// @cp/node-sdk/dist/cjs/models/AdminCreateMerchantCredentialsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAdminCreateMerchantCredentialsDto = instanceOfAdminCreateMerchantCredentialsDto;
exports.AdminCreateMerchantCredentialsDtoFromJSON = AdminCreateMerchantCredentialsDtoFromJSON;
exports.AdminCreateMerchantCredentialsDtoFromJSONTyped = AdminCreateMerchantCredentialsDtoFromJSONTyped;
exports.AdminCreateMerchantCredentialsDtoToJSON = AdminCreateMerchantCredentialsDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the AdminCreateMerchantCredentialsDto interface.
 */
function instanceOfAdminCreateMerchantCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "credentialsName" in value;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
function AdminCreateMerchantCredentialsDtoFromJSON(json) {
    return AdminCreateMerchantCredentialsDtoFromJSONTyped(json, false);
}
function AdminCreateMerchantCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'credentialsName': json['credentialsName'],
        'merchantId': json['merchantId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function AdminCreateMerchantCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'credentialsName': value.credentialsName,
        'merchantId': value.merchantId,
    };
}
