// @cp/node-sdk/dist/cjs/models/AdminMerchantSettingsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAdminMerchantSettingsDto = instanceOfAdminMerchantSettingsDto;
exports.AdminMerchantSettingsDtoFromJSON = AdminMerchantSettingsDtoFromJSON;
exports.AdminMerchantSettingsDtoFromJSONTyped = AdminMerchantSettingsDtoFromJSONTyped;
exports.AdminMerchantSettingsDtoToJSON = AdminMerchantSettingsDtoToJSON;
const runtime_1 = require("../runtime");
const MerchantThemeDto_1 = require("./MerchantThemeDto");
/**
 * Check if a given object implements the AdminMerchantSettingsDto interface.
 */
function instanceOfAdminMerchantSettingsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
function AdminMerchantSettingsDtoFromJSON(json) {
    return AdminMerchantSettingsDtoFromJSONTyped(json, false);
}
function AdminMerchantSettingsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'checkoutTheme': !(0, runtime_1.exists)(json, 'checkoutTheme') ? undefined : (0, MerchantThemeDto_1.MerchantThemeDtoFromJSON)(json['checkoutTheme']),
        'isSplitCFeeEnabled': !(0, runtime_1.exists)(json, 'isSplitCFeeEnabled') ? undefined : json['isSplitCFeeEnabled'],
        'isDebtRepaymentEnabled': !(0, runtime_1.exists)(json, 'isDebtRepaymentEnabled') ? undefined : json['isDebtRepaymentEnabled'],
        'isPartialAuthEnabled': !(0, runtime_1.exists)(json, 'isPartialAuthEnabled') ? undefined : json['isPartialAuthEnabled'],
        'merchantId': json['merchantId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function AdminMerchantSettingsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'checkoutTheme': (0, MerchantThemeDto_1.MerchantThemeDtoToJSON)(value.checkoutTheme),
        'isSplitCFeeEnabled': value.isSplitCFeeEnabled,
        'isDebtRepaymentEnabled': value.isDebtRepaymentEnabled,
        'isPartialAuthEnabled': value.isPartialAuthEnabled,
        'merchantId': value.merchantId,
    };
}
