// @cp/node-sdk/dist/cjs/models/AdminVoidTransactionDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAdminVoidTransactionDto = instanceOfAdminVoidTransactionDto;
exports.AdminVoidTransactionDtoFromJSON = AdminVoidTransactionDtoFromJSON;
exports.AdminVoidTransactionDtoFromJSONTyped = AdminVoidTransactionDtoFromJSONTyped;
exports.AdminVoidTransactionDtoToJSON = AdminVoidTransactionDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the AdminVoidTransactionDto interface.
 */
function instanceOfAdminVoidTransactionDto(value) {
    let isInstance = true;
    isInstance = isInstance && "transactionId" in value;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
function AdminVoidTransactionDtoFromJSON(json) {
    return AdminVoidTransactionDtoFromJSONTyped(json, false);
}
function AdminVoidTransactionDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'transactionId': json['transactionId'],
        'merchantId': json['merchantId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function AdminVoidTransactionDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transactionId': value.transactionId,
        'merchantId': value.merchantId,
    };
}
