// @cp/node-sdk/dist/cjs/models/AuthorizedPaymentInstrumentsEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfAuthorizedPaymentInstrumentsEntity = instanceOfAuthorizedPaymentInstrumentsEntity;
exports.AuthorizedPaymentInstrumentsEntityFromJSON = AuthorizedPaymentInstrumentsEntityFromJSON;
exports.AuthorizedPaymentInstrumentsEntityFromJSONTyped = AuthorizedPaymentInstrumentsEntityFromJSONTyped;
exports.AuthorizedPaymentInstrumentsEntityToJSON = AuthorizedPaymentInstrumentsEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the AuthorizedPaymentInstrumentsEntity interface.
 */
function instanceOfAuthorizedPaymentInstrumentsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "card" in value;
    isInstance = isInstance && "ach" in value;
    return isInstance;
}
function AuthorizedPaymentInstrumentsEntityFromJSON(json) {
    return AuthorizedPaymentInstrumentsEntityFromJSONTyped(json, false);
}
function AuthorizedPaymentInstrumentsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'card': json['card'],
        'ach': json['ach'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function AuthorizedPaymentInstrumentsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'card': value.card,
        'ach': value.ach,
    };
}
