// @cp/node-sdk/dist/cjs/models/BillingDetailsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBillingDetailsDto = instanceOfBillingDetailsDto;
exports.BillingDetailsDtoFromJSON = BillingDetailsDtoFromJSON;
exports.BillingDetailsDtoFromJSONTyped = BillingDetailsDtoFromJSONTyped;
exports.BillingDetailsDtoToJSON = BillingDetailsDtoToJSON;
const runtime_1 = require("../runtime");
const BillingAddressDto_1 = require("./BillingAddressDto");
/**
 * Check if a given object implements the BillingDetailsDto interface.
 */
function instanceOfBillingDetailsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "billingAddress" in value;
    return isInstance;
}
function BillingDetailsDtoFromJSON(json) {
    return BillingDetailsDtoFromJSONTyped(json, false);
}
function BillingDetailsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'email': json['email'],
        'billingAddress': (0, BillingAddressDto_1.BillingAddressDtoFromJSON)(json['billingAddress']),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function BillingDetailsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'billingAddress': (0, BillingAddressDto_1.BillingAddressDtoToJSON)(value.billingAddress),
    };
}
