// @cp/node-sdk/dist/cjs/models/BillingDetailsEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBillingDetailsEntity = instanceOfBillingDetailsEntity;
exports.BillingDetailsEntityFromJSON = BillingDetailsEntityFromJSON;
exports.BillingDetailsEntityFromJSONTyped = BillingDetailsEntityFromJSONTyped;
exports.BillingDetailsEntityToJSON = BillingDetailsEntityToJSON;
const runtime_1 = require("../runtime");
const BillingAddressEntity_1 = require("./BillingAddressEntity");
/**
 * Check if a given object implements the BillingDetailsEntity interface.
 */
function instanceOfBillingDetailsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "billingAddress" in value;
    return isInstance;
}
function BillingDetailsEntityFromJSON(json) {
    return BillingDetailsEntityFromJSONTyped(json, false);
}
function BillingDetailsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'email': json['email'],
        'billingAddress': (0, BillingAddressEntity_1.BillingAddressEntityFromJSON)(json['billingAddress']),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function BillingDetailsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'billingAddress': (0, BillingAddressEntity_1.BillingAddressEntityToJSON)(value.billingAddress),
    };
}
