// @cp/node-sdk/dist/cjs/models/BillingAddressDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBillingAddressDto = instanceOfBillingAddressDto;
exports.BillingAddressDtoFromJSON = BillingAddressDtoFromJSON;
exports.BillingAddressDtoFromJSONTyped = BillingAddressDtoFromJSONTyped;
exports.BillingAddressDtoToJSON = BillingAddressDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the BillingAddressDto interface.
 */
function instanceOfBillingAddressDto(value) {
    let isInstance = true;
    isInstance = isInstance && "city" in value;
    isInstance = isInstance && "streetAddress" in value;
    isInstance = isInstance && "postalCode" in value;
    return isInstance;
}
function BillingAddressDtoFromJSON(json) {
    return BillingAddressDtoFromJSONTyped(json, false);
}
function BillingAddressDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'city': json['city'],
        'streetAddress': json['streetAddress'],
        'postalCode': json['postalCode'],
        'countryCode': !(0, runtime_1.exists)(json, 'countryCode') ? undefined : json['countryCode'],
        'subdivisionCode': !(0, runtime_1.exists)(json, 'subdivisionCode') ? undefined : json['subdivisionCode'],
        'streetAddress2': !(0, runtime_1.exists)(json, 'streetAddress2') ? undefined : json['streetAddress2'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function BillingAddressDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'city': value.city,
        'streetAddress': value.streetAddress,
        'postalCode': value.postalCode,
        'countryCode': value.countryCode,
        'subdivisionCode': value.subdivisionCode,
        'streetAddress2': value.streetAddress2,
    };
}
