// @cp/node-sdk/dist/cjs/models/BuyersSearchEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBuyersSearchEntity = instanceOfBuyersSearchEntity;
exports.BuyersSearchEntityFromJSON = BuyersSearchEntityFromJSON;
exports.BuyersSearchEntityFromJSONTyped = BuyersSearchEntityFromJSONTyped;
exports.BuyersSearchEntityToJSON = BuyersSearchEntityToJSON;
const runtime_1 = require("../runtime");
const BuyersServiceEntity_1 = require("./BuyersServiceEntity");
/**
 * Check if a given object implements the BuyersSearchEntity interface.
 */
function instanceOfBuyersSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "buyers" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
function BuyersSearchEntityFromJSON(json) {
    return BuyersSearchEntityFromJSONTyped(json, false);
}
function BuyersSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'buyers': (json['buyers'].map(BuyersServiceEntity_1.BuyersServiceEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !(0, runtime_1.exists)(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function BuyersSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'buyers': (value.buyers.map(BuyersServiceEntity_1.BuyersServiceEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
