// @cp/node-sdk/dist/cjs/models/BuyersServiceEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBuyersServiceEntity = instanceOfBuyersServiceEntity;
exports.BuyersServiceEntityFromJSON = BuyersServiceEntityFromJSON;
exports.BuyersServiceEntityFromJSONTyped = BuyersServiceEntityFromJSONTyped;
exports.BuyersServiceEntityToJSON = BuyersServiceEntityToJSON;
const runtime_1 = require("../runtime");
const BuyersAddressEntity_1 = require("./BuyersAddressEntity");
/**
 * Check if a given object implements the BuyersServiceEntity interface.
 */
function instanceOfBuyersServiceEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "buyerId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "address" in value;
    return isInstance;
}
function BuyersServiceEntityFromJSON(json) {
    return BuyersServiceEntityFromJSONTyped(json, false);
}
function BuyersServiceEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'buyerId': json['buyerId'],
        'merchantId': json['merchantId'],
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'email': json['email'],
        'address': (0, BuyersAddressEntity_1.BuyersAddressEntityFromJSON)(json['address']),
        'buyerIntegrationId': !(0, runtime_1.exists)(json, 'buyerIntegrationId') ? undefined : json['buyerIntegrationId'],
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'createdAt': !(0, runtime_1.exists)(json, 'createdAt') ? undefined : (new Date(json['createdAt'])),
        'updatedAt': !(0, runtime_1.exists)(json, 'updatedAt') ? undefined : (new Date(json['updatedAt'])),
        'deleted': !(0, runtime_1.exists)(json, 'deleted') ? undefined : json['deleted'],
        'deletedAt': !(0, runtime_1.exists)(json, 'deletedAt') ? undefined : (new Date(json['deletedAt'])),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function BuyersServiceEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'buyerId': value.buyerId,
        'merchantId': value.merchantId,
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'address': (0, BuyersAddressEntity_1.BuyersAddressEntityToJSON)(value.address),
        'buyerIntegrationId': value.buyerIntegrationId,
        'phoneNumber': value.phoneNumber,
        'externalBuyerId': value.externalBuyerId,
        'createdAt': value.createdAt === undefined ? undefined : (value.createdAt.toISOString()),
        'updatedAt': value.updatedAt === undefined ? undefined : (value.updatedAt.toISOString()),
        'deleted': value.deleted,
        'deletedAt': value.deletedAt === undefined ? undefined : (value.deletedAt.toISOString()),
    };
}
