// @cp/node-sdk/dist/cjs/models/BuyerEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfBuyerEntity = instanceOfBuyerEntity;
exports.BuyerEntityFromJSON = BuyerEntityFromJSON;
exports.BuyerEntityFromJSONTyped = BuyerEntityFromJSONTyped;
exports.BuyerEntityToJSON = BuyerEntityToJSON;
const runtime_1 = require("../runtime");
const AddressEntity_1 = require("./AddressEntity");
/**
 * Check if a given object implements the BuyerEntity interface.
 */
function instanceOfBuyerEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "buyerId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "address" in value;
    return isInstance;
}
function BuyerEntityFromJSON(json) {
    return BuyerEntityFromJSONTyped(json, false);
}
function BuyerEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'buyerId': json['buyerId'],
        'merchantId': json['merchantId'],
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'email': json['email'],
        'address': (0, AddressEntity_1.AddressEntityFromJSON)(json['address']),
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'id': !(0, runtime_1.exists)(json, 'id') ? undefined : json['id'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function BuyerEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'buyerId': value.buyerId,
        'merchantId': value.merchantId,
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'address': (0, AddressEntity_1.AddressEntityToJSON)(value.address),
        'phoneNumber': value.phoneNumber,
        'externalBuyerId': value.externalBuyerId,
        'id': value.id,
    };
}
