// @cp/node-sdk/dist/cjs/models/CaptureTransactionDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCaptureTransactionDto = instanceOfCaptureTransactionDto;
exports.CaptureTransactionDtoFromJSON = CaptureTransactionDtoFromJSON;
exports.CaptureTransactionDtoFromJSONTyped = CaptureTransactionDtoFromJSONTyped;
exports.CaptureTransactionDtoToJSON = CaptureTransactionDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CaptureTransactionDto interface.
 */
function instanceOfCaptureTransactionDto(value) {
    let isInstance = true;
    isInstance = isInstance && "transactionId" in value;
    isInstance = isInstance && "amount" in value;
    return isInstance;
}
function CaptureTransactionDtoFromJSON(json) {
    return CaptureTransactionDtoFromJSONTyped(json, false);
}
function CaptureTransactionDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'transactionId': json['transactionId'],
        'amount': json['amount'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CaptureTransactionDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transactionId': value.transactionId,
        'amount': value.amount,
    };
}
