// @cp/node-sdk/dist/cjs/models/CardDetailsServiceEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CardDetailsServiceEntityCardTypeEnum = void 0;
exports.instanceOfCardDetailsServiceEntity = instanceOfCardDetailsServiceEntity;
exports.CardDetailsServiceEntityFromJSON = CardDetailsServiceEntityFromJSON;
exports.CardDetailsServiceEntityFromJSONTyped = CardDetailsServiceEntityFromJSONTyped;
exports.CardDetailsServiceEntityToJSON = CardDetailsServiceEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CardDetailsServiceEntityCardTypeEnum = {
    Credit: 'CREDIT',
    Debit: 'DEBIT',
    Prepaid: 'PREPAID',
    Other: 'OTHER'
};
/**
 * Check if a given object implements the CardDetailsServiceEntity interface.
 */
function instanceOfCardDetailsServiceEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "cardType" in value;
    isInstance = isInstance && "bin" in value;
    isInstance = isInstance && "scheme" in value;
    isInstance = isInstance && "label" in value;
    isInstance = isInstance && "expirationDate" in value;
    return isInstance;
}
function CardDetailsServiceEntityFromJSON(json) {
    return CardDetailsServiceEntityFromJSONTyped(json, false);
}
function CardDetailsServiceEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'cardType': json['cardType'],
        'bin': json['bin'],
        'scheme': json['scheme'],
        'label': json['label'],
        'expirationDate': json['expirationDate'],
        'accountHolderName': !(0, runtime_1.exists)(json, 'accountHolderName') ? undefined : json['accountHolderName'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CardDetailsServiceEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'cardType': value.cardType,
        'bin': value.bin,
        'scheme': value.scheme,
        'label': value.label,
        'expirationDate': value.expirationDate,
        'accountHolderName': value.accountHolderName,
    };
}
