// @cp/node-sdk/dist/cjs/models/CFeeEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CFeeEntityTypeEnum = void 0;
exports.instanceOfCFeeEntity = instanceOfCFeeEntity;
exports.CFeeEntityFromJSON = CFeeEntityFromJSON;
exports.CFeeEntityFromJSONTyped = CFeeEntityFromJSONTyped;
exports.CFeeEntityToJSON = CFeeEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CFeeEntityTypeEnum = {
    Percentage: 'PERCENTAGE',
    Fixed: 'FIXED'
};
/**
 * Check if a given object implements the CFeeEntity interface.
 */
function instanceOfCFeeEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "rate" in value;
    return isInstance;
}
function CFeeEntityFromJSON(json) {
    return CFeeEntityFromJSONTyped(json, false);
}
function CFeeEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'type': json['type'],
        'rate': json['rate'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CFeeEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'rate': value.rate,
    };
}
