// @cp/node-sdk/dist/cjs/models/ChangeNotificationsSearchEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfChangeNotificationsSearchEntity = instanceOfChangeNotificationsSearchEntity;
exports.ChangeNotificationsSearchEntityFromJSON = ChangeNotificationsSearchEntityFromJSON;
exports.ChangeNotificationsSearchEntityFromJSONTyped = ChangeNotificationsSearchEntityFromJSONTyped;
exports.ChangeNotificationsSearchEntityToJSON = ChangeNotificationsSearchEntityToJSON;
const runtime_1 = require("../runtime");
const ChangeNotificationEntity_1 = require("./ChangeNotificationEntity");
/**
 * Check if a given object implements the ChangeNotificationsSearchEntity interface.
 */
function instanceOfChangeNotificationsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "changeNotifications" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
function ChangeNotificationsSearchEntityFromJSON(json) {
    return ChangeNotificationsSearchEntityFromJSONTyped(json, false);
}
function ChangeNotificationsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'changeNotifications': (json['changeNotifications'].map(ChangeNotificationEntity_1.ChangeNotificationEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !(0, runtime_1.exists)(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function ChangeNotificationsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'changeNotifications': (value.changeNotifications.map(ChangeNotificationEntity_1.ChangeNotificationEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
