// @cp/node-sdk/dist/cjs/models/CheckoutSessionItemBaseEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCheckoutSessionItemBaseEntity = instanceOfCheckoutSessionItemBaseEntity;
exports.CheckoutSessionItemBaseEntityFromJSON = CheckoutSessionItemBaseEntityFromJSON;
exports.CheckoutSessionItemBaseEntityFromJSONTyped = CheckoutSessionItemBaseEntityFromJSONTyped;
exports.CheckoutSessionItemBaseEntityToJSON = CheckoutSessionItemBaseEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CheckoutSessionItemBaseEntity interface.
 */
function instanceOfCheckoutSessionItemBaseEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "itemId" in value;
    isInstance = isInstance && "sessionId" in value;
    isInstance = isInstance && "description" in value;
    isInstance = isInstance && "quantity" in value;
    isInstance = isInstance && "unitPrice" in value;
    isInstance = isInstance && "total" in value;
    return isInstance;
}
function CheckoutSessionItemBaseEntityFromJSON(json) {
    return CheckoutSessionItemBaseEntityFromJSONTyped(json, false);
}
function CheckoutSessionItemBaseEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'itemId': json['itemId'],
        'sessionId': json['sessionId'],
        'description': json['description'],
        'quantity': json['quantity'],
        'unitPrice': json['unitPrice'],
        'total': json['total'],
        'subtotal': !(0, runtime_1.exists)(json, 'subtotal') ? undefined : json['subtotal'],
        'discountTotal': !(0, runtime_1.exists)(json, 'discountTotal') ? undefined : json['discountTotal'],
        'taxTotal': !(0, runtime_1.exists)(json, 'taxTotal') ? undefined : json['taxTotal'],
        'feesTotal': !(0, runtime_1.exists)(json, 'feesTotal') ? undefined : json['feesTotal'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CheckoutSessionItemBaseEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'itemId': value.itemId,
        'sessionId': value.sessionId,
        'description': value.description,
        'quantity': value.quantity,
        'unitPrice': value.unitPrice,
        'total': value.total,
        'subtotal': value.subtotal,
        'discountTotal': value.discountTotal,
        'taxTotal': value.taxTotal,
        'feesTotal': value.feesTotal,
    };
}
