// @cp/node-sdk/dist/cjs/models/CheckoutSessionEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CheckoutSessionEntityTransactionInitiatorEnum = exports.CheckoutSessionEntityStatusEnum = exports.CheckoutSessionEntityUiModeEnum = void 0;
exports.instanceOfCheckoutSessionEntity = instanceOfCheckoutSessionEntity;
exports.CheckoutSessionEntityFromJSON = CheckoutSessionEntityFromJSON;
exports.CheckoutSessionEntityFromJSONTyped = CheckoutSessionEntityFromJSONTyped;
exports.CheckoutSessionEntityToJSON = CheckoutSessionEntityToJSON;
const runtime_1 = require("../runtime");
const AvailablePaymentMethodEntity_1 = require("./AvailablePaymentMethodEntity");
const CheckoutSessionItemEntity_1 = require("./CheckoutSessionItemEntity");
const TagEntity_1 = require("./TagEntity");
/**
 * @export
 */
exports.CheckoutSessionEntityUiModeEnum = {
    Hosted: 'HOSTED',
    Embedded: 'EMBEDDED'
};
/**
 * @export
 */
exports.CheckoutSessionEntityStatusEnum = {
    Pending: 'PENDING',
    Processing: 'PROCESSING',
    Expired: 'EXPIRED',
    Completed: 'COMPLETED'
};
/**
 * @export
 */
exports.CheckoutSessionEntityTransactionInitiatorEnum = {
    Mit: 'MIT',
    Cit: 'CIT',
    Sit: 'SIT'
};
/**
 * Check if a given object implements the CheckoutSessionEntity interface.
 */
function instanceOfCheckoutSessionEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "sessionId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "uiMode" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "currency" in value;
    isInstance = isInstance && "total" in value;
    isInstance = isInstance && "subtotal" in value;
    isInstance = isInstance && "feesTotal" in value;
    isInstance = isInstance && "taxTotal" in value;
    isInstance = isInstance && "discountTotal" in value;
    isInstance = isInstance && "shippingTotal" in value;
    isInstance = isInstance && "availablePaymentMethods" in value;
    isInstance = isInstance && "items" in value;
    isInstance = isInstance && "successRedirectUrl" in value;
    isInstance = isInstance && "cancelRedirectUrl" in value;
    isInstance = isInstance && "country" in value;
    isInstance = isInstance && "canExtend" in value;
    isInstance = isInstance && "excludeCFee" in value;
    return isInstance;
}
function CheckoutSessionEntityFromJSON(json) {
    return CheckoutSessionEntityFromJSONTyped(json, false);
}
function CheckoutSessionEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'sessionId': json['sessionId'],
        'merchantId': json['merchantId'],
        'uiMode': json['uiMode'],
        'status': json['status'],
        'currency': json['currency'],
        'total': json['total'],
        'subtotal': json['subtotal'],
        'feesTotal': json['feesTotal'],
        'taxTotal': json['taxTotal'],
        'discountTotal': json['discountTotal'],
        'shippingTotal': json['shippingTotal'],
        'availablePaymentMethods': (json['availablePaymentMethods'].map(AvailablePaymentMethodEntity_1.AvailablePaymentMethodEntityFromJSON)),
        'items': (json['items'].map(CheckoutSessionItemEntity_1.CheckoutSessionItemEntityFromJSON)),
        'successRedirectUrl': json['successRedirectUrl'],
        'cancelRedirectUrl': json['cancelRedirectUrl'],
        'country': json['country'],
        'expiresAt': !(0, runtime_1.exists)(json, 'expiresAt') ? undefined : (new Date(json['expiresAt'])),
        'expiredAt': !(0, runtime_1.exists)(json, 'expiredAt') ? undefined : (new Date(json['expiredAt'])),
        'canExtend': json['canExtend'],
        'expiredReason': !(0, runtime_1.exists)(json, 'expiredReason') ? undefined : json['expiredReason'],
        'tags': !(0, runtime_1.exists)(json, 'tags') ? undefined : (json['tags'].map(TagEntity_1.TagEntityFromJSON)),
        'canStore': !(0, runtime_1.exists)(json, 'canStore') ? undefined : json['canStore'],
        'buyerId': !(0, runtime_1.exists)(json, 'buyerId') ? undefined : json['buyerId'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'transactionInitiator': !(0, runtime_1.exists)(json, 'transactionInitiator') ? undefined : json['transactionInitiator'],
        'redirectUrl': !(0, runtime_1.exists)(json, 'redirectUrl') ? undefined : json['redirectUrl'],
        'externalTransactionId': !(0, runtime_1.exists)(json, 'externalTransactionId') ? undefined : json['externalTransactionId'],
        'secCode': !(0, runtime_1.exists)(json, 'secCode') ? undefined : json['secCode'],
        'excludeCFee': json['excludeCFee'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CheckoutSessionEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'sessionId': value.sessionId,
        'merchantId': value.merchantId,
        'uiMode': value.uiMode,
        'status': value.status,
        'currency': value.currency,
        'total': value.total,
        'subtotal': value.subtotal,
        'feesTotal': value.feesTotal,
        'taxTotal': value.taxTotal,
        'discountTotal': value.discountTotal,
        'shippingTotal': value.shippingTotal,
        'availablePaymentMethods': (value.availablePaymentMethods.map(AvailablePaymentMethodEntity_1.AvailablePaymentMethodEntityToJSON)),
        'items': (value.items.map(CheckoutSessionItemEntity_1.CheckoutSessionItemEntityToJSON)),
        'successRedirectUrl': value.successRedirectUrl,
        'cancelRedirectUrl': value.cancelRedirectUrl,
        'country': value.country,
        'expiresAt': value.expiresAt === undefined ? undefined : (value.expiresAt.toISOString()),
        'expiredAt': value.expiredAt === undefined ? undefined : (value.expiredAt.toISOString()),
        'canExtend': value.canExtend,
        'expiredReason': value.expiredReason,
        'tags': value.tags === undefined ? undefined : (value.tags.map(TagEntity_1.TagEntityToJSON)),
        'canStore': value.canStore,
        'buyerId': value.buyerId,
        'externalBuyerId': value.externalBuyerId,
        'transactionInitiator': value.transactionInitiator,
        'redirectUrl': value.redirectUrl,
        'externalTransactionId': value.externalTransactionId,
        'secCode': value.secCode,
        'excludeCFee': value.excludeCFee,
    };
}
