// @cp/node-sdk/dist/cjs/models/CreateServicePaymentMethodDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateServicePaymentMethodDtoAccountTypeEnum = exports.CreateServicePaymentMethodDtoMethodEnum = void 0;
exports.instanceOfCreateServicePaymentMethodDto = instanceOfCreateServicePaymentMethodDto;
exports.CreateServicePaymentMethodDtoFromJSON = CreateServicePaymentMethodDtoFromJSON;
exports.CreateServicePaymentMethodDtoFromJSONTyped = CreateServicePaymentMethodDtoFromJSONTyped;
exports.CreateServicePaymentMethodDtoToJSON = CreateServicePaymentMethodDtoToJSON;
const runtime_1 = require("../runtime");
const BillingDetailsDto_1 = require("./BillingDetailsDto");
/**
 * @export
 */
exports.CreateServicePaymentMethodDtoMethodEnum = {
    Card: 'CARD',
    Bank: 'BANK'
};
/**
 * @export
 */
exports.CreateServicePaymentMethodDtoAccountTypeEnum = {
    Checking: 'CHECKING',
    Savings: 'SAVINGS'
};
/**
 * Check if a given object implements the CreateServicePaymentMethodDto interface.
 */
function instanceOfCreateServicePaymentMethodDto(value) {
    let isInstance = true;
    isInstance = isInstance && "method" in value;
    return isInstance;
}
function CreateServicePaymentMethodDtoFromJSON(json) {
    return CreateServicePaymentMethodDtoFromJSONTyped(json, false);
}
function CreateServicePaymentMethodDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'method': json['method'],
        'number': !(0, runtime_1.exists)(json, 'number') ? undefined : json['number'],
        'expirationDate': !(0, runtime_1.exists)(json, 'expirationDate') ? undefined : json['expirationDate'],
        'abaNumber': !(0, runtime_1.exists)(json, 'abaNumber') ? undefined : json['abaNumber'],
        'accountNumber': !(0, runtime_1.exists)(json, 'accountNumber') ? undefined : json['accountNumber'],
        'accountType': !(0, runtime_1.exists)(json, 'accountType') ? undefined : json['accountType'],
        'accountHolderName': !(0, runtime_1.exists)(json, 'accountHolderName') ? undefined : json['accountHolderName'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'buyerId': !(0, runtime_1.exists)(json, 'buyerId') ? undefined : json['buyerId'],
        'externalPaymentMethodId': !(0, runtime_1.exists)(json, 'externalPaymentMethodId') ? undefined : json['externalPaymentMethodId'],
        'billingDetails': !(0, runtime_1.exists)(json, 'billingDetails') ? undefined : (0, BillingDetailsDto_1.BillingDetailsDtoFromJSON)(json['billingDetails']),
        'isCustomerStored': !(0, runtime_1.exists)(json, 'isCustomerStored') ? undefined : json['isCustomerStored'],
        'merchantId': !(0, runtime_1.exists)(json, 'merchantId') ? undefined : json['merchantId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateServicePaymentMethodDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'method': value.method,
        'number': value.number,
        'expirationDate': value.expirationDate,
        'abaNumber': value.abaNumber,
        'accountNumber': value.accountNumber,
        'accountType': value.accountType,
        'accountHolderName': value.accountHolderName,
        'externalBuyerId': value.externalBuyerId,
        'buyerId': value.buyerId,
        'externalPaymentMethodId': value.externalPaymentMethodId,
        'billingDetails': (0, BillingDetailsDto_1.BillingDetailsDtoToJSON)(value.billingDetails),
        'isCustomerStored': value.isCustomerStored,
        'merchantId': value.merchantId,
    };
}
