// @cp/node-sdk/dist/cjs/models/CreateMerchantCredentialsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateMerchantCredentialsDto = instanceOfCreateMerchantCredentialsDto;
exports.CreateMerchantCredentialsDtoFromJSON = CreateMerchantCredentialsDtoFromJSON;
exports.CreateMerchantCredentialsDtoFromJSONTyped = CreateMerchantCredentialsDtoFromJSONTyped;
exports.CreateMerchantCredentialsDtoToJSON = CreateMerchantCredentialsDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CreateMerchantCredentialsDto interface.
 */
function instanceOfCreateMerchantCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "credentialsName" in value;
    return isInstance;
}
function CreateMerchantCredentialsDtoFromJSON(json) {
    return CreateMerchantCredentialsDtoFromJSONTyped(json, false);
}
function CreateMerchantCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'credentialsName': json['credentialsName'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateMerchantCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'credentialsName': value.credentialsName,
    };
}
