// @cp/node-sdk/dist/cjs/models/CreateEventDomainDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateEventDomainDto = instanceOfCreateEventDomainDto;
exports.CreateEventDomainDtoFromJSON = CreateEventDomainDtoFromJSON;
exports.CreateEventDomainDtoFromJSONTyped = CreateEventDomainDtoFromJSONTyped;
exports.CreateEventDomainDtoToJSON = CreateEventDomainDtoToJSON;
const runtime_1 = require("../runtime");
const CreateEventConfigDto_1 = require("./CreateEventConfigDto");
/**
 * Check if a given object implements the CreateEventDomainDto interface.
 */
function instanceOfCreateEventDomainDto(value) {
    let isInstance = true;
    isInstance = isInstance && "transactions" in value;
    isInstance = isInstance && "buyers" in value;
    isInstance = isInstance && "users" in value;
    isInstance = isInstance && "paymentMethods" in value;
    isInstance = isInstance && "merchants" in value;
    return isInstance;
}
function CreateEventDomainDtoFromJSON(json) {
    return CreateEventDomainDtoFromJSONTyped(json, false);
}
function CreateEventDomainDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'transactions': (0, CreateEventConfigDto_1.CreateEventConfigDtoFromJSON)(json['transactions']),
        'buyers': (0, CreateEventConfigDto_1.CreateEventConfigDtoFromJSON)(json['buyers']),
        'users': (0, CreateEventConfigDto_1.CreateEventConfigDtoFromJSON)(json['users']),
        'paymentMethods': (0, CreateEventConfigDto_1.CreateEventConfigDtoFromJSON)(json['paymentMethods']),
        'merchants': (0, CreateEventConfigDto_1.CreateEventConfigDtoFromJSON)(json['merchants']),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateEventDomainDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transactions': (0, CreateEventConfigDto_1.CreateEventConfigDtoToJSON)(value.transactions),
        'buyers': (0, CreateEventConfigDto_1.CreateEventConfigDtoToJSON)(value.buyers),
        'users': (0, CreateEventConfigDto_1.CreateEventConfigDtoToJSON)(value.users),
        'paymentMethods': (0, CreateEventConfigDto_1.CreateEventConfigDtoToJSON)(value.paymentMethods),
        'merchants': (0, CreateEventConfigDto_1.CreateEventConfigDtoToJSON)(value.merchants),
    };
}
