// @cp/node-sdk/dist/cjs/models/CreateWebhookDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateWebhookDto = instanceOfCreateWebhookDto;
exports.CreateWebhookDtoFromJSON = CreateWebhookDtoFromJSON;
exports.CreateWebhookDtoFromJSONTyped = CreateWebhookDtoFromJSONTyped;
exports.CreateWebhookDtoToJSON = CreateWebhookDtoToJSON;
const runtime_1 = require("../runtime");
const CreateEventDomainDto_1 = require("./CreateEventDomainDto");
/**
 * Check if a given object implements the CreateWebhookDto interface.
 */
function instanceOfCreateWebhookDto(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "url" in value;
    isInstance = isInstance && "eventDomains" in value;
    return isInstance;
}
function CreateWebhookDtoFromJSON(json) {
    return CreateWebhookDtoFromJSONTyped(json, false);
}
function CreateWebhookDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': json['merchantId'],
        'url': json['url'],
        'eventDomains': (0, CreateEventDomainDto_1.CreateEventDomainDtoFromJSON)(json['eventDomains']),
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'username': !(0, runtime_1.exists)(json, 'username') ? undefined : json['username'],
        'password': !(0, runtime_1.exists)(json, 'password') ? undefined : json['password'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateWebhookDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'url': value.url,
        'eventDomains': (0, CreateEventDomainDto_1.CreateEventDomainDtoToJSON)(value.eventDomains),
        'name': value.name,
        'description': value.description,
        'username': value.username,
        'password': value.password,
    };
}
