// @cp/node-sdk/dist/cjs/models/CreateMerchantAccountWithoutMerchantIdDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateMerchantAccountWithoutMerchantIdDtoCurrenciesEnum = exports.CreateMerchantAccountWithoutMerchantIdDtoCardSchemesEnum = exports.CreateMerchantAccountWithoutMerchantIdDtoCardTypesEnum = exports.CreateMerchantAccountWithoutMerchantIdDtoPspEnum = exports.CreateMerchantAccountWithoutMerchantIdDtoMethodEnum = void 0;
exports.instanceOfCreateMerchantAccountWithoutMerchantIdDto = instanceOfCreateMerchantAccountWithoutMerchantIdDto;
exports.CreateMerchantAccountWithoutMerchantIdDtoFromJSON = CreateMerchantAccountWithoutMerchantIdDtoFromJSON;
exports.CreateMerchantAccountWithoutMerchantIdDtoFromJSONTyped = CreateMerchantAccountWithoutMerchantIdDtoFromJSONTyped;
exports.CreateMerchantAccountWithoutMerchantIdDtoToJSON = CreateMerchantAccountWithoutMerchantIdDtoToJSON;
const runtime_1 = require("../runtime");
const CreateCFeeDto_1 = require("./CreateCFeeDto");
const CreateMerchantConnectorWithoutMerchantAccountIdDto_1 = require("./CreateMerchantConnectorWithoutMerchantAccountIdDto");
/**
 * @export
 */
exports.CreateMerchantAccountWithoutMerchantIdDtoMethodEnum = {
    Card: 'CARD',
    Ach: 'ACH'
};
/**
 * @export
 */
exports.CreateMerchantAccountWithoutMerchantIdDtoPspEnum = {
    Tsys: 'TSYS',
    Nashville: 'NASHVILLE',
    North: 'NORTH',
    ChannelPayments: 'CHANNEL_PAYMENTS'
};
/**
 * @export
 */
exports.CreateMerchantAccountWithoutMerchantIdDtoCardTypesEnum = {
    Credit: 'CREDIT',
    Debit: 'DEBIT',
    Prepaid: 'PREPAID'
};
/**
 * @export
 */
exports.CreateMerchantAccountWithoutMerchantIdDtoCardSchemesEnum = {
    Visa: 'VISA',
    Mastercard: 'MASTERCARD',
    Amex: 'AMEX',
    Discover: 'DISCOVER',
    Jcb: 'JCB',
    DinersClub: 'DINERS_CLUB',
    Maestro: 'MAESTRO'
};
/**
 * @export
 */
exports.CreateMerchantAccountWithoutMerchantIdDtoCurrenciesEnum = {
    Usd: 'USD',
    Aud: 'AUD',
    Cad: 'CAD'
};
/**
 * Check if a given object implements the CreateMerchantAccountWithoutMerchantIdDto interface.
 */
function instanceOfCreateMerchantAccountWithoutMerchantIdDto(value) {
    let isInstance = true;
    isInstance = isInstance && "issuedMid" in value;
    isInstance = isInstance && "method" in value;
    isInstance = isInstance && "psp" in value;
    isInstance = isInstance && "cFee" in value;
    isInstance = isInstance && "connectors" in value;
    return isInstance;
}
function CreateMerchantAccountWithoutMerchantIdDtoFromJSON(json) {
    return CreateMerchantAccountWithoutMerchantIdDtoFromJSONTyped(json, false);
}
function CreateMerchantAccountWithoutMerchantIdDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'issuedMid': json['issuedMid'],
        'method': json['method'],
        'psp': json['psp'],
        'cFee': (0, CreateCFeeDto_1.CreateCFeeDtoFromJSON)(json['cFee']),
        'cardTypes': !(0, runtime_1.exists)(json, 'cardTypes') ? undefined : json['cardTypes'],
        'cardSchemes': !(0, runtime_1.exists)(json, 'cardSchemes') ? undefined : json['cardSchemes'],
        'currencies': !(0, runtime_1.exists)(json, 'currencies') ? undefined : json['currencies'],
        'connectors': (json['connectors'].map(CreateMerchantConnectorWithoutMerchantAccountIdDto_1.CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSON)),
        'mcc': !(0, runtime_1.exists)(json, 'mcc') ? undefined : json['mcc'],
        'bin': !(0, runtime_1.exists)(json, 'bin') ? undefined : json['bin'],
        'chain': !(0, runtime_1.exists)(json, 'chain') ? undefined : json['chain'],
        'store': !(0, runtime_1.exists)(json, 'store') ? undefined : json['store'],
        'agent': !(0, runtime_1.exists)(json, 'agent') ? undefined : json['agent'],
        'isDefault': !(0, runtime_1.exists)(json, 'isDefault') ? undefined : json['isDefault'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateMerchantAccountWithoutMerchantIdDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'issuedMid': value.issuedMid,
        'method': value.method,
        'psp': value.psp,
        'cFee': (0, CreateCFeeDto_1.CreateCFeeDtoToJSON)(value.cFee),
        'cardTypes': value.cardTypes,
        'cardSchemes': value.cardSchemes,
        'currencies': value.currencies,
        'connectors': (value.connectors.map(CreateMerchantConnectorWithoutMerchantAccountIdDto_1.CreateMerchantConnectorWithoutMerchantAccountIdDtoToJSON)),
        'mcc': value.mcc,
        'bin': value.bin,
        'chain': value.chain,
        'store': value.store,
        'agent': value.agent,
        'isDefault': value.isDefault,
    };
}
