// @cp/node-sdk/dist/cjs/models/CreateCFeeDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCFeeDtoTypeEnum = void 0;
exports.instanceOfCreateCFeeDto = instanceOfCreateCFeeDto;
exports.CreateCFeeDtoFromJSON = CreateCFeeDtoFromJSON;
exports.CreateCFeeDtoFromJSONTyped = CreateCFeeDtoFromJSONTyped;
exports.CreateCFeeDtoToJSON = CreateCFeeDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CreateCFeeDtoTypeEnum = {
    Percentage: 'PERCENTAGE',
    Fixed: 'FIXED'
};
/**
 * Check if a given object implements the CreateCFeeDto interface.
 */
function instanceOfCreateCFeeDto(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "rate" in value;
    return isInstance;
}
function CreateCFeeDtoFromJSON(json) {
    return CreateCFeeDtoFromJSONTyped(json, false);
}
function CreateCFeeDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'type': json['type'],
        'rate': json['rate'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateCFeeDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'rate': value.rate,
    };
}
