// @cp/node-sdk/dist/cjs/models/CreateTransactionsReportsTemplateDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTransactionsReportsTemplateDtoDateRangeEnum = exports.CreateTransactionsReportsTemplateDtoOrderEnum = void 0;
exports.instanceOfCreateTransactionsReportsTemplateDto = instanceOfCreateTransactionsReportsTemplateDto;
exports.CreateTransactionsReportsTemplateDtoFromJSON = CreateTransactionsReportsTemplateDtoFromJSON;
exports.CreateTransactionsReportsTemplateDtoFromJSONTyped = CreateTransactionsReportsTemplateDtoFromJSONTyped;
exports.CreateTransactionsReportsTemplateDtoToJSON = CreateTransactionsReportsTemplateDtoToJSON;
const runtime_1 = require("../runtime");
const CreateTransactionsReportsFiltersDto_1 = require("./CreateTransactionsReportsFiltersDto");
/**
 * @export
 */
exports.CreateTransactionsReportsTemplateDtoOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * @export
 */
exports.CreateTransactionsReportsTemplateDtoDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the CreateTransactionsReportsTemplateDto interface.
 */
function instanceOfCreateTransactionsReportsTemplateDto(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "dateRange" in value;
    isInstance = isInstance && "startTime" in value;
    isInstance = isInstance && "endTime" in value;
    isInstance = isInstance && "columns" in value;
    isInstance = isInstance && "transactionsFilters" in value;
    return isInstance;
}
function CreateTransactionsReportsTemplateDtoFromJSON(json) {
    return CreateTransactionsReportsTemplateDtoFromJSONTyped(json, false);
}
function CreateTransactionsReportsTemplateDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': json['name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'order': !(0, runtime_1.exists)(json, 'order') ? undefined : json['order'],
        'dateRange': json['dateRange'],
        'startTime': json['startTime'],
        'endTime': json['endTime'],
        'columns': json['columns'],
        'transactionsFilters': (0, CreateTransactionsReportsFiltersDto_1.CreateTransactionsReportsFiltersDtoFromJSON)(json['transactionsFilters']),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateTransactionsReportsTemplateDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'order': value.order,
        'dateRange': value.dateRange,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'columns': value.columns,
        'transactionsFilters': (0, CreateTransactionsReportsFiltersDto_1.CreateTransactionsReportsFiltersDtoToJSON)(value.transactionsFilters),
    };
}
