// @cp/node-sdk/dist/cjs/models/CreateInitialUserDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateInitialUserDtoRoleEnum = void 0;
exports.instanceOfCreateInitialUserDto = instanceOfCreateInitialUserDto;
exports.CreateInitialUserDtoFromJSON = CreateInitialUserDtoFromJSON;
exports.CreateInitialUserDtoFromJSONTyped = CreateInitialUserDtoFromJSONTyped;
exports.CreateInitialUserDtoToJSON = CreateInitialUserDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CreateInitialUserDtoRoleEnum = {
    User: 'user',
    Agent: 'agent',
    Admin: 'admin',
    SuperAdmin: 'superAdmin'
};
/**
 * Check if a given object implements the CreateInitialUserDto interface.
 */
function instanceOfCreateInitialUserDto(value) {
    let isInstance = true;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "role" in value;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
function CreateInitialUserDtoFromJSON(json) {
    return CreateInitialUserDtoFromJSONTyped(json, false);
}
function CreateInitialUserDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'email': json['email'],
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'role': json['role'],
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'merchantId': json['merchantId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateInitialUserDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'email': value.email,
        'givenName': value.givenName,
        'familyName': value.familyName,
        'role': value.role,
        'phoneNumber': value.phoneNumber,
        'merchantId': value.merchantId,
    };
}
