// @cp/node-sdk/dist/cjs/models/CreateFeatureDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateFeatureDto = instanceOfCreateFeatureDto;
exports.CreateFeatureDtoFromJSON = CreateFeatureDtoFromJSON;
exports.CreateFeatureDtoFromJSONTyped = CreateFeatureDtoFromJSONTyped;
exports.CreateFeatureDtoToJSON = CreateFeatureDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CreateFeatureDto interface.
 */
function instanceOfCreateFeatureDto(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "description" in value;
    return isInstance;
}
function CreateFeatureDtoFromJSON(json) {
    return CreateFeatureDtoFromJSONTyped(json, false);
}
function CreateFeatureDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': json['name'],
        'description': json['description'],
        'version': !(0, runtime_1.exists)(json, 'version') ? undefined : json['version'],
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateFeatureDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'version': value.version,
        'enabled': value.enabled,
    };
}
