// @cp/node-sdk/dist/cjs/models/CreateFeatureTargetDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateFeatureTargetDtoTargetTypeEnum = void 0;
exports.instanceOfCreateFeatureTargetDto = instanceOfCreateFeatureTargetDto;
exports.CreateFeatureTargetDtoFromJSON = CreateFeatureTargetDtoFromJSON;
exports.CreateFeatureTargetDtoFromJSONTyped = CreateFeatureTargetDtoFromJSONTyped;
exports.CreateFeatureTargetDtoToJSON = CreateFeatureTargetDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CreateFeatureTargetDtoTargetTypeEnum = {
    Merchant: 'merchant',
    User: 'user',
    PaymentMethod: 'payment_method',
    Buyer: 'buyer',
    Webhook: 'webhook',
    Job: 'job'
};
/**
 * Check if a given object implements the CreateFeatureTargetDto interface.
 */
function instanceOfCreateFeatureTargetDto(value) {
    let isInstance = true;
    isInstance = isInstance && "featureId" in value;
    isInstance = isInstance && "targetEntityId" in value;
    return isInstance;
}
function CreateFeatureTargetDtoFromJSON(json) {
    return CreateFeatureTargetDtoFromJSONTyped(json, false);
}
function CreateFeatureTargetDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'featureId': json['featureId'],
        'targetEntityId': json['targetEntityId'],
        'targetType': !(0, runtime_1.exists)(json, 'targetType') ? undefined : json['targetType'],
        'rolloutPercentage': !(0, runtime_1.exists)(json, 'rolloutPercentage') ? undefined : json['rolloutPercentage'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateFeatureTargetDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'featureId': value.featureId,
        'targetEntityId': value.targetEntityId,
        'targetType': value.targetType,
        'rolloutPercentage': value.rolloutPercentage,
    };
}
