// @cp/node-sdk/dist/cjs/models/CreateCheckoutSessionDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCheckoutSessionDtoCurrencyEnum = exports.CreateCheckoutSessionDtoSecCodeEnum = void 0;
exports.instanceOfCreateCheckoutSessionDto = instanceOfCreateCheckoutSessionDto;
exports.CreateCheckoutSessionDtoFromJSON = CreateCheckoutSessionDtoFromJSON;
exports.CreateCheckoutSessionDtoFromJSONTyped = CreateCheckoutSessionDtoFromJSONTyped;
exports.CreateCheckoutSessionDtoToJSON = CreateCheckoutSessionDtoToJSON;
const runtime_1 = require("../runtime");
const CreateCheckoutSessionItemDto_1 = require("./CreateCheckoutSessionItemDto");
const TagDto_1 = require("./TagDto");
/**
 * @export
 */
exports.CreateCheckoutSessionDtoSecCodeEnum = {
    Ppd: 'PPD',
    Ccd: 'CCD',
    Web: 'WEB',
    Tel: 'TEL'
};
/**
 * @export
 */
exports.CreateCheckoutSessionDtoCurrencyEnum = {
    Usd: 'USD',
    Cad: 'CAD',
    Aud: 'AUD'
};
/**
 * Check if a given object implements the CreateCheckoutSessionDto interface.
 */
function instanceOfCreateCheckoutSessionDto(value) {
    let isInstance = true;
    isInstance = isInstance && "items" in value;
    return isInstance;
}
function CreateCheckoutSessionDtoFromJSON(json) {
    return CreateCheckoutSessionDtoFromJSONTyped(json, false);
}
function CreateCheckoutSessionDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'items': (json['items'].map(CreateCheckoutSessionItemDto_1.CreateCheckoutSessionItemDtoFromJSON)),
        'excludeCFee': !(0, runtime_1.exists)(json, 'excludeCFee') ? undefined : json['excludeCFee'],
        'successRedirectUrl': !(0, runtime_1.exists)(json, 'successRedirectUrl') ? undefined : json['successRedirectUrl'],
        'cancelRedirectUrl': !(0, runtime_1.exists)(json, 'cancelRedirectUrl') ? undefined : json['cancelRedirectUrl'],
        'buyerId': !(0, runtime_1.exists)(json, 'buyerId') ? undefined : json['buyerId'],
        'secCode': !(0, runtime_1.exists)(json, 'secCode') ? undefined : json['secCode'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'currency': !(0, runtime_1.exists)(json, 'currency') ? undefined : json['currency'],
        'country': !(0, runtime_1.exists)(json, 'country') ? undefined : json['country'],
        'maxAttempts': !(0, runtime_1.exists)(json, 'maxAttempts') ? undefined : json['maxAttempts'],
        'duration': !(0, runtime_1.exists)(json, 'duration') ? undefined : json['duration'],
        'externalTransactionId': !(0, runtime_1.exists)(json, 'externalTransactionId') ? undefined : json['externalTransactionId'],
        'tags': !(0, runtime_1.exists)(json, 'tags') ? undefined : (json['tags'].map(TagDto_1.TagDtoFromJSON)),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateCheckoutSessionDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'items': (value.items.map(CreateCheckoutSessionItemDto_1.CreateCheckoutSessionItemDtoToJSON)),
        'excludeCFee': value.excludeCFee,
        'successRedirectUrl': value.successRedirectUrl,
        'cancelRedirectUrl': value.cancelRedirectUrl,
        'buyerId': value.buyerId,
        'secCode': value.secCode,
        'externalBuyerId': value.externalBuyerId,
        'currency': value.currency,
        'country': value.country,
        'maxAttempts': value.maxAttempts,
        'duration': value.duration,
        'externalTransactionId': value.externalTransactionId,
        'tags': value.tags === undefined ? undefined : (value.tags.map(TagDto_1.TagDtoToJSON)),
    };
}
