// @cp/node-sdk/dist/cjs/models/CreateMerchantConnectorCredentialsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateMerchantConnectorCredentialsDto = instanceOfCreateMerchantConnectorCredentialsDto;
exports.CreateMerchantConnectorCredentialsDtoFromJSON = CreateMerchantConnectorCredentialsDtoFromJSON;
exports.CreateMerchantConnectorCredentialsDtoFromJSONTyped = CreateMerchantConnectorCredentialsDtoFromJSONTyped;
exports.CreateMerchantConnectorCredentialsDtoToJSON = CreateMerchantConnectorCredentialsDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CreateMerchantConnectorCredentialsDto interface.
 */
function instanceOfCreateMerchantConnectorCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "connectorId" in value;
    return isInstance;
}
function CreateMerchantConnectorCredentialsDtoFromJSON(json) {
    return CreateMerchantConnectorCredentialsDtoFromJSONTyped(json, false);
}
function CreateMerchantConnectorCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'connectorId': json['connectorId'],
        'username': !(0, runtime_1.exists)(json, 'username') ? undefined : json['username'],
        'password': !(0, runtime_1.exists)(json, 'password') ? undefined : json['password'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateMerchantConnectorCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'connectorId': value.connectorId,
        'username': value.username,
        'password': value.password,
    };
}
