// @cp/node-sdk/dist/cjs/models/CreateTransactionsReportsScheduleDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTransactionsReportsScheduleDtoOrderEnum = exports.CreateTransactionsReportsScheduleDtoDateRangeEnum = exports.CreateTransactionsReportsScheduleDtoIntervalEnum = void 0;
exports.instanceOfCreateTransactionsReportsScheduleDto = instanceOfCreateTransactionsReportsScheduleDto;
exports.CreateTransactionsReportsScheduleDtoFromJSON = CreateTransactionsReportsScheduleDtoFromJSON;
exports.CreateTransactionsReportsScheduleDtoFromJSONTyped = CreateTransactionsReportsScheduleDtoFromJSONTyped;
exports.CreateTransactionsReportsScheduleDtoToJSON = CreateTransactionsReportsScheduleDtoToJSON;
const runtime_1 = require("../runtime");
const CreateTransactionsReportsFiltersDto_1 = require("./CreateTransactionsReportsFiltersDto");
/**
 * @export
 */
exports.CreateTransactionsReportsScheduleDtoIntervalEnum = {
    Daily: 'DAILY',
    Weekly: 'WEEKLY',
    Monthly: 'MONTHLY'
};
/**
 * @export
 */
exports.CreateTransactionsReportsScheduleDtoDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * @export
 */
exports.CreateTransactionsReportsScheduleDtoOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * Check if a given object implements the CreateTransactionsReportsScheduleDto interface.
 */
function instanceOfCreateTransactionsReportsScheduleDto(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "interval" in value;
    isInstance = isInstance && "dateRange" in value;
    isInstance = isInstance && "recipients" in value;
    isInstance = isInstance && "startTime" in value;
    isInstance = isInstance && "endTime" in value;
    isInstance = isInstance && "nextRun" in value;
    isInstance = isInstance && "columns" in value;
    isInstance = isInstance && "transactionsFilters" in value;
    return isInstance;
}
function CreateTransactionsReportsScheduleDtoFromJSON(json) {
    return CreateTransactionsReportsScheduleDtoFromJSONTyped(json, false);
}
function CreateTransactionsReportsScheduleDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': json['name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'interval': json['interval'],
        'dateRange': json['dateRange'],
        'recipients': json['recipients'],
        'order': !(0, runtime_1.exists)(json, 'order') ? undefined : json['order'],
        'startTime': json['startTime'],
        'endTime': json['endTime'],
        'nextRun': json['nextRun'],
        'columns': json['columns'],
        'transactionsFilters': (0, CreateTransactionsReportsFiltersDto_1.CreateTransactionsReportsFiltersDtoFromJSON)(json['transactionsFilters']),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateTransactionsReportsScheduleDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'interval': value.interval,
        'dateRange': value.dateRange,
        'recipients': value.recipients,
        'order': value.order,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'nextRun': value.nextRun,
        'columns': value.columns,
        'transactionsFilters': (0, CreateTransactionsReportsFiltersDto_1.CreateTransactionsReportsFiltersDtoToJSON)(value.transactionsFilters),
    };
}
