// @cp/node-sdk/dist/cjs/models/CreateCheckoutSessionPaymentMethodDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateCheckoutSessionPaymentMethodDtoAccountTypeEnum = exports.CreateCheckoutSessionPaymentMethodDtoMethodEnum = void 0;
exports.instanceOfCreateCheckoutSessionPaymentMethodDto = instanceOfCreateCheckoutSessionPaymentMethodDto;
exports.CreateCheckoutSessionPaymentMethodDtoFromJSON = CreateCheckoutSessionPaymentMethodDtoFromJSON;
exports.CreateCheckoutSessionPaymentMethodDtoFromJSONTyped = CreateCheckoutSessionPaymentMethodDtoFromJSONTyped;
exports.CreateCheckoutSessionPaymentMethodDtoToJSON = CreateCheckoutSessionPaymentMethodDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CreateCheckoutSessionPaymentMethodDtoMethodEnum = {
    Ach: 'ACH',
    Bank: 'BANK',
    Card: 'CARD'
};
/**
 * @export
 */
exports.CreateCheckoutSessionPaymentMethodDtoAccountTypeEnum = {
    Checking: 'CHECKING',
    Savings: 'SAVINGS'
};
/**
 * Check if a given object implements the CreateCheckoutSessionPaymentMethodDto interface.
 */
function instanceOfCreateCheckoutSessionPaymentMethodDto(value) {
    let isInstance = true;
    isInstance = isInstance && "method" in value;
    return isInstance;
}
function CreateCheckoutSessionPaymentMethodDtoFromJSON(json) {
    return CreateCheckoutSessionPaymentMethodDtoFromJSONTyped(json, false);
}
function CreateCheckoutSessionPaymentMethodDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'method': json['method'],
        'number': !(0, runtime_1.exists)(json, 'number') ? undefined : json['number'],
        'securityCode': !(0, runtime_1.exists)(json, 'securityCode') ? undefined : json['securityCode'],
        'expirationDate': !(0, runtime_1.exists)(json, 'expirationDate') ? undefined : json['expirationDate'],
        'paymentMethodId': !(0, runtime_1.exists)(json, 'paymentMethodId') ? undefined : json['paymentMethodId'],
        'store': !(0, runtime_1.exists)(json, 'store') ? undefined : json['store'],
        'abaNumber': !(0, runtime_1.exists)(json, 'abaNumber') ? undefined : json['abaNumber'],
        'accountNumber': !(0, runtime_1.exists)(json, 'accountNumber') ? undefined : json['accountNumber'],
        'accountType': !(0, runtime_1.exists)(json, 'accountType') ? undefined : json['accountType'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateCheckoutSessionPaymentMethodDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'method': value.method,
        'number': value.number,
        'securityCode': value.securityCode,
        'expirationDate': value.expirationDate,
        'paymentMethodId': value.paymentMethodId,
        'store': value.store,
        'abaNumber': value.abaNumber,
        'accountNumber': value.accountNumber,
        'accountType': value.accountType,
    };
}
