// @cp/node-sdk/dist/cjs/models/CreatePaymentMethodSessionDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreatePaymentMethodSessionDto = instanceOfCreatePaymentMethodSessionDto;
exports.CreatePaymentMethodSessionDtoFromJSON = CreatePaymentMethodSessionDtoFromJSON;
exports.CreatePaymentMethodSessionDtoFromJSONTyped = CreatePaymentMethodSessionDtoFromJSONTyped;
exports.CreatePaymentMethodSessionDtoToJSON = CreatePaymentMethodSessionDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CreatePaymentMethodSessionDto interface.
 */
function instanceOfCreatePaymentMethodSessionDto(value) {
    let isInstance = true;
    isInstance = isInstance && "origin" in value;
    return isInstance;
}
function CreatePaymentMethodSessionDtoFromJSON(json) {
    return CreatePaymentMethodSessionDtoFromJSONTyped(json, false);
}
function CreatePaymentMethodSessionDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'origin': json['origin'],
        'buyerId': !(0, runtime_1.exists)(json, 'buyerId') ? undefined : json['buyerId'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreatePaymentMethodSessionDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'origin': value.origin,
        'buyerId': value.buyerId,
        'externalBuyerId': value.externalBuyerId,
    };
}
