// @cp/node-sdk/dist/cjs/models/CreateMerchantAddressDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateMerchantAddressDto = instanceOfCreateMerchantAddressDto;
exports.CreateMerchantAddressDtoFromJSON = CreateMerchantAddressDtoFromJSON;
exports.CreateMerchantAddressDtoFromJSONTyped = CreateMerchantAddressDtoFromJSONTyped;
exports.CreateMerchantAddressDtoToJSON = CreateMerchantAddressDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CreateMerchantAddressDto interface.
 */
function instanceOfCreateMerchantAddressDto(value) {
    let isInstance = true;
    isInstance = isInstance && "city" in value;
    isInstance = isInstance && "country" in value;
    isInstance = isInstance && "subdivision" in value;
    isInstance = isInstance && "streetAddress" in value;
    isInstance = isInstance && "postalCode" in value;
    return isInstance;
}
function CreateMerchantAddressDtoFromJSON(json) {
    return CreateMerchantAddressDtoFromJSONTyped(json, false);
}
function CreateMerchantAddressDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'city': json['city'],
        'country': json['country'],
        'subdivision': json['subdivision'],
        'streetAddress': json['streetAddress'],
        'postalCode': json['postalCode'],
        'countryCode': !(0, runtime_1.exists)(json, 'countryCode') ? undefined : json['countryCode'],
        'streetAddress2': !(0, runtime_1.exists)(json, 'streetAddress2') ? undefined : json['streetAddress2'],
        'subdivisionCode': !(0, runtime_1.exists)(json, 'subdivisionCode') ? undefined : json['subdivisionCode'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateMerchantAddressDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'city': value.city,
        'country': value.country,
        'subdivision': value.subdivision,
        'streetAddress': value.streetAddress,
        'postalCode': value.postalCode,
        'countryCode': value.countryCode,
        'streetAddress2': value.streetAddress2,
        'subdivisionCode': value.subdivisionCode,
    };
}
