// @cp/node-sdk/dist/cjs/models/CreateCheckoutSessionItemDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateCheckoutSessionItemDto = instanceOfCreateCheckoutSessionItemDto;
exports.CreateCheckoutSessionItemDtoFromJSON = CreateCheckoutSessionItemDtoFromJSON;
exports.CreateCheckoutSessionItemDtoFromJSONTyped = CreateCheckoutSessionItemDtoFromJSONTyped;
exports.CreateCheckoutSessionItemDtoToJSON = CreateCheckoutSessionItemDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CreateCheckoutSessionItemDto interface.
 */
function instanceOfCreateCheckoutSessionItemDto(value) {
    let isInstance = true;
    isInstance = isInstance && "unitPrice" in value;
    isInstance = isInstance && "description" in value;
    return isInstance;
}
function CreateCheckoutSessionItemDtoFromJSON(json) {
    return CreateCheckoutSessionItemDtoFromJSONTyped(json, false);
}
function CreateCheckoutSessionItemDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'unitPrice': json['unitPrice'],
        'description': json['description'],
        'quantity': !(0, runtime_1.exists)(json, 'quantity') ? undefined : json['quantity'],
        'taxTotal': !(0, runtime_1.exists)(json, 'taxTotal') ? undefined : json['taxTotal'],
        'feesTotal': !(0, runtime_1.exists)(json, 'feesTotal') ? undefined : json['feesTotal'],
        'discountTotal': !(0, runtime_1.exists)(json, 'discountTotal') ? undefined : json['discountTotal'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateCheckoutSessionItemDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'unitPrice': value.unitPrice,
        'description': value.description,
        'quantity': value.quantity,
        'taxTotal': value.taxTotal,
        'feesTotal': value.feesTotal,
        'discountTotal': value.discountTotal,
    };
}
