// @cp/node-sdk/dist/cjs/models/CreateEventConfigDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateEventConfigDto = instanceOfCreateEventConfigDto;
exports.CreateEventConfigDtoFromJSON = CreateEventConfigDtoFromJSON;
exports.CreateEventConfigDtoFromJSONTyped = CreateEventConfigDtoFromJSONTyped;
exports.CreateEventConfigDtoToJSON = CreateEventConfigDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the CreateEventConfigDto interface.
 */
function instanceOfCreateEventConfigDto(value) {
    let isInstance = true;
    isInstance = isInstance && "selected" in value;
    return isInstance;
}
function CreateEventConfigDtoFromJSON(json) {
    return CreateEventConfigDtoFromJSONTyped(json, false);
}
function CreateEventConfigDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'selected': json['selected'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateEventConfigDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'selected': value.selected,
    };
}
