// @cp/node-sdk/dist/cjs/models/CreateTransactionsReportsFiltersDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateTransactionsReportsFiltersDtoTransactionStatusesEnum = exports.CreateTransactionsReportsFiltersDtoTransactionIntentsEnum = exports.CreateTransactionsReportsFiltersDtoCardSchemesEnum = exports.CreateTransactionsReportsFiltersDtoCardTypesEnum = exports.CreateTransactionsReportsFiltersDtoAccountTypesEnum = exports.CreateTransactionsReportsFiltersDtoPaymentMethodsEnum = void 0;
exports.instanceOfCreateTransactionsReportsFiltersDto = instanceOfCreateTransactionsReportsFiltersDto;
exports.CreateTransactionsReportsFiltersDtoFromJSON = CreateTransactionsReportsFiltersDtoFromJSON;
exports.CreateTransactionsReportsFiltersDtoFromJSONTyped = CreateTransactionsReportsFiltersDtoFromJSONTyped;
exports.CreateTransactionsReportsFiltersDtoToJSON = CreateTransactionsReportsFiltersDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CreateTransactionsReportsFiltersDtoPaymentMethodsEnum = {
    Card: 'CARD',
    Bank: 'BANK'
};
/**
 * @export
 */
exports.CreateTransactionsReportsFiltersDtoAccountTypesEnum = {
    Checking: 'CHECKING',
    Savings: 'SAVINGS'
};
/**
 * @export
 */
exports.CreateTransactionsReportsFiltersDtoCardTypesEnum = {
    Credit: 'CREDIT',
    Debit: 'DEBIT',
    Prepaid: 'PREPAID'
};
/**
 * @export
 */
exports.CreateTransactionsReportsFiltersDtoCardSchemesEnum = {
    Visa: 'visa',
    Mastercard: 'mastercard',
    Amex: 'amex',
    Discover: 'discover',
    Jcb: 'jcb',
    DinersClub: 'diners_club',
    Maestro: 'maestro'
};
/**
 * @export
 */
exports.CreateTransactionsReportsFiltersDtoTransactionIntentsEnum = {
    Capture: 'capture',
    Authorize: 'authorize',
    Debit: 'debit',
    Credit: 'credit'
};
/**
 * @export
 */
exports.CreateTransactionsReportsFiltersDtoTransactionStatusesEnum = {
    Authorized: 'authorized',
    AuthorizationFailed: 'authorizationFailed',
    Declined: 'declined',
    Captured: 'captured',
    Voided: 'voided',
    Hold: 'hold',
    Submitted: 'submitted',
    Transmitted: 'transmitted',
    Settled: 'settled',
    Returned: 'returned',
    Rejected: 'rejected'
};
/**
 * Check if a given object implements the CreateTransactionsReportsFiltersDto interface.
 */
function instanceOfCreateTransactionsReportsFiltersDto(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethods" in value;
    return isInstance;
}
function CreateTransactionsReportsFiltersDtoFromJSON(json) {
    return CreateTransactionsReportsFiltersDtoFromJSONTyped(json, false);
}
function CreateTransactionsReportsFiltersDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethods': json['paymentMethods'],
        'accountTypes': !(0, runtime_1.exists)(json, 'accountTypes') ? undefined : json['accountTypes'],
        'cardTypes': !(0, runtime_1.exists)(json, 'cardTypes') ? undefined : json['cardTypes'],
        'cardSchemes': !(0, runtime_1.exists)(json, 'cardSchemes') ? undefined : json['cardSchemes'],
        'transactionIntents': !(0, runtime_1.exists)(json, 'transactionIntents') ? undefined : json['transactionIntents'],
        'transactionStatuses': !(0, runtime_1.exists)(json, 'transactionStatuses') ? undefined : json['transactionStatuses'],
        'amount': !(0, runtime_1.exists)(json, 'amount') ? undefined : json['amount'],
        'amountGreaterThan': !(0, runtime_1.exists)(json, 'amountGreaterThan') ? undefined : json['amountGreaterThan'],
        'amountLessThan': !(0, runtime_1.exists)(json, 'amountLessThan') ? undefined : json['amountLessThan'],
        'amountMin': !(0, runtime_1.exists)(json, 'amountMin') ? undefined : json['amountMin'],
        'amountMax': !(0, runtime_1.exists)(json, 'amountMax') ? undefined : json['amountMax'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateTransactionsReportsFiltersDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethods': value.paymentMethods,
        'accountTypes': value.accountTypes,
        'cardTypes': value.cardTypes,
        'cardSchemes': value.cardSchemes,
        'transactionIntents': value.transactionIntents,
        'transactionStatuses': value.transactionStatuses,
        'amount': value.amount,
        'amountGreaterThan': value.amountGreaterThan,
        'amountLessThan': value.amountLessThan,
        'amountMin': value.amountMin,
        'amountMax': value.amountMax,
    };
}
