// @cp/node-sdk/dist/cjs/models/CreateProvisionedCardTokenByMerchantDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateProvisionedCardTokenByMerchantDtoConnectorEnum = void 0;
exports.instanceOfCreateProvisionedCardTokenByMerchantDto = instanceOfCreateProvisionedCardTokenByMerchantDto;
exports.CreateProvisionedCardTokenByMerchantDtoFromJSON = CreateProvisionedCardTokenByMerchantDtoFromJSON;
exports.CreateProvisionedCardTokenByMerchantDtoFromJSONTyped = CreateProvisionedCardTokenByMerchantDtoFromJSONTyped;
exports.CreateProvisionedCardTokenByMerchantDtoToJSON = CreateProvisionedCardTokenByMerchantDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CreateProvisionedCardTokenByMerchantDtoConnectorEnum = {
    Epay: 'EPAY',
    Tsys: 'TSYS'
};
/**
 * Check if a given object implements the CreateProvisionedCardTokenByMerchantDto interface.
 */
function instanceOfCreateProvisionedCardTokenByMerchantDto(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethodId" in value;
    isInstance = isInstance && "connector" in value;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
function CreateProvisionedCardTokenByMerchantDtoFromJSON(json) {
    return CreateProvisionedCardTokenByMerchantDtoFromJSONTyped(json, false);
}
function CreateProvisionedCardTokenByMerchantDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethodId': json['paymentMethodId'],
        'securityCode': !(0, runtime_1.exists)(json, 'securityCode') ? undefined : json['securityCode'],
        'connector': json['connector'],
        'redirectUrl': !(0, runtime_1.exists)(json, 'redirectUrl') ? undefined : json['redirectUrl'],
        'merchantId': json['merchantId'],
        'token': !(0, runtime_1.exists)(json, 'token') ? undefined : json['token'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateProvisionedCardTokenByMerchantDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethodId': value.paymentMethodId,
        'securityCode': value.securityCode,
        'connector': value.connector,
        'redirectUrl': value.redirectUrl,
        'merchantId': value.merchantId,
        'token': value.token,
    };
}
