// @cp/node-sdk/dist/cjs/models/CreateMerchantConnectorWithoutMerchantAccountIdDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateMerchantConnectorWithoutMerchantAccountIdDtoTypeEnum = exports.CreateMerchantConnectorWithoutMerchantAccountIdDtoConnectorEnum = void 0;
exports.instanceOfCreateMerchantConnectorWithoutMerchantAccountIdDto = instanceOfCreateMerchantConnectorWithoutMerchantAccountIdDto;
exports.CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSON = CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSON;
exports.CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSONTyped = CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSONTyped;
exports.CreateMerchantConnectorWithoutMerchantAccountIdDtoToJSON = CreateMerchantConnectorWithoutMerchantAccountIdDtoToJSON;
const runtime_1 = require("../runtime");
const CreateMerchantConnectorCredentialsDto_1 = require("./CreateMerchantConnectorCredentialsDto");
/**
 * @export
 */
exports.CreateMerchantConnectorWithoutMerchantAccountIdDtoConnectorEnum = {
    Epay: 'EPAY',
    AuthorizeNet: 'AUTHORIZE_NET',
    ChannelPayments: 'CHANNEL_PAYMENTS',
    Tsys: 'TSYS'
};
/**
 * @export
 */
exports.CreateMerchantConnectorWithoutMerchantAccountIdDtoTypeEnum = {
    Gateway: 'GATEWAY',
    Terminal: 'TERMINAL'
};
/**
 * Check if a given object implements the CreateMerchantConnectorWithoutMerchantAccountIdDto interface.
 */
function instanceOfCreateMerchantConnectorWithoutMerchantAccountIdDto(value) {
    let isInstance = true;
    isInstance = isInstance && "connector" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSON(json) {
    return CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSONTyped(json, false);
}
function CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'connector': json['connector'],
        'type': json['type'],
        'connectorCredentials': !(0, runtime_1.exists)(json, 'connectorCredentials') ? undefined : (0, CreateMerchantConnectorCredentialsDto_1.CreateMerchantConnectorCredentialsDtoFromJSON)(json['connectorCredentials']),
        'issuedConnectorId': !(0, runtime_1.exists)(json, 'issuedConnectorId') ? undefined : json['issuedConnectorId'],
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateMerchantConnectorWithoutMerchantAccountIdDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'connector': value.connector,
        'type': value.type,
        'connectorCredentials': (0, CreateMerchantConnectorCredentialsDto_1.CreateMerchantConnectorCredentialsDtoToJSON)(value.connectorCredentials),
        'issuedConnectorId': value.issuedConnectorId,
        'name': value.name,
    };
}
