// @cp/node-sdk/dist/cjs/models/CreatePaymentMethodDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreatePaymentMethodDtoAccountTypeEnum = exports.CreatePaymentMethodDtoMethodEnum = void 0;
exports.instanceOfCreatePaymentMethodDto = instanceOfCreatePaymentMethodDto;
exports.CreatePaymentMethodDtoFromJSON = CreatePaymentMethodDtoFromJSON;
exports.CreatePaymentMethodDtoFromJSONTyped = CreatePaymentMethodDtoFromJSONTyped;
exports.CreatePaymentMethodDtoToJSON = CreatePaymentMethodDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CreatePaymentMethodDtoMethodEnum = {
    Card: 'CARD',
    Bank: 'BANK'
};
/**
 * @export
 */
exports.CreatePaymentMethodDtoAccountTypeEnum = {
    Checking: 'CHECKING',
    Savings: 'SAVINGS'
};
/**
 * Check if a given object implements the CreatePaymentMethodDto interface.
 */
function instanceOfCreatePaymentMethodDto(value) {
    let isInstance = true;
    isInstance = isInstance && "method" in value;
    return isInstance;
}
function CreatePaymentMethodDtoFromJSON(json) {
    return CreatePaymentMethodDtoFromJSONTyped(json, false);
}
function CreatePaymentMethodDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'method': json['method'],
        'number': !(0, runtime_1.exists)(json, 'number') ? undefined : json['number'],
        'expirationDate': !(0, runtime_1.exists)(json, 'expirationDate') ? undefined : json['expirationDate'],
        'abaNumber': !(0, runtime_1.exists)(json, 'abaNumber') ? undefined : json['abaNumber'],
        'accountNumber': !(0, runtime_1.exists)(json, 'accountNumber') ? undefined : json['accountNumber'],
        'accountType': !(0, runtime_1.exists)(json, 'accountType') ? undefined : json['accountType'],
        'accountHolderName': !(0, runtime_1.exists)(json, 'accountHolderName') ? undefined : json['accountHolderName'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'buyerId': !(0, runtime_1.exists)(json, 'buyerId') ? undefined : json['buyerId'],
        'externalPaymentMethodId': !(0, runtime_1.exists)(json, 'externalPaymentMethodId') ? undefined : json['externalPaymentMethodId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreatePaymentMethodDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'method': value.method,
        'number': value.number,
        'expirationDate': value.expirationDate,
        'abaNumber': value.abaNumber,
        'accountNumber': value.accountNumber,
        'accountType': value.accountType,
        'accountHolderName': value.accountHolderName,
        'externalBuyerId': value.externalBuyerId,
        'buyerId': value.buyerId,
        'externalPaymentMethodId': value.externalPaymentMethodId,
    };
}
