// @cp/node-sdk/dist/cjs/models/CreateBuyerDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfCreateBuyerDto = instanceOfCreateBuyerDto;
exports.CreateBuyerDtoFromJSON = CreateBuyerDtoFromJSON;
exports.CreateBuyerDtoFromJSONTyped = CreateBuyerDtoFromJSONTyped;
exports.CreateBuyerDtoToJSON = CreateBuyerDtoToJSON;
const runtime_1 = require("../runtime");
const CreateBuyerAddressDto_1 = require("./CreateBuyerAddressDto");
/**
 * Check if a given object implements the CreateBuyerDto interface.
 */
function instanceOfCreateBuyerDto(value) {
    let isInstance = true;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "address" in value;
    return isInstance;
}
function CreateBuyerDtoFromJSON(json) {
    return CreateBuyerDtoFromJSONTyped(json, false);
}
function CreateBuyerDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'email': json['email'],
        'address': (0, CreateBuyerAddressDto_1.CreateBuyerAddressDtoFromJSON)(json['address']),
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateBuyerDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'address': (0, CreateBuyerAddressDto_1.CreateBuyerAddressDtoToJSON)(value.address),
        'phoneNumber': value.phoneNumber,
        'externalBuyerId': value.externalBuyerId,
    };
}
