// @cp/node-sdk/dist/cjs/models/CreateUserDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.CreateUserDtoRoleEnum = void 0;
exports.instanceOfCreateUserDto = instanceOfCreateUserDto;
exports.CreateUserDtoFromJSON = CreateUserDtoFromJSON;
exports.CreateUserDtoFromJSONTyped = CreateUserDtoFromJSONTyped;
exports.CreateUserDtoToJSON = CreateUserDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.CreateUserDtoRoleEnum = {
    User: 'user',
    Agent: 'agent',
    Admin: 'admin',
    SuperAdmin: 'superAdmin'
};
/**
 * Check if a given object implements the CreateUserDto interface.
 */
function instanceOfCreateUserDto(value) {
    let isInstance = true;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "role" in value;
    return isInstance;
}
function CreateUserDtoFromJSON(json) {
    return CreateUserDtoFromJSONTyped(json, false);
}
function CreateUserDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'email': json['email'],
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'role': json['role'],
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function CreateUserDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'email': value.email,
        'givenName': value.givenName,
        'familyName': value.familyName,
        'role': value.role,
        'phoneNumber': value.phoneNumber,
    };
}
