// @cp/node-sdk/dist/cjs/models/ErrorEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfErrorEntity = instanceOfErrorEntity;
exports.ErrorEntityFromJSON = ErrorEntityFromJSON;
exports.ErrorEntityFromJSONTyped = ErrorEntityFromJSONTyped;
exports.ErrorEntityToJSON = ErrorEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ErrorEntity interface.
 */
function instanceOfErrorEntity(value) {
    let isInstance = true;
    return isInstance;
}
function ErrorEntityFromJSON(json) {
    return ErrorEntityFromJSONTyped(json, false);
}
function ErrorEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'statusCode': !(0, runtime_1.exists)(json, 'statusCode') ? undefined : json['statusCode'],
        'message': !(0, runtime_1.exists)(json, 'message') ? undefined : json['message'],
        'path': !(0, runtime_1.exists)(json, 'path') ? undefined : json['path'],
        'timestamp': !(0, runtime_1.exists)(json, 'timestamp') ? undefined : (new Date(json['timestamp'])),
        'method': !(0, runtime_1.exists)(json, 'method') ? undefined : json['method'],
        'requestId': !(0, runtime_1.exists)(json, 'requestId') ? undefined : json['requestId'],
        'details': !(0, runtime_1.exists)(json, 'details') ? undefined : json['details'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function ErrorEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'statusCode': value.statusCode,
        'message': value.message,
        'path': value.path,
        'timestamp': value.timestamp === undefined ? undefined : (value.timestamp.toISOString()),
        'method': value.method,
        'requestId': value.requestId,
        'details': value.details,
    };
}
