// @cp/node-sdk/dist/cjs/models/FeaturesEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfFeaturesEntity = instanceOfFeaturesEntity;
exports.FeaturesEntityFromJSON = FeaturesEntityFromJSON;
exports.FeaturesEntityFromJSONTyped = FeaturesEntityFromJSONTyped;
exports.FeaturesEntityToJSON = FeaturesEntityToJSON;
const runtime_1 = require("../runtime");
const FeatureEntity_1 = require("./FeatureEntity");
/**
 * Check if a given object implements the FeaturesEntity interface.
 */
function instanceOfFeaturesEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "features" in value;
    return isInstance;
}
function FeaturesEntityFromJSON(json) {
    return FeaturesEntityFromJSONTyped(json, false);
}
function FeaturesEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'features': (json['features'].map(FeatureEntity_1.FeatureEntityFromJSON)),
        'featuresDict': !(0, runtime_1.exists)(json, 'featuresDict') ? undefined : json['featuresDict'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function FeaturesEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'features': (value.features.map(FeatureEntity_1.FeatureEntityToJSON)),
        'featuresDict': value.featuresDict,
    };
}
