// @cp/node-sdk/dist/cjs/models/FeatureTargetEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeatureTargetEntityTargetTypeEnum = void 0;
exports.instanceOfFeatureTargetEntity = instanceOfFeatureTargetEntity;
exports.FeatureTargetEntityFromJSON = FeatureTargetEntityFromJSON;
exports.FeatureTargetEntityFromJSONTyped = FeatureTargetEntityFromJSONTyped;
exports.FeatureTargetEntityToJSON = FeatureTargetEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.FeatureTargetEntityTargetTypeEnum = {
    Merchant: 'merchant',
    User: 'user',
    PaymentMethod: 'payment_method',
    Buyer: 'buyer',
    Webhook: 'webhook',
    Job: 'job'
};
/**
 * Check if a given object implements the FeatureTargetEntity interface.
 */
function instanceOfFeatureTargetEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "featureId" in value;
    isInstance = isInstance && "targetId" in value;
    isInstance = isInstance && "targetEntityId" in value;
    isInstance = isInstance && "targetType" in value;
    isInstance = isInstance && "rolloutPercentage" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    return isInstance;
}
function FeatureTargetEntityFromJSON(json) {
    return FeatureTargetEntityFromJSONTyped(json, false);
}
function FeatureTargetEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'featureId': json['featureId'],
        'targetId': json['targetId'],
        'targetEntityId': json['targetEntityId'],
        'targetType': json['targetType'],
        'rolloutPercentage': json['rolloutPercentage'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function FeatureTargetEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'featureId': value.featureId,
        'targetId': value.targetId,
        'targetEntityId': value.targetEntityId,
        'targetType': value.targetType,
        'rolloutPercentage': value.rolloutPercentage,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
    };
}
