// @cp/node-sdk/dist/cjs/models/FeeDetailsEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.FeeDetailsEntityChargeTypeEnum = exports.FeeDetailsEntityTypeEnum = void 0;
exports.instanceOfFeeDetailsEntity = instanceOfFeeDetailsEntity;
exports.FeeDetailsEntityFromJSON = FeeDetailsEntityFromJSON;
exports.FeeDetailsEntityFromJSONTyped = FeeDetailsEntityFromJSONTyped;
exports.FeeDetailsEntityToJSON = FeeDetailsEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.FeeDetailsEntityTypeEnum = {
    ConvenienceFee: 'CONVENIENCE_FEE'
};
/**
 * @export
 */
exports.FeeDetailsEntityChargeTypeEnum = {
    IncludedInAmount: 'INCLUDED_IN_AMOUNT',
    ChargedSeparately: 'CHARGED_SEPARATELY'
};
/**
 * Check if a given object implements the FeeDetailsEntity interface.
 */
function instanceOfFeeDetailsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "chargeType" in value;
    isInstance = isInstance && "amount" in value;
    return isInstance;
}
function FeeDetailsEntityFromJSON(json) {
    return FeeDetailsEntityFromJSONTyped(json, false);
}
function FeeDetailsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'type': json['type'],
        'chargeType': json['chargeType'],
        'amount': json['amount'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function FeeDetailsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'chargeType': value.chargeType,
        'amount': value.amount,
    };
}
