// @cp/node-sdk/dist/cjs/models/LogoUploadEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfLogoUploadEntity = instanceOfLogoUploadEntity;
exports.LogoUploadEntityFromJSON = LogoUploadEntityFromJSON;
exports.LogoUploadEntityFromJSONTyped = LogoUploadEntityFromJSONTyped;
exports.LogoUploadEntityToJSON = LogoUploadEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the LogoUploadEntity interface.
 */
function instanceOfLogoUploadEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "imageUrl" in value;
    isInstance = isInstance && "uploadUrl" in value;
    return isInstance;
}
function LogoUploadEntityFromJSON(json) {
    return LogoUploadEntityFromJSONTyped(json, false);
}
function LogoUploadEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'imageUrl': json['imageUrl'],
        'uploadUrl': json['uploadUrl'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function LogoUploadEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'imageUrl': value.imageUrl,
        'uploadUrl': value.uploadUrl,
    };
}
