// @cp/node-sdk/dist/cjs/models/MerchantSettingsBaseDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantSettingsBaseDto = instanceOfMerchantSettingsBaseDto;
exports.MerchantSettingsBaseDtoFromJSON = MerchantSettingsBaseDtoFromJSON;
exports.MerchantSettingsBaseDtoFromJSONTyped = MerchantSettingsBaseDtoFromJSONTyped;
exports.MerchantSettingsBaseDtoToJSON = MerchantSettingsBaseDtoToJSON;
const runtime_1 = require("../runtime");
const MerchantThemeDto_1 = require("./MerchantThemeDto");
/**
 * Check if a given object implements the MerchantSettingsBaseDto interface.
 */
function instanceOfMerchantSettingsBaseDto(value) {
    let isInstance = true;
    return isInstance;
}
function MerchantSettingsBaseDtoFromJSON(json) {
    return MerchantSettingsBaseDtoFromJSONTyped(json, false);
}
function MerchantSettingsBaseDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'checkoutTheme': !(0, runtime_1.exists)(json, 'checkoutTheme') ? undefined : (0, MerchantThemeDto_1.MerchantThemeDtoFromJSON)(json['checkoutTheme']),
        'isSplitCFeeEnabled': !(0, runtime_1.exists)(json, 'isSplitCFeeEnabled') ? undefined : json['isSplitCFeeEnabled'],
        'isDebtRepaymentEnabled': !(0, runtime_1.exists)(json, 'isDebtRepaymentEnabled') ? undefined : json['isDebtRepaymentEnabled'],
        'isPartialAuthEnabled': !(0, runtime_1.exists)(json, 'isPartialAuthEnabled') ? undefined : json['isPartialAuthEnabled'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantSettingsBaseDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'checkoutTheme': (0, MerchantThemeDto_1.MerchantThemeDtoToJSON)(value.checkoutTheme),
        'isSplitCFeeEnabled': value.isSplitCFeeEnabled,
        'isDebtRepaymentEnabled': value.isDebtRepaymentEnabled,
        'isPartialAuthEnabled': value.isPartialAuthEnabled,
    };
}
