// @cp/node-sdk/dist/cjs/models/MerchantThemeLogoDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MerchantThemeLogoDtoTypeEnum = void 0;
exports.instanceOfMerchantThemeLogoDto = instanceOfMerchantThemeLogoDto;
exports.MerchantThemeLogoDtoFromJSON = MerchantThemeLogoDtoFromJSON;
exports.MerchantThemeLogoDtoFromJSONTyped = MerchantThemeLogoDtoFromJSONTyped;
exports.MerchantThemeLogoDtoToJSON = MerchantThemeLogoDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.MerchantThemeLogoDtoTypeEnum = {
    None: 'NONE',
    Default: 'DEFAULT',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the MerchantThemeLogoDto interface.
 */
function instanceOfMerchantThemeLogoDto(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function MerchantThemeLogoDtoFromJSON(json) {
    return MerchantThemeLogoDtoFromJSONTyped(json, false);
}
function MerchantThemeLogoDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'type': json['type'],
        'customLogo': !(0, runtime_1.exists)(json, 'customLogo') ? undefined : json['customLogo'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantThemeLogoDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'customLogo': value.customLogo,
    };
}
