// @cp/node-sdk/dist/cjs/models/MerchantSettingsEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantSettingsEntity = instanceOfMerchantSettingsEntity;
exports.MerchantSettingsEntityFromJSON = MerchantSettingsEntityFromJSON;
exports.MerchantSettingsEntityFromJSONTyped = MerchantSettingsEntityFromJSONTyped;
exports.MerchantSettingsEntityToJSON = MerchantSettingsEntityToJSON;
const runtime_1 = require("../runtime");
const MerchantThemeEntity_1 = require("./MerchantThemeEntity");
/**
 * Check if a given object implements the MerchantSettingsEntity interface.
 */
function instanceOfMerchantSettingsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "isSplitCFeeEnabled" in value;
    isInstance = isInstance && "isDebtRepaymentEnabled" in value;
    isInstance = isInstance && "isPartialAuthEnabled" in value;
    return isInstance;
}
function MerchantSettingsEntityFromJSON(json) {
    return MerchantSettingsEntityFromJSONTyped(json, false);
}
function MerchantSettingsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': json['merchantId'],
        'checkoutTheme': !(0, runtime_1.exists)(json, 'checkoutTheme') ? undefined : (0, MerchantThemeEntity_1.MerchantThemeEntityFromJSON)(json['checkoutTheme']),
        'isSplitCFeeEnabled': json['isSplitCFeeEnabled'],
        'isDebtRepaymentEnabled': json['isDebtRepaymentEnabled'],
        'isPartialAuthEnabled': json['isPartialAuthEnabled'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantSettingsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'checkoutTheme': (0, MerchantThemeEntity_1.MerchantThemeEntityToJSON)(value.checkoutTheme),
        'isSplitCFeeEnabled': value.isSplitCFeeEnabled,
        'isDebtRepaymentEnabled': value.isDebtRepaymentEnabled,
        'isPartialAuthEnabled': value.isPartialAuthEnabled,
    };
}
