// @cp/node-sdk/dist/cjs/models/MerchantColorsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantColorsDto = instanceOfMerchantColorsDto;
exports.MerchantColorsDtoFromJSON = MerchantColorsDtoFromJSON;
exports.MerchantColorsDtoFromJSONTyped = MerchantColorsDtoFromJSONTyped;
exports.MerchantColorsDtoToJSON = MerchantColorsDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the MerchantColorsDto interface.
 */
function instanceOfMerchantColorsDto(value) {
    let isInstance = true;
    return isInstance;
}
function MerchantColorsDtoFromJSON(json) {
    return MerchantColorsDtoFromJSONTyped(json, false);
}
function MerchantColorsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'primary': !(0, runtime_1.exists)(json, 'primary') ? undefined : json['primary'],
        'secondary': !(0, runtime_1.exists)(json, 'secondary') ? undefined : json['secondary'],
        'tertiary': !(0, runtime_1.exists)(json, 'tertiary') ? undefined : json['tertiary'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantColorsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'primary': value.primary,
        'secondary': value.secondary,
        'tertiary': value.tertiary,
    };
}
