// @cp/node-sdk/dist/cjs/models/MerchantAddressEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantAddressEntity = instanceOfMerchantAddressEntity;
exports.MerchantAddressEntityFromJSON = MerchantAddressEntityFromJSON;
exports.MerchantAddressEntityFromJSONTyped = MerchantAddressEntityFromJSONTyped;
exports.MerchantAddressEntityToJSON = MerchantAddressEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the MerchantAddressEntity interface.
 */
function instanceOfMerchantAddressEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "country" in value;
    isInstance = isInstance && "subdivision" in value;
    isInstance = isInstance && "city" in value;
    isInstance = isInstance && "streetAddress" in value;
    isInstance = isInstance && "postalCode" in value;
    return isInstance;
}
function MerchantAddressEntityFromJSON(json) {
    return MerchantAddressEntityFromJSONTyped(json, false);
}
function MerchantAddressEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'country': json['country'],
        'subdivision': json['subdivision'],
        'city': json['city'],
        'streetAddress': json['streetAddress'],
        'postalCode': json['postalCode'],
        'countryCode': !(0, runtime_1.exists)(json, 'countryCode') ? undefined : json['countryCode'],
        'subdivisionCode': !(0, runtime_1.exists)(json, 'subdivisionCode') ? undefined : json['subdivisionCode'],
        'streetAddress2': !(0, runtime_1.exists)(json, 'streetAddress2') ? undefined : json['streetAddress2'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantAddressEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'country': value.country,
        'subdivision': value.subdivision,
        'city': value.city,
        'streetAddress': value.streetAddress,
        'postalCode': value.postalCode,
        'countryCode': value.countryCode,
        'subdivisionCode': value.subdivisionCode,
        'streetAddress2': value.streetAddress2,
    };
}
