// @cp/node-sdk/dist/cjs/models/MerchantConnectorCredentialsServiceEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantConnectorCredentialsServiceEntity = instanceOfMerchantConnectorCredentialsServiceEntity;
exports.MerchantConnectorCredentialsServiceEntityFromJSON = MerchantConnectorCredentialsServiceEntityFromJSON;
exports.MerchantConnectorCredentialsServiceEntityFromJSONTyped = MerchantConnectorCredentialsServiceEntityFromJSONTyped;
exports.MerchantConnectorCredentialsServiceEntityToJSON = MerchantConnectorCredentialsServiceEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the MerchantConnectorCredentialsServiceEntity interface.
 */
function instanceOfMerchantConnectorCredentialsServiceEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "credentialsId" in value;
    isInstance = isInstance && "connectorId" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    return isInstance;
}
function MerchantConnectorCredentialsServiceEntityFromJSON(json) {
    return MerchantConnectorCredentialsServiceEntityFromJSONTyped(json, false);
}
function MerchantConnectorCredentialsServiceEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'credentialsId': json['credentialsId'],
        'connectorId': json['connectorId'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'username': !(0, runtime_1.exists)(json, 'username') ? undefined : json['username'],
        'password': !(0, runtime_1.exists)(json, 'password') ? undefined : json['password'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantConnectorCredentialsServiceEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'credentialsId': value.credentialsId,
        'connectorId': value.connectorId,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'username': value.username,
        'password': value.password,
    };
}
