// @cp/node-sdk/dist/cjs/models/MerchantThemeEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantThemeEntity = instanceOfMerchantThemeEntity;
exports.MerchantThemeEntityFromJSON = MerchantThemeEntityFromJSON;
exports.MerchantThemeEntityFromJSONTyped = MerchantThemeEntityFromJSONTyped;
exports.MerchantThemeEntityToJSON = MerchantThemeEntityToJSON;
const runtime_1 = require("../runtime");
const MerchantColorsEntity_1 = require("./MerchantColorsEntity");
const MerchantThemeLogoEntity_1 = require("./MerchantThemeLogoEntity");
/**
 * Check if a given object implements the MerchantThemeEntity interface.
 */
function instanceOfMerchantThemeEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "logo" in value;
    return isInstance;
}
function MerchantThemeEntityFromJSON(json) {
    return MerchantThemeEntityFromJSONTyped(json, false);
}
function MerchantThemeEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'enabled': json['enabled'],
        'logo': (0, MerchantThemeLogoEntity_1.MerchantThemeLogoEntityFromJSON)(json['logo']),
        'colors': !(0, runtime_1.exists)(json, 'colors') ? undefined : (0, MerchantColorsEntity_1.MerchantColorsEntityFromJSON)(json['colors']),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantThemeEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
        'logo': (0, MerchantThemeLogoEntity_1.MerchantThemeLogoEntityToJSON)(value.logo),
        'colors': (0, MerchantColorsEntity_1.MerchantColorsEntityToJSON)(value.colors),
    };
}
