// @cp/node-sdk/dist/cjs/models/MerchantThemeDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantThemeDto = instanceOfMerchantThemeDto;
exports.MerchantThemeDtoFromJSON = MerchantThemeDtoFromJSON;
exports.MerchantThemeDtoFromJSONTyped = MerchantThemeDtoFromJSONTyped;
exports.MerchantThemeDtoToJSON = MerchantThemeDtoToJSON;
const runtime_1 = require("../runtime");
const MerchantColorsDto_1 = require("./MerchantColorsDto");
const MerchantThemeLogoDto_1 = require("./MerchantThemeLogoDto");
/**
 * Check if a given object implements the MerchantThemeDto interface.
 */
function instanceOfMerchantThemeDto(value) {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "logo" in value;
    return isInstance;
}
function MerchantThemeDtoFromJSON(json) {
    return MerchantThemeDtoFromJSONTyped(json, false);
}
function MerchantThemeDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'enabled': json['enabled'],
        'logo': (0, MerchantThemeLogoDto_1.MerchantThemeLogoDtoFromJSON)(json['logo']),
        'colors': !(0, runtime_1.exists)(json, 'colors') ? undefined : (0, MerchantColorsDto_1.MerchantColorsDtoFromJSON)(json['colors']),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantThemeDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
        'logo': (0, MerchantThemeLogoDto_1.MerchantThemeLogoDtoToJSON)(value.logo),
        'colors': (0, MerchantColorsDto_1.MerchantColorsDtoToJSON)(value.colors),
    };
}
