// @cp/node-sdk/dist/cjs/models/MerchantAccountServiceEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MerchantAccountServiceEntityCardTypesEnum = exports.MerchantAccountServiceEntityCurrenciesEnum = exports.MerchantAccountServiceEntityCardSchemesEnum = exports.MerchantAccountServiceEntityStatusEnum = exports.MerchantAccountServiceEntityMethodEnum = exports.MerchantAccountServiceEntityPspEnum = void 0;
exports.instanceOfMerchantAccountServiceEntity = instanceOfMerchantAccountServiceEntity;
exports.MerchantAccountServiceEntityFromJSON = MerchantAccountServiceEntityFromJSON;
exports.MerchantAccountServiceEntityFromJSONTyped = MerchantAccountServiceEntityFromJSONTyped;
exports.MerchantAccountServiceEntityToJSON = MerchantAccountServiceEntityToJSON;
const runtime_1 = require("../runtime");
const CFeeEntity_1 = require("./CFeeEntity");
const MerchantConnectorServiceEntity_1 = require("./MerchantConnectorServiceEntity");
/**
 * @export
 */
exports.MerchantAccountServiceEntityPspEnum = {
    Tsys: 'TSYS',
    Nashville: 'NASHVILLE',
    North: 'NORTH',
    ChannelPayments: 'CHANNEL_PAYMENTS'
};
/**
 * @export
 */
exports.MerchantAccountServiceEntityMethodEnum = {
    Card: 'CARD',
    Ach: 'ACH'
};
/**
 * @export
 */
exports.MerchantAccountServiceEntityStatusEnum = {
    Pending: 'PENDING',
    Active: 'ACTIVE',
    Inactive: 'INACTIVE',
    Suspended: 'SUSPENDED',
    Closed: 'CLOSED'
};
/**
 * @export
 */
exports.MerchantAccountServiceEntityCardSchemesEnum = {
    Visa: 'VISA',
    Mastercard: 'MASTERCARD',
    Amex: 'AMEX',
    Discover: 'DISCOVER',
    Jcb: 'JCB',
    DinersClub: 'DINERS_CLUB',
    Maestro: 'MAESTRO'
};
/**
 * @export
 */
exports.MerchantAccountServiceEntityCurrenciesEnum = {
    Usd: 'USD',
    Aud: 'AUD',
    Cad: 'CAD'
};
/**
 * @export
 */
exports.MerchantAccountServiceEntityCardTypesEnum = {
    Credit: 'CREDIT',
    Debit: 'DEBIT',
    Prepaid: 'PREPAID'
};
/**
 * Check if a given object implements the MerchantAccountServiceEntity interface.
 */
function instanceOfMerchantAccountServiceEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantAccountId" in value;
    isInstance = isInstance && "issuedMid" in value;
    isInstance = isInstance && "psp" in value;
    isInstance = isInstance && "cFee" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    isInstance = isInstance && "method" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "cardSchemes" in value;
    isInstance = isInstance && "currencies" in value;
    isInstance = isInstance && "cardTypes" in value;
    return isInstance;
}
function MerchantAccountServiceEntityFromJSON(json) {
    return MerchantAccountServiceEntityFromJSONTyped(json, false);
}
function MerchantAccountServiceEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantAccountId': json['merchantAccountId'],
        'merchantId': !(0, runtime_1.exists)(json, 'merchantId') ? undefined : json['merchantId'],
        'issuedMid': json['issuedMid'],
        'psp': json['psp'],
        'cFee': (0, CFeeEntity_1.CFeeEntityFromJSON)(json['cFee']),
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'method': json['method'],
        'status': json['status'],
        'cardSchemes': json['cardSchemes'],
        'currencies': json['currencies'],
        'cardTypes': json['cardTypes'],
        'isDefault': !(0, runtime_1.exists)(json, 'isDefault') ? undefined : json['isDefault'],
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'connectors': !(0, runtime_1.exists)(json, 'connectors') ? undefined : (json['connectors'].map(MerchantConnectorServiceEntity_1.MerchantConnectorServiceEntityFromJSON)),
        'lastPolledAt': !(0, runtime_1.exists)(json, 'lastPolledAt') ? undefined : (new Date(json['lastPolledAt'])),
        'mcc': !(0, runtime_1.exists)(json, 'mcc') ? undefined : json['mcc'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantAccountServiceEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantAccountId': value.merchantAccountId,
        'merchantId': value.merchantId,
        'issuedMid': value.issuedMid,
        'psp': value.psp,
        'cFee': (0, CFeeEntity_1.CFeeEntityToJSON)(value.cFee),
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'method': value.method,
        'status': value.status,
        'cardSchemes': value.cardSchemes,
        'currencies': value.currencies,
        'cardTypes': value.cardTypes,
        'isDefault': value.isDefault,
        'name': value.name,
        'connectors': value.connectors === undefined ? undefined : (value.connectors.map(MerchantConnectorServiceEntity_1.MerchantConnectorServiceEntityToJSON)),
        'lastPolledAt': value.lastPolledAt === undefined ? undefined : (value.lastPolledAt.toISOString()),
        'mcc': value.mcc,
    };
}
