// @cp/node-sdk/dist/cjs/models/MerchantAccountsSearchEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantAccountsSearchEntity = instanceOfMerchantAccountsSearchEntity;
exports.MerchantAccountsSearchEntityFromJSON = MerchantAccountsSearchEntityFromJSON;
exports.MerchantAccountsSearchEntityFromJSONTyped = MerchantAccountsSearchEntityFromJSONTyped;
exports.MerchantAccountsSearchEntityToJSON = MerchantAccountsSearchEntityToJSON;
const runtime_1 = require("../runtime");
const MerchantAccountServiceEntity_1 = require("./MerchantAccountServiceEntity");
/**
 * Check if a given object implements the MerchantAccountsSearchEntity interface.
 */
function instanceOfMerchantAccountsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantAccounts" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
function MerchantAccountsSearchEntityFromJSON(json) {
    return MerchantAccountsSearchEntityFromJSONTyped(json, false);
}
function MerchantAccountsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantAccounts': (json['merchantAccounts'].map(MerchantAccountServiceEntity_1.MerchantAccountServiceEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !(0, runtime_1.exists)(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantAccountsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantAccounts': (value.merchantAccounts.map(MerchantAccountServiceEntity_1.MerchantAccountServiceEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
