// @cp/node-sdk/dist/cjs/models/MerchantSettingsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantSettingsDto = instanceOfMerchantSettingsDto;
exports.MerchantSettingsDtoFromJSON = MerchantSettingsDtoFromJSON;
exports.MerchantSettingsDtoFromJSONTyped = MerchantSettingsDtoFromJSONTyped;
exports.MerchantSettingsDtoToJSON = MerchantSettingsDtoToJSON;
const runtime_1 = require("../runtime");
const MerchantThemeDto_1 = require("./MerchantThemeDto");
/**
 * Check if a given object implements the MerchantSettingsDto interface.
 */
function instanceOfMerchantSettingsDto(value) {
    let isInstance = true;
    return isInstance;
}
function MerchantSettingsDtoFromJSON(json) {
    return MerchantSettingsDtoFromJSONTyped(json, false);
}
function MerchantSettingsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'checkoutTheme': !(0, runtime_1.exists)(json, 'checkoutTheme') ? undefined : (0, MerchantThemeDto_1.MerchantThemeDtoFromJSON)(json['checkoutTheme']),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantSettingsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'checkoutTheme': (0, MerchantThemeDto_1.MerchantThemeDtoToJSON)(value.checkoutTheme),
    };
}
