// @cp/node-sdk/dist/cjs/models/MerchantThemeLogoEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.MerchantThemeLogoEntityTypeEnum = void 0;
exports.instanceOfMerchantThemeLogoEntity = instanceOfMerchantThemeLogoEntity;
exports.MerchantThemeLogoEntityFromJSON = MerchantThemeLogoEntityFromJSON;
exports.MerchantThemeLogoEntityFromJSONTyped = MerchantThemeLogoEntityFromJSONTyped;
exports.MerchantThemeLogoEntityToJSON = MerchantThemeLogoEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.MerchantThemeLogoEntityTypeEnum = {
    None: 'NONE',
    Default: 'DEFAULT',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the MerchantThemeLogoEntity interface.
 */
function instanceOfMerchantThemeLogoEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
function MerchantThemeLogoEntityFromJSON(json) {
    return MerchantThemeLogoEntityFromJSONTyped(json, false);
}
function MerchantThemeLogoEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'type': json['type'],
        'customLogo': !(0, runtime_1.exists)(json, 'customLogo') ? undefined : json['customLogo'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantThemeLogoEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'customLogo': value.customLogo,
    };
}
