// @cp/node-sdk/dist/cjs/models/MerchantFeeDetailsEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfMerchantFeeDetailsEntity = instanceOfMerchantFeeDetailsEntity;
exports.MerchantFeeDetailsEntityFromJSON = MerchantFeeDetailsEntityFromJSON;
exports.MerchantFeeDetailsEntityFromJSONTyped = MerchantFeeDetailsEntityFromJSONTyped;
exports.MerchantFeeDetailsEntityToJSON = MerchantFeeDetailsEntityToJSON;
const runtime_1 = require("../runtime");
const PaymentMethodFeeDetailsEntity_1 = require("./PaymentMethodFeeDetailsEntity");
/**
 * Check if a given object implements the MerchantFeeDetailsEntity interface.
 */
function instanceOfMerchantFeeDetailsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "subtotal" in value;
    isInstance = isInstance && "feeDetails" in value;
    return isInstance;
}
function MerchantFeeDetailsEntityFromJSON(json) {
    return MerchantFeeDetailsEntityFromJSONTyped(json, false);
}
function MerchantFeeDetailsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': json['merchantId'],
        'subtotal': json['subtotal'],
        'feeDetails': (json['feeDetails'].map(PaymentMethodFeeDetailsEntity_1.PaymentMethodFeeDetailsEntityFromJSON)),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function MerchantFeeDetailsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'subtotal': value.subtotal,
        'feeDetails': (value.feeDetails.map(PaymentMethodFeeDetailsEntity_1.PaymentMethodFeeDetailsEntityToJSON)),
    };
}
