// @cp/node-sdk/dist/cjs/models/PaymentMethodDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodDtoAccountTypeEnum = exports.PaymentMethodDtoMethodEnum = void 0;
exports.instanceOfPaymentMethodDto = instanceOfPaymentMethodDto;
exports.PaymentMethodDtoFromJSON = PaymentMethodDtoFromJSON;
exports.PaymentMethodDtoFromJSONTyped = PaymentMethodDtoFromJSONTyped;
exports.PaymentMethodDtoToJSON = PaymentMethodDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.PaymentMethodDtoMethodEnum = {
    Ach: 'ACH',
    Bank: 'BANK',
    Card: 'CARD'
};
/**
 * @export
 */
exports.PaymentMethodDtoAccountTypeEnum = {
    Checking: 'CHECKING',
    Savings: 'SAVINGS'
};
/**
 * Check if a given object implements the PaymentMethodDto interface.
 */
function instanceOfPaymentMethodDto(value) {
    let isInstance = true;
    isInstance = isInstance && "method" in value;
    return isInstance;
}
function PaymentMethodDtoFromJSON(json) {
    return PaymentMethodDtoFromJSONTyped(json, false);
}
function PaymentMethodDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'method': json['method'],
        'number': !(0, runtime_1.exists)(json, 'number') ? undefined : json['number'],
        'securityCode': !(0, runtime_1.exists)(json, 'securityCode') ? undefined : json['securityCode'],
        'expirationDate': !(0, runtime_1.exists)(json, 'expirationDate') ? undefined : json['expirationDate'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'buyerId': !(0, runtime_1.exists)(json, 'buyerId') ? undefined : json['buyerId'],
        'paymentMethodId': !(0, runtime_1.exists)(json, 'paymentMethodId') ? undefined : json['paymentMethodId'],
        'store': !(0, runtime_1.exists)(json, 'store') ? undefined : json['store'],
        'abaNumber': !(0, runtime_1.exists)(json, 'abaNumber') ? undefined : json['abaNumber'],
        'accountNumber': !(0, runtime_1.exists)(json, 'accountNumber') ? undefined : json['accountNumber'],
        'accountType': !(0, runtime_1.exists)(json, 'accountType') ? undefined : json['accountType'],
        'accountHolderName': !(0, runtime_1.exists)(json, 'accountHolderName') ? undefined : json['accountHolderName'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function PaymentMethodDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'method': value.method,
        'number': value.number,
        'securityCode': value.securityCode,
        'expirationDate': value.expirationDate,
        'externalBuyerId': value.externalBuyerId,
        'buyerId': value.buyerId,
        'paymentMethodId': value.paymentMethodId,
        'store': value.store,
        'abaNumber': value.abaNumber,
        'accountNumber': value.accountNumber,
        'accountType': value.accountType,
        'accountHolderName': value.accountHolderName,
    };
}
