// @cp/node-sdk/dist/cjs/models/PaymentMethodSessionAttemptEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodSessionAttemptEntityStatusEnum = exports.PaymentMethodSessionAttemptEntityMethodEnum = void 0;
exports.instanceOfPaymentMethodSessionAttemptEntity = instanceOfPaymentMethodSessionAttemptEntity;
exports.PaymentMethodSessionAttemptEntityFromJSON = PaymentMethodSessionAttemptEntityFromJSON;
exports.PaymentMethodSessionAttemptEntityFromJSONTyped = PaymentMethodSessionAttemptEntityFromJSONTyped;
exports.PaymentMethodSessionAttemptEntityToJSON = PaymentMethodSessionAttemptEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.PaymentMethodSessionAttemptEntityMethodEnum = {
    Card: 'CARD',
    Bank: 'BANK'
};
/**
 * @export
 */
exports.PaymentMethodSessionAttemptEntityStatusEnum = {
    Success: 'SUCCESS',
    Failed: 'FAILED'
};
/**
 * Check if a given object implements the PaymentMethodSessionAttemptEntity interface.
 */
function instanceOfPaymentMethodSessionAttemptEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "method" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "attemptedAt" in value;
    return isInstance;
}
function PaymentMethodSessionAttemptEntityFromJSON(json) {
    return PaymentMethodSessionAttemptEntityFromJSONTyped(json, false);
}
function PaymentMethodSessionAttemptEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'method': json['method'],
        'status': json['status'],
        'attemptedAt': (new Date(json['attemptedAt'])),
        'reason': !(0, runtime_1.exists)(json, 'reason') ? undefined : json['reason'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function PaymentMethodSessionAttemptEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'method': value.method,
        'status': value.status,
        'attemptedAt': (value.attemptedAt.toISOString()),
        'reason': value.reason,
    };
}
