// @cp/node-sdk/dist/cjs/models/PaymentMethodFeeDetailsEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.PaymentMethodFeeDetailsEntityMethodEnum = void 0;
exports.instanceOfPaymentMethodFeeDetailsEntity = instanceOfPaymentMethodFeeDetailsEntity;
exports.PaymentMethodFeeDetailsEntityFromJSON = PaymentMethodFeeDetailsEntityFromJSON;
exports.PaymentMethodFeeDetailsEntityFromJSONTyped = PaymentMethodFeeDetailsEntityFromJSONTyped;
exports.PaymentMethodFeeDetailsEntityToJSON = PaymentMethodFeeDetailsEntityToJSON;
const runtime_1 = require("../runtime");
const FeeDetailsEntity_1 = require("./FeeDetailsEntity");
/**
 * @export
 */
exports.PaymentMethodFeeDetailsEntityMethodEnum = {
    Ach: 'ACH',
    Bank: 'BANK',
    Card: 'CARD'
};
/**
 * Check if a given object implements the PaymentMethodFeeDetailsEntity interface.
 */
function instanceOfPaymentMethodFeeDetailsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "method" in value;
    isInstance = isInstance && "fees" in value;
    isInstance = isInstance && "feesTotal" in value;
    isInstance = isInstance && "adjustedTotal" in value;
    return isInstance;
}
function PaymentMethodFeeDetailsEntityFromJSON(json) {
    return PaymentMethodFeeDetailsEntityFromJSONTyped(json, false);
}
function PaymentMethodFeeDetailsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'method': json['method'],
        'fees': (json['fees'].map(FeeDetailsEntity_1.FeeDetailsEntityFromJSON)),
        'feesTotal': json['feesTotal'],
        'adjustedTotal': json['adjustedTotal'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function PaymentMethodFeeDetailsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'method': value.method,
        'fees': (value.fees.map(FeeDetailsEntity_1.FeeDetailsEntityToJSON)),
        'feesTotal': value.feesTotal,
        'adjustedTotal': value.adjustedTotal,
    };
}
