// @cp/node-sdk/dist/cjs/models/PaymentMethodsSearchEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfPaymentMethodsSearchEntity = instanceOfPaymentMethodsSearchEntity;
exports.PaymentMethodsSearchEntityFromJSON = PaymentMethodsSearchEntityFromJSON;
exports.PaymentMethodsSearchEntityFromJSONTyped = PaymentMethodsSearchEntityFromJSONTyped;
exports.PaymentMethodsSearchEntityToJSON = PaymentMethodsSearchEntityToJSON;
const runtime_1 = require("../runtime");
const PaymentMethodEntity_1 = require("./PaymentMethodEntity");
/**
 * Check if a given object implements the PaymentMethodsSearchEntity interface.
 */
function instanceOfPaymentMethodsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethods" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
function PaymentMethodsSearchEntityFromJSON(json) {
    return PaymentMethodsSearchEntityFromJSONTyped(json, false);
}
function PaymentMethodsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethods': (json['paymentMethods'].map(PaymentMethodEntity_1.PaymentMethodEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !(0, runtime_1.exists)(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function PaymentMethodsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethods': (value.paymentMethods.map(PaymentMethodEntity_1.PaymentMethodEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
