// @cp/node-sdk/dist/cjs/models/ReportCreatorEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfReportCreatorEntity = instanceOfReportCreatorEntity;
exports.ReportCreatorEntityFromJSON = ReportCreatorEntityFromJSON;
exports.ReportCreatorEntityFromJSONTyped = ReportCreatorEntityFromJSONTyped;
exports.ReportCreatorEntityToJSON = ReportCreatorEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ReportCreatorEntity interface.
 */
function instanceOfReportCreatorEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "userId" in value;
    isInstance = isInstance && "fullName" in value;
    return isInstance;
}
function ReportCreatorEntityFromJSON(json) {
    return ReportCreatorEntityFromJSONTyped(json, false);
}
function ReportCreatorEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'userId': json['userId'],
        'fullName': json['fullName'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function ReportCreatorEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'userId': value.userId,
        'fullName': value.fullName,
    };
}
