// @cp/node-sdk/dist/cjs/models/ReturnNotificationEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfReturnNotificationEntity = instanceOfReturnNotificationEntity;
exports.ReturnNotificationEntityFromJSON = ReturnNotificationEntityFromJSON;
exports.ReturnNotificationEntityFromJSONTyped = ReturnNotificationEntityFromJSONTyped;
exports.ReturnNotificationEntityToJSON = ReturnNotificationEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ReturnNotificationEntity interface.
 */
function instanceOfReturnNotificationEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "returnNotificationId" in value;
    isInstance = isInstance && "returnCode" in value;
    isInstance = isInstance && "transactionId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "merchantAccountId" in value;
    isInstance = isInstance && "acknowledged" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    return isInstance;
}
function ReturnNotificationEntityFromJSON(json) {
    return ReturnNotificationEntityFromJSONTyped(json, false);
}
function ReturnNotificationEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'returnNotificationId': json['returnNotificationId'],
        'returnedDate': !(0, runtime_1.exists)(json, 'returnedDate') ? undefined : (new Date(json['returnedDate'])),
        'returnCode': json['returnCode'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'transactionId': json['transactionId'],
        'merchantId': json['merchantId'],
        'merchantAccountId': json['merchantAccountId'],
        'acknowledged': json['acknowledged'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'acknowledgedDate': !(0, runtime_1.exists)(json, 'acknowledgedDate') ? undefined : (new Date(json['acknowledgedDate'])),
        'customerId': !(0, runtime_1.exists)(json, 'customerId') ? undefined : json['customerId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function ReturnNotificationEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'returnNotificationId': value.returnNotificationId,
        'returnedDate': value.returnedDate === undefined ? undefined : (value.returnedDate.toISOString()),
        'returnCode': value.returnCode,
        'description': value.description,
        'transactionId': value.transactionId,
        'merchantId': value.merchantId,
        'merchantAccountId': value.merchantAccountId,
        'acknowledged': value.acknowledged,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'acknowledgedDate': value.acknowledgedDate === undefined ? undefined : (value.acknowledgedDate.toISOString()),
        'customerId': value.customerId,
    };
}
