// @cp/node-sdk/dist/cjs/models/TagDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTagDto = instanceOfTagDto;
exports.TagDtoFromJSON = TagDtoFromJSON;
exports.TagDtoFromJSONTyped = TagDtoFromJSONTyped;
exports.TagDtoToJSON = TagDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the TagDto interface.
 */
function instanceOfTagDto(value) {
    let isInstance = true;
    isInstance = isInstance && "key" in value;
    isInstance = isInstance && "value" in value;
    return isInstance;
}
function TagDtoFromJSON(json) {
    return TagDtoFromJSONTyped(json, false);
}
function TagDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'key': json['key'],
        'value': json['value'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TagDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'key': value.key,
        'value': value.value,
    };
}
