// @cp/node-sdk/dist/cjs/models/TagEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTagEntity = instanceOfTagEntity;
exports.TagEntityFromJSON = TagEntityFromJSON;
exports.TagEntityFromJSONTyped = TagEntityFromJSONTyped;
exports.TagEntityToJSON = TagEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the TagEntity interface.
 */
function instanceOfTagEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "key" in value;
    isInstance = isInstance && "value" in value;
    return isInstance;
}
function TagEntityFromJSON(json) {
    return TagEntityFromJSONTyped(json, false);
}
function TagEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'key': json['key'],
        'value': json['value'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TagEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'key': value.key,
        'value': value.value,
    };
}
