// @cp/node-sdk/dist/cjs/models/TransactionsReportsSearchQueryResponseEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTransactionsReportsSearchQueryResponseEntity = instanceOfTransactionsReportsSearchQueryResponseEntity;
exports.TransactionsReportsSearchQueryResponseEntityFromJSON = TransactionsReportsSearchQueryResponseEntityFromJSON;
exports.TransactionsReportsSearchQueryResponseEntityFromJSONTyped = TransactionsReportsSearchQueryResponseEntityFromJSONTyped;
exports.TransactionsReportsSearchQueryResponseEntityToJSON = TransactionsReportsSearchQueryResponseEntityToJSON;
const runtime_1 = require("../runtime");
const TransactionsReportEntity_1 = require("./TransactionsReportEntity");
/**
 * Check if a given object implements the TransactionsReportsSearchQueryResponseEntity interface.
 */
function instanceOfTransactionsReportsSearchQueryResponseEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    isInstance = isInstance && "items" in value;
    return isInstance;
}
function TransactionsReportsSearchQueryResponseEntityFromJSON(json) {
    return TransactionsReportsSearchQueryResponseEntityFromJSONTyped(json, false);
}
function TransactionsReportsSearchQueryResponseEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !(0, runtime_1.exists)(json, 'prevCursor') ? undefined : json['prevCursor'],
        'items': (json['items'].map(TransactionsReportEntity_1.TransactionsReportEntityFromJSON)),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TransactionsReportsSearchQueryResponseEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
        'items': (value.items.map(TransactionsReportEntity_1.TransactionsReportEntityToJSON)),
    };
}
