// @cp/node-sdk/dist/cjs/models/TransactionsReportSchedulesSearchQueryResponseEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTransactionsReportSchedulesSearchQueryResponseEntity = instanceOfTransactionsReportSchedulesSearchQueryResponseEntity;
exports.TransactionsReportSchedulesSearchQueryResponseEntityFromJSON = TransactionsReportSchedulesSearchQueryResponseEntityFromJSON;
exports.TransactionsReportSchedulesSearchQueryResponseEntityFromJSONTyped = TransactionsReportSchedulesSearchQueryResponseEntityFromJSONTyped;
exports.TransactionsReportSchedulesSearchQueryResponseEntityToJSON = TransactionsReportSchedulesSearchQueryResponseEntityToJSON;
const runtime_1 = require("../runtime");
const TransactionsReportScheduleEntity_1 = require("./TransactionsReportScheduleEntity");
/**
 * Check if a given object implements the TransactionsReportSchedulesSearchQueryResponseEntity interface.
 */
function instanceOfTransactionsReportSchedulesSearchQueryResponseEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    isInstance = isInstance && "items" in value;
    return isInstance;
}
function TransactionsReportSchedulesSearchQueryResponseEntityFromJSON(json) {
    return TransactionsReportSchedulesSearchQueryResponseEntityFromJSONTyped(json, false);
}
function TransactionsReportSchedulesSearchQueryResponseEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !(0, runtime_1.exists)(json, 'prevCursor') ? undefined : json['prevCursor'],
        'items': (json['items'].map(TransactionsReportScheduleEntity_1.TransactionsReportScheduleEntityFromJSON)),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TransactionsReportSchedulesSearchQueryResponseEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
        'items': (value.items.map(TransactionsReportScheduleEntity_1.TransactionsReportScheduleEntityToJSON)),
    };
}
