// @cp/node-sdk/dist/cjs/models/TransactionRefundsSearchEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTransactionRefundsSearchEntity = instanceOfTransactionRefundsSearchEntity;
exports.TransactionRefundsSearchEntityFromJSON = TransactionRefundsSearchEntityFromJSON;
exports.TransactionRefundsSearchEntityFromJSONTyped = TransactionRefundsSearchEntityFromJSONTyped;
exports.TransactionRefundsSearchEntityToJSON = TransactionRefundsSearchEntityToJSON;
const runtime_1 = require("../runtime");
const TransactionRefundBaseEntity_1 = require("./TransactionRefundBaseEntity");
/**
 * Check if a given object implements the TransactionRefundsSearchEntity interface.
 */
function instanceOfTransactionRefundsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "refunds" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "prevCursor" in value;
    return isInstance;
}
function TransactionRefundsSearchEntityFromJSON(json) {
    return TransactionRefundsSearchEntityFromJSONTyped(json, false);
}
function TransactionRefundsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'refunds': (json['refunds'].map(TransactionRefundBaseEntity_1.TransactionRefundBaseEntityFromJSON)),
        'limit': json['limit'],
        'page': !(0, runtime_1.exists)(json, 'page') ? undefined : json['page'],
        'pageCount': !(0, runtime_1.exists)(json, 'pageCount') ? undefined : json['pageCount'],
        'totalCount': !(0, runtime_1.exists)(json, 'totalCount') ? undefined : json['totalCount'],
        'filteredCount': !(0, runtime_1.exists)(json, 'filteredCount') ? undefined : json['filteredCount'],
        'prevCursor': json['prevCursor'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TransactionRefundsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'refunds': (value.refunds.map(TransactionRefundBaseEntity_1.TransactionRefundBaseEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'prevCursor': value.prevCursor,
        'nextCursor': value.nextCursor,
    };
}
