// @cp/node-sdk/dist/cjs/models/TransactionNoteHistoryEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTransactionNoteHistoryEntity = instanceOfTransactionNoteHistoryEntity;
exports.TransactionNoteHistoryEntityFromJSON = TransactionNoteHistoryEntityFromJSON;
exports.TransactionNoteHistoryEntityFromJSONTyped = TransactionNoteHistoryEntityFromJSONTyped;
exports.TransactionNoteHistoryEntityToJSON = TransactionNoteHistoryEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the TransactionNoteHistoryEntity interface.
 */
function instanceOfTransactionNoteHistoryEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "content" in value;
    isInstance = isInstance && "modifiedBy" in value;
    isInstance = isInstance && "modifiedAt" in value;
    return isInstance;
}
function TransactionNoteHistoryEntityFromJSON(json) {
    return TransactionNoteHistoryEntityFromJSONTyped(json, false);
}
function TransactionNoteHistoryEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'content': json['content'],
        'modifiedBy': json['modifiedBy'],
        'modifiedAt': (new Date(json['modifiedAt'])),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TransactionNoteHistoryEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'content': value.content,
        'modifiedBy': value.modifiedBy,
        'modifiedAt': (value.modifiedAt.toISOString()),
    };
}
