// @cp/node-sdk/dist/cjs/models/TransactionPaymentInstrumentBaseEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionPaymentInstrumentBaseEntityAccountTypeEnum = exports.TransactionPaymentInstrumentBaseEntityCardTypeEnum = exports.TransactionPaymentInstrumentBaseEntityMethodEnum = void 0;
exports.instanceOfTransactionPaymentInstrumentBaseEntity = instanceOfTransactionPaymentInstrumentBaseEntity;
exports.TransactionPaymentInstrumentBaseEntityFromJSON = TransactionPaymentInstrumentBaseEntityFromJSON;
exports.TransactionPaymentInstrumentBaseEntityFromJSONTyped = TransactionPaymentInstrumentBaseEntityFromJSONTyped;
exports.TransactionPaymentInstrumentBaseEntityToJSON = TransactionPaymentInstrumentBaseEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.TransactionPaymentInstrumentBaseEntityMethodEnum = {
    Ach: 'ACH',
    Bank: 'BANK',
    Card: 'CARD'
};
/**
 * @export
 */
exports.TransactionPaymentInstrumentBaseEntityCardTypeEnum = {
    Credit: 'CREDIT',
    Debit: 'DEBIT',
    Prepaid: 'PREPAID'
};
/**
 * @export
 */
exports.TransactionPaymentInstrumentBaseEntityAccountTypeEnum = {
    Checking: 'CHECKING',
    Savings: 'SAVINGS'
};
/**
 * Check if a given object implements the TransactionPaymentInstrumentBaseEntity interface.
 */
function instanceOfTransactionPaymentInstrumentBaseEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethodId" in value;
    isInstance = isInstance && "method" in value;
    return isInstance;
}
function TransactionPaymentInstrumentBaseEntityFromJSON(json) {
    return TransactionPaymentInstrumentBaseEntityFromJSONTyped(json, false);
}
function TransactionPaymentInstrumentBaseEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethodId': json['paymentMethodId'],
        'method': json['method'],
        'externalPaymentMethodId': !(0, runtime_1.exists)(json, 'externalPaymentMethodId') ? undefined : json['externalPaymentMethodId'],
        'label': !(0, runtime_1.exists)(json, 'label') ? undefined : json['label'],
        'scheme': !(0, runtime_1.exists)(json, 'scheme') ? undefined : json['scheme'],
        'expirationDate': !(0, runtime_1.exists)(json, 'expirationDate') ? undefined : json['expirationDate'],
        'currency': !(0, runtime_1.exists)(json, 'currency') ? undefined : json['currency'],
        'country': !(0, runtime_1.exists)(json, 'country') ? undefined : json['country'],
        'bin': !(0, runtime_1.exists)(json, 'bin') ? undefined : json['bin'],
        'cardType': !(0, runtime_1.exists)(json, 'cardType') ? undefined : json['cardType'],
        'accountHolderName': !(0, runtime_1.exists)(json, 'accountHolderName') ? undefined : json['accountHolderName'],
        'accountNumber': !(0, runtime_1.exists)(json, 'accountNumber') ? undefined : json['accountNumber'],
        'accountEnding': !(0, runtime_1.exists)(json, 'accountEnding') ? undefined : json['accountEnding'],
        'accountType': !(0, runtime_1.exists)(json, 'accountType') ? undefined : json['accountType'],
        'abaNumber': !(0, runtime_1.exists)(json, 'abaNumber') ? undefined : json['abaNumber'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TransactionPaymentInstrumentBaseEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethodId': value.paymentMethodId,
        'method': value.method,
        'externalPaymentMethodId': value.externalPaymentMethodId,
        'label': value.label,
        'scheme': value.scheme,
        'expirationDate': value.expirationDate,
        'currency': value.currency,
        'country': value.country,
        'bin': value.bin,
        'cardType': value.cardType,
        'accountHolderName': value.accountHolderName,
        'accountNumber': value.accountNumber,
        'accountEnding': value.accountEnding,
        'accountType': value.accountType,
        'abaNumber': value.abaNumber,
    };
}
