// @cp/node-sdk/dist/cjs/models/TransactionsSearchEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTransactionsSearchEntity = instanceOfTransactionsSearchEntity;
exports.TransactionsSearchEntityFromJSON = TransactionsSearchEntityFromJSON;
exports.TransactionsSearchEntityFromJSONTyped = TransactionsSearchEntityFromJSONTyped;
exports.TransactionsSearchEntityToJSON = TransactionsSearchEntityToJSON;
const runtime_1 = require("../runtime");
const TransactionEntity_1 = require("./TransactionEntity");
/**
 * Check if a given object implements the TransactionsSearchEntity interface.
 */
function instanceOfTransactionsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "transactions" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
function TransactionsSearchEntityFromJSON(json) {
    return TransactionsSearchEntityFromJSONTyped(json, false);
}
function TransactionsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'transactions': (json['transactions'].map(TransactionEntity_1.TransactionEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !(0, runtime_1.exists)(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TransactionsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transactions': (value.transactions.map(TransactionEntity_1.TransactionEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
