// @cp/node-sdk/dist/cjs/models/TransactionsReportTemplatesSearchQueryResponseEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTransactionsReportTemplatesSearchQueryResponseEntity = instanceOfTransactionsReportTemplatesSearchQueryResponseEntity;
exports.TransactionsReportTemplatesSearchQueryResponseEntityFromJSON = TransactionsReportTemplatesSearchQueryResponseEntityFromJSON;
exports.TransactionsReportTemplatesSearchQueryResponseEntityFromJSONTyped = TransactionsReportTemplatesSearchQueryResponseEntityFromJSONTyped;
exports.TransactionsReportTemplatesSearchQueryResponseEntityToJSON = TransactionsReportTemplatesSearchQueryResponseEntityToJSON;
const runtime_1 = require("../runtime");
const TransactionsReportTemplateEntity_1 = require("./TransactionsReportTemplateEntity");
/**
 * Check if a given object implements the TransactionsReportTemplatesSearchQueryResponseEntity interface.
 */
function instanceOfTransactionsReportTemplatesSearchQueryResponseEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    isInstance = isInstance && "items" in value;
    return isInstance;
}
function TransactionsReportTemplatesSearchQueryResponseEntityFromJSON(json) {
    return TransactionsReportTemplatesSearchQueryResponseEntityFromJSONTyped(json, false);
}
function TransactionsReportTemplatesSearchQueryResponseEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !(0, runtime_1.exists)(json, 'prevCursor') ? undefined : json['prevCursor'],
        'items': (json['items'].map(TransactionsReportTemplateEntity_1.TransactionsReportTemplateEntityFromJSON)),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TransactionsReportTemplatesSearchQueryResponseEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
        'items': (value.items.map(TransactionsReportTemplateEntity_1.TransactionsReportTemplateEntityToJSON)),
    };
}
