// @cp/node-sdk/dist/cjs/models/TransactionNoteEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfTransactionNoteEntity = instanceOfTransactionNoteEntity;
exports.TransactionNoteEntityFromJSON = TransactionNoteEntityFromJSON;
exports.TransactionNoteEntityFromJSONTyped = TransactionNoteEntityFromJSONTyped;
exports.TransactionNoteEntityToJSON = TransactionNoteEntityToJSON;
const runtime_1 = require("../runtime");
const TransactionNoteHistoryEntity_1 = require("./TransactionNoteHistoryEntity");
/**
 * Check if a given object implements the TransactionNoteEntity interface.
 */
function instanceOfTransactionNoteEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "noteId" in value;
    isInstance = isInstance && "refundId" in value;
    isInstance = isInstance && "content" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "createdBy" in value;
    return isInstance;
}
function TransactionNoteEntityFromJSON(json) {
    return TransactionNoteEntityFromJSONTyped(json, false);
}
function TransactionNoteEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'noteId': json['noteId'],
        'refundId': json['refundId'],
        'content': json['content'],
        'createdAt': (new Date(json['createdAt'])),
        'createdBy': json['createdBy'],
        'history': !(0, runtime_1.exists)(json, 'history') ? undefined : (json['history'].map(TransactionNoteHistoryEntity_1.TransactionNoteHistoryEntityFromJSON)),
        'updatedBy': !(0, runtime_1.exists)(json, 'updatedBy') ? undefined : json['updatedBy'],
        'deleted': !(0, runtime_1.exists)(json, 'deleted') ? undefined : json['deleted'],
        'deletedAt': !(0, runtime_1.exists)(json, 'deletedAt') ? undefined : (new Date(json['deletedAt'])),
        'deletedBy': !(0, runtime_1.exists)(json, 'deletedBy') ? undefined : json['deletedBy'],
        'updatedAt': !(0, runtime_1.exists)(json, 'updatedAt') ? undefined : (new Date(json['updatedAt'])),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TransactionNoteEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'noteId': value.noteId,
        'refundId': value.refundId,
        'content': value.content,
        'createdAt': (value.createdAt.toISOString()),
        'createdBy': value.createdBy,
        'history': value.history === undefined ? undefined : (value.history.map(TransactionNoteHistoryEntity_1.TransactionNoteHistoryEntityToJSON)),
        'updatedBy': value.updatedBy,
        'deleted': value.deleted,
        'deletedAt': value.deletedAt === undefined ? undefined : (value.deletedAt.toISOString()),
        'deletedBy': value.deletedBy,
        'updatedAt': value.updatedAt === undefined ? undefined : (value.updatedAt.toISOString()),
    };
}
