// @cp/node-sdk/dist/cjs/models/TransactionsReportTemplateEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.TransactionsReportTemplateEntityDateRangeEnum = exports.TransactionsReportTemplateEntityOrderEnum = exports.TransactionsReportTemplateEntityReportDomainEnum = void 0;
exports.instanceOfTransactionsReportTemplateEntity = instanceOfTransactionsReportTemplateEntity;
exports.TransactionsReportTemplateEntityFromJSON = TransactionsReportTemplateEntityFromJSON;
exports.TransactionsReportTemplateEntityFromJSONTyped = TransactionsReportTemplateEntityFromJSONTyped;
exports.TransactionsReportTemplateEntityToJSON = TransactionsReportTemplateEntityToJSON;
const runtime_1 = require("../runtime");
const ReportCreatorEntity_1 = require("./ReportCreatorEntity");
const TransactionsReportsFiltersEntity_1 = require("./TransactionsReportsFiltersEntity");
/**
 * @export
 */
exports.TransactionsReportTemplateEntityReportDomainEnum = {
    Transactions: 'TRANSACTIONS'
};
/**
 * @export
 */
exports.TransactionsReportTemplateEntityOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * @export
 */
exports.TransactionsReportTemplateEntityDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the TransactionsReportTemplateEntity interface.
 */
function instanceOfTransactionsReportTemplateEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "reportTemplateId" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "reportDomain" in value;
    isInstance = isInstance && "isFavorite" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    isInstance = isInstance && "dateRange" in value;
    isInstance = isInstance && "startTime" in value;
    isInstance = isInstance && "endTime" in value;
    isInstance = isInstance && "columns" in value;
    isInstance = isInstance && "transactionsFilters" in value;
    return isInstance;
}
function TransactionsReportTemplateEntityFromJSON(json) {
    return TransactionsReportTemplateEntityFromJSONTyped(json, false);
}
function TransactionsReportTemplateEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'reportTemplateId': json['reportTemplateId'],
        'merchantId': !(0, runtime_1.exists)(json, 'merchantId') ? undefined : json['merchantId'],
        'createdBy': (0, ReportCreatorEntity_1.ReportCreatorEntityFromJSON)(json['createdBy']),
        'reportDomain': json['reportDomain'],
        'isFavorite': json['isFavorite'],
        'name': json['name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'order': !(0, runtime_1.exists)(json, 'order') ? undefined : json['order'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'deleted': !(0, runtime_1.exists)(json, 'deleted') ? undefined : json['deleted'],
        'deletedAt': !(0, runtime_1.exists)(json, 'deletedAt') ? undefined : (new Date(json['deletedAt'])),
        'dateRange': json['dateRange'],
        'startTime': json['startTime'],
        'endTime': json['endTime'],
        'columns': json['columns'],
        'transactionsFilters': (0, TransactionsReportsFiltersEntity_1.TransactionsReportsFiltersEntityFromJSON)(json['transactionsFilters']),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function TransactionsReportTemplateEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'reportTemplateId': value.reportTemplateId,
        'merchantId': value.merchantId,
        'createdBy': (0, ReportCreatorEntity_1.ReportCreatorEntityToJSON)(value.createdBy),
        'reportDomain': value.reportDomain,
        'isFavorite': value.isFavorite,
        'name': value.name,
        'description': value.description,
        'order': value.order,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'deleted': value.deleted,
        'deletedAt': value.deletedAt === undefined ? undefined : (value.deletedAt.toISOString()),
        'dateRange': value.dateRange,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'columns': value.columns,
        'transactionsFilters': (0, TransactionsReportsFiltersEntity_1.TransactionsReportsFiltersEntityToJSON)(value.transactionsFilters),
    };
}
