// @cp/node-sdk/dist/cjs/models/UpdateMerchantConnectorCredentialsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateMerchantConnectorCredentialsDto = instanceOfUpdateMerchantConnectorCredentialsDto;
exports.UpdateMerchantConnectorCredentialsDtoFromJSON = UpdateMerchantConnectorCredentialsDtoFromJSON;
exports.UpdateMerchantConnectorCredentialsDtoFromJSONTyped = UpdateMerchantConnectorCredentialsDtoFromJSONTyped;
exports.UpdateMerchantConnectorCredentialsDtoToJSON = UpdateMerchantConnectorCredentialsDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UpdateMerchantConnectorCredentialsDto interface.
 */
function instanceOfUpdateMerchantConnectorCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "credentialsId" in value;
    return isInstance;
}
function UpdateMerchantConnectorCredentialsDtoFromJSON(json) {
    return UpdateMerchantConnectorCredentialsDtoFromJSONTyped(json, false);
}
function UpdateMerchantConnectorCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'username': !(0, runtime_1.exists)(json, 'username') ? undefined : json['username'],
        'password': !(0, runtime_1.exists)(json, 'password') ? undefined : json['password'],
        'credentialsId': json['credentialsId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateMerchantConnectorCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'password': value.password,
        'credentialsId': value.credentialsId,
    };
}
