// @cp/node-sdk/dist/cjs/models/UpdateSelfDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateSelfDtoStatusEnum = exports.UpdateSelfDtoRoleEnum = void 0;
exports.instanceOfUpdateSelfDto = instanceOfUpdateSelfDto;
exports.UpdateSelfDtoFromJSON = UpdateSelfDtoFromJSON;
exports.UpdateSelfDtoFromJSONTyped = UpdateSelfDtoFromJSONTyped;
exports.UpdateSelfDtoToJSON = UpdateSelfDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.UpdateSelfDtoRoleEnum = {
    User: 'user',
    Agent: 'agent',
    Admin: 'admin',
    SuperAdmin: 'superAdmin'
};
/**
 * @export
 */
exports.UpdateSelfDtoStatusEnum = {
    Invited: 'invited',
    Active: 'active',
    Inactive: 'inactive',
    Blocked: 'blocked'
};
/**
 * Check if a given object implements the UpdateSelfDto interface.
 */
function instanceOfUpdateSelfDto(value) {
    let isInstance = true;
    return isInstance;
}
function UpdateSelfDtoFromJSON(json) {
    return UpdateSelfDtoFromJSONTyped(json, false);
}
function UpdateSelfDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': !(0, runtime_1.exists)(json, 'givenName') ? undefined : json['givenName'],
        'familyName': !(0, runtime_1.exists)(json, 'familyName') ? undefined : json['familyName'],
        'role': !(0, runtime_1.exists)(json, 'role') ? undefined : json['role'],
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'status': !(0, runtime_1.exists)(json, 'status') ? undefined : json['status'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateSelfDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'role': value.role,
        'phoneNumber': value.phoneNumber,
        'status': value.status,
    };
}
