// @cp/node-sdk/dist/cjs/models/UpdatePaymentMethodStatusDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdatePaymentMethodStatusDtoStatusEnum = void 0;
exports.instanceOfUpdatePaymentMethodStatusDto = instanceOfUpdatePaymentMethodStatusDto;
exports.UpdatePaymentMethodStatusDtoFromJSON = UpdatePaymentMethodStatusDtoFromJSON;
exports.UpdatePaymentMethodStatusDtoFromJSONTyped = UpdatePaymentMethodStatusDtoFromJSONTyped;
exports.UpdatePaymentMethodStatusDtoToJSON = UpdatePaymentMethodStatusDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.UpdatePaymentMethodStatusDtoStatusEnum = {
    Processing: 'processing',
    ApprovalRequired: 'approvalRequired',
    Succeeded: 'succeeded',
    Failed: 'failed',
    Expired: 'expired'
};
/**
 * Check if a given object implements the UpdatePaymentMethodStatusDto interface.
 */
function instanceOfUpdatePaymentMethodStatusDto(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethodId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "status" in value;
    return isInstance;
}
function UpdatePaymentMethodStatusDtoFromJSON(json) {
    return UpdatePaymentMethodStatusDtoFromJSONTyped(json, false);
}
function UpdatePaymentMethodStatusDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethodId': json['paymentMethodId'],
        'merchantId': json['merchantId'],
        'status': json['status'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdatePaymentMethodStatusDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethodId': value.paymentMethodId,
        'merchantId': value.merchantId,
        'status': value.status,
    };
}
