// @cp/node-sdk/dist/cjs/models/UpdateFeatureTargetDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateFeatureTargetDto = instanceOfUpdateFeatureTargetDto;
exports.UpdateFeatureTargetDtoFromJSON = UpdateFeatureTargetDtoFromJSON;
exports.UpdateFeatureTargetDtoFromJSONTyped = UpdateFeatureTargetDtoFromJSONTyped;
exports.UpdateFeatureTargetDtoToJSON = UpdateFeatureTargetDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UpdateFeatureTargetDto interface.
 */
function instanceOfUpdateFeatureTargetDto(value) {
    let isInstance = true;
    return isInstance;
}
function UpdateFeatureTargetDtoFromJSON(json) {
    return UpdateFeatureTargetDtoFromJSONTyped(json, false);
}
function UpdateFeatureTargetDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'rolloutPercentage': !(0, runtime_1.exists)(json, 'rolloutPercentage') ? undefined : json['rolloutPercentage'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateFeatureTargetDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'rolloutPercentage': value.rolloutPercentage,
    };
}
