// @cp/node-sdk/dist/cjs/models/UpdateWebhookDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateWebhookDto = instanceOfUpdateWebhookDto;
exports.UpdateWebhookDtoFromJSON = UpdateWebhookDtoFromJSON;
exports.UpdateWebhookDtoFromJSONTyped = UpdateWebhookDtoFromJSONTyped;
exports.UpdateWebhookDtoToJSON = UpdateWebhookDtoToJSON;
const runtime_1 = require("../runtime");
const CreateEventDomainDto_1 = require("./CreateEventDomainDto");
/**
 * Check if a given object implements the UpdateWebhookDto interface.
 */
function instanceOfUpdateWebhookDto(value) {
    let isInstance = true;
    isInstance = isInstance && "webhookId" in value;
    return isInstance;
}
function UpdateWebhookDtoFromJSON(json) {
    return UpdateWebhookDtoFromJSONTyped(json, false);
}
function UpdateWebhookDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': !(0, runtime_1.exists)(json, 'merchantId') ? undefined : json['merchantId'],
        'url': !(0, runtime_1.exists)(json, 'url') ? undefined : json['url'],
        'eventDomains': !(0, runtime_1.exists)(json, 'eventDomains') ? undefined : (0, CreateEventDomainDto_1.CreateEventDomainDtoFromJSON)(json['eventDomains']),
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'username': !(0, runtime_1.exists)(json, 'username') ? undefined : json['username'],
        'password': !(0, runtime_1.exists)(json, 'password') ? undefined : json['password'],
        'webhookId': json['webhookId'],
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateWebhookDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'url': value.url,
        'eventDomains': (0, CreateEventDomainDto_1.CreateEventDomainDtoToJSON)(value.eventDomains),
        'name': value.name,
        'description': value.description,
        'username': value.username,
        'password': value.password,
        'webhookId': value.webhookId,
        'enabled': value.enabled,
    };
}
