// @cp/node-sdk/dist/cjs/models/UpdateMerchantDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateMerchantDto = instanceOfUpdateMerchantDto;
exports.UpdateMerchantDtoFromJSON = UpdateMerchantDtoFromJSON;
exports.UpdateMerchantDtoFromJSONTyped = UpdateMerchantDtoFromJSONTyped;
exports.UpdateMerchantDtoToJSON = UpdateMerchantDtoToJSON;
const runtime_1 = require("../runtime");
const CreateMerchantAddressDto_1 = require("./CreateMerchantAddressDto");
/**
 * Check if a given object implements the UpdateMerchantDto interface.
 */
function instanceOfUpdateMerchantDto(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
function UpdateMerchantDtoFromJSON(json) {
    return UpdateMerchantDtoFromJSONTyped(json, false);
}
function UpdateMerchantDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'address': !(0, runtime_1.exists)(json, 'address') ? undefined : (0, CreateMerchantAddressDto_1.CreateMerchantAddressDtoFromJSON)(json['address']),
        'website': !(0, runtime_1.exists)(json, 'website') ? undefined : json['website'],
        'merchantId': json['merchantId'],
        'cancelRedirectUrl': !(0, runtime_1.exists)(json, 'cancelRedirectUrl') ? undefined : json['cancelRedirectUrl'],
        'successRedirectUrl': !(0, runtime_1.exists)(json, 'successRedirectUrl') ? undefined : json['successRedirectUrl'],
        'logo': !(0, runtime_1.exists)(json, 'logo') ? undefined : json['logo'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateMerchantDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'email': value.email,
        'phoneNumber': value.phoneNumber,
        'address': (0, CreateMerchantAddressDto_1.CreateMerchantAddressDtoToJSON)(value.address),
        'website': value.website,
        'merchantId': value.merchantId,
        'cancelRedirectUrl': value.cancelRedirectUrl,
        'successRedirectUrl': value.successRedirectUrl,
        'logo': value.logo,
    };
}
