// @cp/node-sdk/dist/cjs/models/UpdateUserDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateUserDtoStatusEnum = exports.UpdateUserDtoRoleEnum = void 0;
exports.instanceOfUpdateUserDto = instanceOfUpdateUserDto;
exports.UpdateUserDtoFromJSON = UpdateUserDtoFromJSON;
exports.UpdateUserDtoFromJSONTyped = UpdateUserDtoFromJSONTyped;
exports.UpdateUserDtoToJSON = UpdateUserDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.UpdateUserDtoRoleEnum = {
    User: 'user',
    Agent: 'agent',
    Admin: 'admin',
    SuperAdmin: 'superAdmin'
};
/**
 * @export
 */
exports.UpdateUserDtoStatusEnum = {
    Invited: 'invited',
    Active: 'active',
    Inactive: 'inactive',
    Blocked: 'blocked'
};
/**
 * Check if a given object implements the UpdateUserDto interface.
 */
function instanceOfUpdateUserDto(value) {
    let isInstance = true;
    isInstance = isInstance && "userId" in value;
    return isInstance;
}
function UpdateUserDtoFromJSON(json) {
    return UpdateUserDtoFromJSONTyped(json, false);
}
function UpdateUserDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': !(0, runtime_1.exists)(json, 'givenName') ? undefined : json['givenName'],
        'familyName': !(0, runtime_1.exists)(json, 'familyName') ? undefined : json['familyName'],
        'role': !(0, runtime_1.exists)(json, 'role') ? undefined : json['role'],
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'userId': json['userId'],
        'status': !(0, runtime_1.exists)(json, 'status') ? undefined : json['status'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateUserDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'role': value.role,
        'phoneNumber': value.phoneNumber,
        'userId': value.userId,
        'status': value.status,
    };
}
