// @cp/node-sdk/dist/cjs/models/UpdateBuyerDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateBuyerDto = instanceOfUpdateBuyerDto;
exports.UpdateBuyerDtoFromJSON = UpdateBuyerDtoFromJSON;
exports.UpdateBuyerDtoFromJSONTyped = UpdateBuyerDtoFromJSONTyped;
exports.UpdateBuyerDtoToJSON = UpdateBuyerDtoToJSON;
const runtime_1 = require("../runtime");
const CreateBuyerAddressDto_1 = require("./CreateBuyerAddressDto");
/**
 * Check if a given object implements the UpdateBuyerDto interface.
 */
function instanceOfUpdateBuyerDto(value) {
    let isInstance = true;
    isInstance = isInstance && "buyerId" in value;
    return isInstance;
}
function UpdateBuyerDtoFromJSON(json) {
    return UpdateBuyerDtoFromJSONTyped(json, false);
}
function UpdateBuyerDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': !(0, runtime_1.exists)(json, 'givenName') ? undefined : json['givenName'],
        'familyName': !(0, runtime_1.exists)(json, 'familyName') ? undefined : json['familyName'],
        'email': !(0, runtime_1.exists)(json, 'email') ? undefined : json['email'],
        'address': !(0, runtime_1.exists)(json, 'address') ? undefined : (0, CreateBuyerAddressDto_1.CreateBuyerAddressDtoFromJSON)(json['address']),
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'externalBuyerId': !(0, runtime_1.exists)(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'buyerId': json['buyerId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateBuyerDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'address': (0, CreateBuyerAddressDto_1.CreateBuyerAddressDtoToJSON)(value.address),
        'phoneNumber': value.phoneNumber,
        'externalBuyerId': value.externalBuyerId,
        'buyerId': value.buyerId,
    };
}
