// @cp/node-sdk/dist/cjs/models/UpdateFeatureDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateFeatureDto = instanceOfUpdateFeatureDto;
exports.UpdateFeatureDtoFromJSON = UpdateFeatureDtoFromJSON;
exports.UpdateFeatureDtoFromJSONTyped = UpdateFeatureDtoFromJSONTyped;
exports.UpdateFeatureDtoToJSON = UpdateFeatureDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UpdateFeatureDto interface.
 */
function instanceOfUpdateFeatureDto(value) {
    let isInstance = true;
    return isInstance;
}
function UpdateFeatureDtoFromJSON(json) {
    return UpdateFeatureDtoFromJSONTyped(json, false);
}
function UpdateFeatureDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'version': !(0, runtime_1.exists)(json, 'version') ? undefined : json['version'],
        'enabled': !(0, runtime_1.exists)(json, 'enabled') ? undefined : json['enabled'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateFeatureDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'description': value.description,
        'version': value.version,
        'enabled': value.enabled,
    };
}
