// @cp/node-sdk/dist/cjs/models/UpdateTransactionsReportsScheduleDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTransactionsReportsScheduleDtoOrderEnum = exports.UpdateTransactionsReportsScheduleDtoDateRangeEnum = exports.UpdateTransactionsReportsScheduleDtoIntervalEnum = void 0;
exports.instanceOfUpdateTransactionsReportsScheduleDto = instanceOfUpdateTransactionsReportsScheduleDto;
exports.UpdateTransactionsReportsScheduleDtoFromJSON = UpdateTransactionsReportsScheduleDtoFromJSON;
exports.UpdateTransactionsReportsScheduleDtoFromJSONTyped = UpdateTransactionsReportsScheduleDtoFromJSONTyped;
exports.UpdateTransactionsReportsScheduleDtoToJSON = UpdateTransactionsReportsScheduleDtoToJSON;
const runtime_1 = require("../runtime");
const CreateTransactionsReportsFiltersDto_1 = require("./CreateTransactionsReportsFiltersDto");
/**
 * @export
 */
exports.UpdateTransactionsReportsScheduleDtoIntervalEnum = {
    Daily: 'DAILY',
    Weekly: 'WEEKLY',
    Monthly: 'MONTHLY'
};
/**
 * @export
 */
exports.UpdateTransactionsReportsScheduleDtoDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * @export
 */
exports.UpdateTransactionsReportsScheduleDtoOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * Check if a given object implements the UpdateTransactionsReportsScheduleDto interface.
 */
function instanceOfUpdateTransactionsReportsScheduleDto(value) {
    let isInstance = true;
    isInstance = isInstance && "reportScheduleId" in value;
    return isInstance;
}
function UpdateTransactionsReportsScheduleDtoFromJSON(json) {
    return UpdateTransactionsReportsScheduleDtoFromJSONTyped(json, false);
}
function UpdateTransactionsReportsScheduleDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'interval': !(0, runtime_1.exists)(json, 'interval') ? undefined : json['interval'],
        'dateRange': !(0, runtime_1.exists)(json, 'dateRange') ? undefined : json['dateRange'],
        'recipients': !(0, runtime_1.exists)(json, 'recipients') ? undefined : json['recipients'],
        'order': !(0, runtime_1.exists)(json, 'order') ? undefined : json['order'],
        'startTime': !(0, runtime_1.exists)(json, 'startTime') ? undefined : json['startTime'],
        'endTime': !(0, runtime_1.exists)(json, 'endTime') ? undefined : json['endTime'],
        'nextRun': !(0, runtime_1.exists)(json, 'nextRun') ? undefined : json['nextRun'],
        'columns': !(0, runtime_1.exists)(json, 'columns') ? undefined : json['columns'],
        'transactionsFilters': !(0, runtime_1.exists)(json, 'transactionsFilters') ? undefined : (0, CreateTransactionsReportsFiltersDto_1.CreateTransactionsReportsFiltersDtoFromJSON)(json['transactionsFilters']),
        'reportScheduleId': json['reportScheduleId'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateTransactionsReportsScheduleDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'interval': value.interval,
        'dateRange': value.dateRange,
        'recipients': value.recipients,
        'order': value.order,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'nextRun': value.nextRun,
        'columns': value.columns,
        'transactionsFilters': (0, CreateTransactionsReportsFiltersDto_1.CreateTransactionsReportsFiltersDtoToJSON)(value.transactionsFilters),
        'reportScheduleId': value.reportScheduleId,
    };
}
