// @cp/node-sdk/dist/cjs/models/UpdateTransactionsReportsTemplateDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UpdateTransactionsReportsTemplateDtoDateRangeEnum = exports.UpdateTransactionsReportsTemplateDtoOrderEnum = void 0;
exports.instanceOfUpdateTransactionsReportsTemplateDto = instanceOfUpdateTransactionsReportsTemplateDto;
exports.UpdateTransactionsReportsTemplateDtoFromJSON = UpdateTransactionsReportsTemplateDtoFromJSON;
exports.UpdateTransactionsReportsTemplateDtoFromJSONTyped = UpdateTransactionsReportsTemplateDtoFromJSONTyped;
exports.UpdateTransactionsReportsTemplateDtoToJSON = UpdateTransactionsReportsTemplateDtoToJSON;
const runtime_1 = require("../runtime");
const CreateTransactionsReportsFiltersDto_1 = require("./CreateTransactionsReportsFiltersDto");
/**
 * @export
 */
exports.UpdateTransactionsReportsTemplateDtoOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * @export
 */
exports.UpdateTransactionsReportsTemplateDtoDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the UpdateTransactionsReportsTemplateDto interface.
 */
function instanceOfUpdateTransactionsReportsTemplateDto(value) {
    let isInstance = true;
    isInstance = isInstance && "reportTemplateId" in value;
    return isInstance;
}
function UpdateTransactionsReportsTemplateDtoFromJSON(json) {
    return UpdateTransactionsReportsTemplateDtoFromJSONTyped(json, false);
}
function UpdateTransactionsReportsTemplateDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': !(0, runtime_1.exists)(json, 'name') ? undefined : json['name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'order': !(0, runtime_1.exists)(json, 'order') ? undefined : json['order'],
        'dateRange': !(0, runtime_1.exists)(json, 'dateRange') ? undefined : json['dateRange'],
        'startTime': !(0, runtime_1.exists)(json, 'startTime') ? undefined : json['startTime'],
        'endTime': !(0, runtime_1.exists)(json, 'endTime') ? undefined : json['endTime'],
        'columns': !(0, runtime_1.exists)(json, 'columns') ? undefined : json['columns'],
        'transactionsFilters': !(0, runtime_1.exists)(json, 'transactionsFilters') ? undefined : (0, CreateTransactionsReportsFiltersDto_1.CreateTransactionsReportsFiltersDtoFromJSON)(json['transactionsFilters']),
        'reportTemplateId': json['reportTemplateId'],
        'isFavorite': !(0, runtime_1.exists)(json, 'isFavorite') ? undefined : json['isFavorite'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateTransactionsReportsTemplateDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'order': value.order,
        'dateRange': value.dateRange,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'columns': value.columns,
        'transactionsFilters': (0, CreateTransactionsReportsFiltersDto_1.CreateTransactionsReportsFiltersDtoToJSON)(value.transactionsFilters),
        'reportTemplateId': value.reportTemplateId,
        'isFavorite': value.isFavorite,
    };
}
