// @cp/node-sdk/dist/cjs/models/UpdateMerchantCredentialsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUpdateMerchantCredentialsDto = instanceOfUpdateMerchantCredentialsDto;
exports.UpdateMerchantCredentialsDtoFromJSON = UpdateMerchantCredentialsDtoFromJSON;
exports.UpdateMerchantCredentialsDtoFromJSONTyped = UpdateMerchantCredentialsDtoFromJSONTyped;
exports.UpdateMerchantCredentialsDtoToJSON = UpdateMerchantCredentialsDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the UpdateMerchantCredentialsDto interface.
 */
function instanceOfUpdateMerchantCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantCredentialsId" in value;
    isInstance = isInstance && "credentialsName" in value;
    return isInstance;
}
function UpdateMerchantCredentialsDtoFromJSON(json) {
    return UpdateMerchantCredentialsDtoFromJSONTyped(json, false);
}
function UpdateMerchantCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantCredentialsId': json['merchantCredentialsId'],
        'credentialsName': json['credentialsName'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UpdateMerchantCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantCredentialsId': value.merchantCredentialsId,
        'credentialsName': value.credentialsName,
    };
}
