// @cp/node-sdk/dist/cjs/models/UsersSearchEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfUsersSearchEntity = instanceOfUsersSearchEntity;
exports.UsersSearchEntityFromJSON = UsersSearchEntityFromJSON;
exports.UsersSearchEntityFromJSONTyped = UsersSearchEntityFromJSONTyped;
exports.UsersSearchEntityToJSON = UsersSearchEntityToJSON;
const runtime_1 = require("../runtime");
const UserEntity_1 = require("./UserEntity");
/**
 * Check if a given object implements the UsersSearchEntity interface.
 */
function instanceOfUsersSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "items" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
function UsersSearchEntityFromJSON(json) {
    return UsersSearchEntityFromJSONTyped(json, false);
}
function UsersSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'items': (json['items'].map(UserEntity_1.UserEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !(0, runtime_1.exists)(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !(0, runtime_1.exists)(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UsersSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'items': (value.items.map(UserEntity_1.UserEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
