// @cp/node-sdk/dist/cjs/models/UserEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserEntityStatusEnum = exports.UserEntityRoleEnum = void 0;
exports.instanceOfUserEntity = instanceOfUserEntity;
exports.UserEntityFromJSON = UserEntityFromJSON;
exports.UserEntityFromJSONTyped = UserEntityFromJSONTyped;
exports.UserEntityToJSON = UserEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * @export
 */
exports.UserEntityRoleEnum = {
    User: 'user',
    Agent: 'agent',
    Admin: 'admin',
    SuperAdmin: 'superAdmin'
};
/**
 * @export
 */
exports.UserEntityStatusEnum = {
    Invited: 'invited',
    Active: 'active',
    Inactive: 'inactive',
    Blocked: 'blocked'
};
/**
 * Check if a given object implements the UserEntity interface.
 */
function instanceOfUserEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "userId" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "role" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    return isInstance;
}
function UserEntityFromJSON(json) {
    return UserEntityFromJSONTyped(json, false);
}
function UserEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'userId': json['userId'],
        'email': json['email'],
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'merchantId': json['merchantId'],
        'role': json['role'],
        'status': json['status'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'phoneNumber': !(0, runtime_1.exists)(json, 'phoneNumber') ? undefined : json['phoneNumber'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function UserEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'userId': value.userId,
        'email': value.email,
        'givenName': value.givenName,
        'familyName': value.familyName,
        'merchantId': value.merchantId,
        'role': value.role,
        'status': value.status,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'phoneNumber': value.phoneNumber,
    };
}
