// @cp/node-sdk/dist/cjs/models/ValidateMerchantCredentialsDto.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfValidateMerchantCredentialsDto = instanceOfValidateMerchantCredentialsDto;
exports.ValidateMerchantCredentialsDtoFromJSON = ValidateMerchantCredentialsDtoFromJSON;
exports.ValidateMerchantCredentialsDtoFromJSONTyped = ValidateMerchantCredentialsDtoFromJSONTyped;
exports.ValidateMerchantCredentialsDtoToJSON = ValidateMerchantCredentialsDtoToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the ValidateMerchantCredentialsDto interface.
 */
function instanceOfValidateMerchantCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "accessToken" in value;
    return isInstance;
}
function ValidateMerchantCredentialsDtoFromJSON(json) {
    return ValidateMerchantCredentialsDtoFromJSONTyped(json, false);
}
function ValidateMerchantCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'accessToken': json['accessToken'],
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function ValidateMerchantCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'accessToken': value.accessToken,
    };
}
