// @cp/node-sdk/dist/cjs/models/WebhooksEntity.js
"use strict";
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
Object.defineProperty(exports, "__esModule", { value: true });
exports.instanceOfWebhooksEntity = instanceOfWebhooksEntity;
exports.WebhooksEntityFromJSON = WebhooksEntityFromJSON;
exports.WebhooksEntityFromJSONTyped = WebhooksEntityFromJSONTyped;
exports.WebhooksEntityToJSON = WebhooksEntityToJSON;
const runtime_1 = require("../runtime");
/**
 * Check if a given object implements the WebhooksEntity interface.
 */
function instanceOfWebhooksEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "id" in value;
    isInstance = isInstance && "webhookId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "url" in value;
    isInstance = isInstance && "eventDomains" in value;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    return isInstance;
}
function WebhooksEntityFromJSON(json) {
    return WebhooksEntityFromJSONTyped(json, false);
}
function WebhooksEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'id': json['id'],
        'webhookId': json['webhookId'],
        'merchantId': json['merchantId'],
        'url': json['url'],
        'eventDomains': json['eventDomains'],
        'enabled': json['enabled'],
        'name': json['name'],
        'description': !(0, runtime_1.exists)(json, 'description') ? undefined : json['description'],
        'username': !(0, runtime_1.exists)(json, 'username') ? undefined : json['username'],
        'password': !(0, runtime_1.exists)(json, 'password') ? undefined : json['password'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
    };
    return (0, runtime_1.removeNullUndefined)(typed);
}
function WebhooksEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'id': value.id,
        'webhookId': value.webhookId,
        'merchantId': value.merchantId,
        'url': value.url,
        'eventDomains': value.eventDomains,
        'enabled': value.enabled,
        'name': value.name,
        'description': value.description,
        'username': value.username,
        'password': value.password,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
    };
}
