// @cp/node-sdk/dist/mjs/models/AdminMerchantSettingsDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { MerchantThemeDtoFromJSON, MerchantThemeDtoToJSON, } from './MerchantThemeDto';
/**
 * Check if a given object implements the AdminMerchantSettingsDto interface.
 */
export function instanceOfAdminMerchantSettingsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
export function AdminMerchantSettingsDtoFromJSON(json) {
    return AdminMerchantSettingsDtoFromJSONTyped(json, false);
}
export function AdminMerchantSettingsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'checkoutTheme': !exists(json, 'checkoutTheme') ? undefined : MerchantThemeDtoFromJSON(json['checkoutTheme']),
        'isSplitCFeeEnabled': !exists(json, 'isSplitCFeeEnabled') ? undefined : json['isSplitCFeeEnabled'],
        'isDebtRepaymentEnabled': !exists(json, 'isDebtRepaymentEnabled') ? undefined : json['isDebtRepaymentEnabled'],
        'isPartialAuthEnabled': !exists(json, 'isPartialAuthEnabled') ? undefined : json['isPartialAuthEnabled'],
        'merchantId': json['merchantId'],
    };
    return removeNullUndefined(typed);
}
export function AdminMerchantSettingsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'checkoutTheme': MerchantThemeDtoToJSON(value.checkoutTheme),
        'isSplitCFeeEnabled': value.isSplitCFeeEnabled,
        'isDebtRepaymentEnabled': value.isDebtRepaymentEnabled,
        'isPartialAuthEnabled': value.isPartialAuthEnabled,
        'merchantId': value.merchantId,
    };
}
