// @cp/node-sdk/dist/mjs/models/AdminCreateMerchantCredentialsDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the AdminCreateMerchantCredentialsDto interface.
 */
export function instanceOfAdminCreateMerchantCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "credentialsName" in value;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
export function AdminCreateMerchantCredentialsDtoFromJSON(json) {
    return AdminCreateMerchantCredentialsDtoFromJSONTyped(json, false);
}
export function AdminCreateMerchantCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'credentialsName': json['credentialsName'],
        'merchantId': json['merchantId'],
    };
    return removeNullUndefined(typed);
}
export function AdminCreateMerchantCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'credentialsName': value.credentialsName,
        'merchantId': value.merchantId,
    };
}
