// @cp/node-sdk/dist/mjs/models/AdminVoidTransactionDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the AdminVoidTransactionDto interface.
 */
export function instanceOfAdminVoidTransactionDto(value) {
    let isInstance = true;
    isInstance = isInstance && "transactionId" in value;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
export function AdminVoidTransactionDtoFromJSON(json) {
    return AdminVoidTransactionDtoFromJSONTyped(json, false);
}
export function AdminVoidTransactionDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'transactionId': json['transactionId'],
        'merchantId': json['merchantId'],
    };
    return removeNullUndefined(typed);
}
export function AdminVoidTransactionDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transactionId': value.transactionId,
        'merchantId': value.merchantId,
    };
}
