// @cp/node-sdk/dist/mjs/models/AuthorizedPaymentInstrumentsEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the AuthorizedPaymentInstrumentsEntity interface.
 */
export function instanceOfAuthorizedPaymentInstrumentsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "card" in value;
    isInstance = isInstance && "ach" in value;
    return isInstance;
}
export function AuthorizedPaymentInstrumentsEntityFromJSON(json) {
    return AuthorizedPaymentInstrumentsEntityFromJSONTyped(json, false);
}
export function AuthorizedPaymentInstrumentsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'card': json['card'],
        'ach': json['ach'],
    };
    return removeNullUndefined(typed);
}
export function AuthorizedPaymentInstrumentsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'card': value.card,
        'ach': value.ach,
    };
}
