// @cp/node-sdk/dist/mjs/models/AvailablePaymentMethodEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { TransactionFeeDetailsEntityFromJSON, TransactionFeeDetailsEntityToJSON, } from './TransactionFeeDetailsEntity';
/**
 * @export
 */
export const AvailablePaymentMethodEntityMethodEnum = {
    Ach: 'ACH',
    Bank: 'BANK',
    Card: 'CARD'
};
/**
 * Check if a given object implements the AvailablePaymentMethodEntity interface.
 */
export function instanceOfAvailablePaymentMethodEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "method" in value;
    isInstance = isInstance && "adjustedTotal" in value;
    isInstance = isInstance && "feesTotal" in value;
    return isInstance;
}
export function AvailablePaymentMethodEntityFromJSON(json) {
    return AvailablePaymentMethodEntityFromJSONTyped(json, false);
}
export function AvailablePaymentMethodEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'method': json['method'],
        'adjustedTotal': json['adjustedTotal'],
        'feesTotal': json['feesTotal'],
        'fees': !exists(json, 'fees') ? undefined : (json['fees'].map(TransactionFeeDetailsEntityFromJSON)),
    };
    return removeNullUndefined(typed);
}
export function AvailablePaymentMethodEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'method': value.method,
        'adjustedTotal': value.adjustedTotal,
        'feesTotal': value.feesTotal,
        'fees': value.fees === undefined ? undefined : (value.fees.map(TransactionFeeDetailsEntityToJSON)),
    };
}
