// @cp/node-sdk/dist/mjs/models/BankDetailsServiceEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const BankDetailsServiceEntityAccountTypeEnum = {
    Checking: 'CHECKING',
    Savings: 'SAVINGS'
};
/**
 * Check if a given object implements the BankDetailsServiceEntity interface.
 */
export function instanceOfBankDetailsServiceEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "abaNumber" in value;
    isInstance = isInstance && "accountType" in value;
    isInstance = isInstance && "accountNumber" in value;
    isInstance = isInstance && "accountHolderName" in value;
    isInstance = isInstance && "accountEnding" in value;
    return isInstance;
}
export function BankDetailsServiceEntityFromJSON(json) {
    return BankDetailsServiceEntityFromJSONTyped(json, false);
}
export function BankDetailsServiceEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'abaNumber': json['abaNumber'],
        'accountType': json['accountType'],
        'accountNumber': json['accountNumber'],
        'accountHolderName': json['accountHolderName'],
        'accountEnding': json['accountEnding'],
    };
    return removeNullUndefined(typed);
}
export function BankDetailsServiceEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'abaNumber': value.abaNumber,
        'accountType': value.accountType,
        'accountNumber': value.accountNumber,
        'accountHolderName': value.accountHolderName,
        'accountEnding': value.accountEnding,
    };
}
