// @cp/node-sdk/dist/mjs/models/BillingDetailsDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
import { BillingAddressDtoFromJSON, BillingAddressDtoToJSON, } from './BillingAddressDto';
/**
 * Check if a given object implements the BillingDetailsDto interface.
 */
export function instanceOfBillingDetailsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "billingAddress" in value;
    return isInstance;
}
export function BillingDetailsDtoFromJSON(json) {
    return BillingDetailsDtoFromJSONTyped(json, false);
}
export function BillingDetailsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'email': json['email'],
        'billingAddress': BillingAddressDtoFromJSON(json['billingAddress']),
    };
    return removeNullUndefined(typed);
}
export function BillingDetailsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'billingAddress': BillingAddressDtoToJSON(value.billingAddress),
    };
}
