// @cp/node-sdk/dist/mjs/models/BuyersAddressEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the BuyersAddressEntity interface.
 */
export function instanceOfBuyersAddressEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "city" in value;
    isInstance = isInstance && "countryCode" in value;
    isInstance = isInstance && "postalCode" in value;
    isInstance = isInstance && "subdivision" in value;
    isInstance = isInstance && "streetAddress" in value;
    return isInstance;
}
export function BuyersAddressEntityFromJSON(json) {
    return BuyersAddressEntityFromJSONTyped(json, false);
}
export function BuyersAddressEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'city': json['city'],
        'countryCode': json['countryCode'],
        'postalCode': json['postalCode'],
        'subdivision': json['subdivision'],
        'streetAddress': json['streetAddress'],
        'streetAddress2': !exists(json, 'streetAddress2') ? undefined : json['streetAddress2'],
        'subdivisionCode': !exists(json, 'subdivisionCode') ? undefined : json['subdivisionCode'],
    };
    return removeNullUndefined(typed);
}
export function BuyersAddressEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'city': value.city,
        'countryCode': value.countryCode,
        'postalCode': value.postalCode,
        'subdivision': value.subdivision,
        'streetAddress': value.streetAddress,
        'streetAddress2': value.streetAddress2,
        'subdivisionCode': value.subdivisionCode,
    };
}
