// @cp/node-sdk/dist/mjs/models/BuyersSearchEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { BuyersServiceEntityFromJSON, BuyersServiceEntityToJSON, } from './BuyersServiceEntity';
/**
 * Check if a given object implements the BuyersSearchEntity interface.
 */
export function instanceOfBuyersSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "buyers" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
export function BuyersSearchEntityFromJSON(json) {
    return BuyersSearchEntityFromJSONTyped(json, false);
}
export function BuyersSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'buyers': (json['buyers'].map(BuyersServiceEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !exists(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !exists(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return removeNullUndefined(typed);
}
export function BuyersSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'buyers': (value.buyers.map(BuyersServiceEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
