// @cp/node-sdk/dist/mjs/models/BuyersServiceEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { BuyersAddressEntityFromJSON, BuyersAddressEntityToJSON, } from './BuyersAddressEntity';
/**
 * Check if a given object implements the BuyersServiceEntity interface.
 */
export function instanceOfBuyersServiceEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "buyerId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "address" in value;
    return isInstance;
}
export function BuyersServiceEntityFromJSON(json) {
    return BuyersServiceEntityFromJSONTyped(json, false);
}
export function BuyersServiceEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'buyerId': json['buyerId'],
        'merchantId': json['merchantId'],
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'email': json['email'],
        'address': BuyersAddressEntityFromJSON(json['address']),
        'buyerIntegrationId': !exists(json, 'buyerIntegrationId') ? undefined : json['buyerIntegrationId'],
        'phoneNumber': !exists(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'externalBuyerId': !exists(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'createdAt': !exists(json, 'createdAt') ? undefined : (new Date(json['createdAt'])),
        'updatedAt': !exists(json, 'updatedAt') ? undefined : (new Date(json['updatedAt'])),
        'deleted': !exists(json, 'deleted') ? undefined : json['deleted'],
        'deletedAt': !exists(json, 'deletedAt') ? undefined : (new Date(json['deletedAt'])),
    };
    return removeNullUndefined(typed);
}
export function BuyersServiceEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'buyerId': value.buyerId,
        'merchantId': value.merchantId,
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'address': BuyersAddressEntityToJSON(value.address),
        'buyerIntegrationId': value.buyerIntegrationId,
        'phoneNumber': value.phoneNumber,
        'externalBuyerId': value.externalBuyerId,
        'createdAt': value.createdAt === undefined ? undefined : (value.createdAt.toISOString()),
        'updatedAt': value.updatedAt === undefined ? undefined : (value.updatedAt.toISOString()),
        'deleted': value.deleted,
        'deletedAt': value.deletedAt === undefined ? undefined : (value.deletedAt.toISOString()),
    };
}
