// @cp/node-sdk/dist/mjs/models/CaptureTransactionDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the CaptureTransactionDto interface.
 */
export function instanceOfCaptureTransactionDto(value) {
    let isInstance = true;
    isInstance = isInstance && "transactionId" in value;
    isInstance = isInstance && "amount" in value;
    return isInstance;
}
export function CaptureTransactionDtoFromJSON(json) {
    return CaptureTransactionDtoFromJSONTyped(json, false);
}
export function CaptureTransactionDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'transactionId': json['transactionId'],
        'amount': json['amount'],
    };
    return removeNullUndefined(typed);
}
export function CaptureTransactionDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transactionId': value.transactionId,
        'amount': value.amount,
    };
}
