// @cp/node-sdk/dist/mjs/models/ChangeNotificationsSearchEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { ChangeNotificationEntityFromJSON, ChangeNotificationEntityToJSON, } from './ChangeNotificationEntity';
/**
 * Check if a given object implements the ChangeNotificationsSearchEntity interface.
 */
export function instanceOfChangeNotificationsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "changeNotifications" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
export function ChangeNotificationsSearchEntityFromJSON(json) {
    return ChangeNotificationsSearchEntityFromJSONTyped(json, false);
}
export function ChangeNotificationsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'changeNotifications': (json['changeNotifications'].map(ChangeNotificationEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !exists(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !exists(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return removeNullUndefined(typed);
}
export function ChangeNotificationsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'changeNotifications': (value.changeNotifications.map(ChangeNotificationEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
