// @cp/node-sdk/dist/mjs/models/CheckoutSessionBaseEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { AvailablePaymentMethodEntityFromJSON, AvailablePaymentMethodEntityToJSON, } from './AvailablePaymentMethodEntity';
import { CheckoutSessionItemBaseEntityFromJSON, CheckoutSessionItemBaseEntityToJSON, } from './CheckoutSessionItemBaseEntity';
import { TagEntityFromJSON, TagEntityToJSON, } from './TagEntity';
/**
 * @export
 */
export const CheckoutSessionBaseEntityUiModeEnum = {
    Hosted: 'HOSTED',
    Embedded: 'EMBEDDED'
};
/**
 * @export
 */
export const CheckoutSessionBaseEntityStatusEnum = {
    Pending: 'PENDING',
    Processing: 'PROCESSING',
    Expired: 'EXPIRED',
    Completed: 'COMPLETED'
};
/**
 * @export
 */
export const CheckoutSessionBaseEntityTransactionInitiatorEnum = {
    Mit: 'MIT',
    Cit: 'CIT',
    Sit: 'SIT'
};
/**
 * Check if a given object implements the CheckoutSessionBaseEntity interface.
 */
export function instanceOfCheckoutSessionBaseEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "sessionId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "uiMode" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "currency" in value;
    isInstance = isInstance && "total" in value;
    isInstance = isInstance && "subtotal" in value;
    isInstance = isInstance && "feesTotal" in value;
    isInstance = isInstance && "taxTotal" in value;
    isInstance = isInstance && "discountTotal" in value;
    isInstance = isInstance && "shippingTotal" in value;
    isInstance = isInstance && "availablePaymentMethods" in value;
    isInstance = isInstance && "successRedirectUrl" in value;
    isInstance = isInstance && "cancelRedirectUrl" in value;
    isInstance = isInstance && "country" in value;
    isInstance = isInstance && "canExtend" in value;
    return isInstance;
}
export function CheckoutSessionBaseEntityFromJSON(json) {
    return CheckoutSessionBaseEntityFromJSONTyped(json, false);
}
export function CheckoutSessionBaseEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'sessionId': json['sessionId'],
        'merchantId': json['merchantId'],
        'uiMode': json['uiMode'],
        'status': json['status'],
        'currency': json['currency'],
        'total': json['total'],
        'subtotal': json['subtotal'],
        'feesTotal': json['feesTotal'],
        'taxTotal': json['taxTotal'],
        'discountTotal': json['discountTotal'],
        'shippingTotal': json['shippingTotal'],
        'availablePaymentMethods': (json['availablePaymentMethods'].map(AvailablePaymentMethodEntityFromJSON)),
        'items': !exists(json, 'items') ? undefined : (json['items'].map(CheckoutSessionItemBaseEntityFromJSON)),
        'successRedirectUrl': json['successRedirectUrl'],
        'cancelRedirectUrl': json['cancelRedirectUrl'],
        'country': json['country'],
        'expiresAt': !exists(json, 'expiresAt') ? undefined : (new Date(json['expiresAt'])),
        'expiredAt': !exists(json, 'expiredAt') ? undefined : (new Date(json['expiredAt'])),
        'canExtend': json['canExtend'],
        'expiredReason': !exists(json, 'expiredReason') ? undefined : json['expiredReason'],
        'tags': !exists(json, 'tags') ? undefined : (json['tags'].map(TagEntityFromJSON)),
        'canStore': !exists(json, 'canStore') ? undefined : json['canStore'],
        'buyerId': !exists(json, 'buyerId') ? undefined : json['buyerId'],
        'externalBuyerId': !exists(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'transactionInitiator': !exists(json, 'transactionInitiator') ? undefined : json['transactionInitiator'],
    };
    return removeNullUndefined(typed);
}
export function CheckoutSessionBaseEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'sessionId': value.sessionId,
        'merchantId': value.merchantId,
        'uiMode': value.uiMode,
        'status': value.status,
        'currency': value.currency,
        'total': value.total,
        'subtotal': value.subtotal,
        'feesTotal': value.feesTotal,
        'taxTotal': value.taxTotal,
        'discountTotal': value.discountTotal,
        'shippingTotal': value.shippingTotal,
        'availablePaymentMethods': (value.availablePaymentMethods.map(AvailablePaymentMethodEntityToJSON)),
        'items': value.items === undefined ? undefined : (value.items.map(CheckoutSessionItemBaseEntityToJSON)),
        'successRedirectUrl': value.successRedirectUrl,
        'cancelRedirectUrl': value.cancelRedirectUrl,
        'country': value.country,
        'expiresAt': value.expiresAt === undefined ? undefined : (value.expiresAt.toISOString()),
        'expiredAt': value.expiredAt === undefined ? undefined : (value.expiredAt.toISOString()),
        'canExtend': value.canExtend,
        'expiredReason': value.expiredReason,
        'tags': value.tags === undefined ? undefined : (value.tags.map(TagEntityToJSON)),
        'canStore': value.canStore,
        'buyerId': value.buyerId,
        'externalBuyerId': value.externalBuyerId,
        'transactionInitiator': value.transactionInitiator,
    };
}
