// @cp/node-sdk/dist/mjs/models/CheckoutSessionItemEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the CheckoutSessionItemEntity interface.
 */
export function instanceOfCheckoutSessionItemEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "itemId" in value;
    isInstance = isInstance && "sessionId" in value;
    isInstance = isInstance && "description" in value;
    isInstance = isInstance && "quantity" in value;
    isInstance = isInstance && "unitPrice" in value;
    isInstance = isInstance && "total" in value;
    return isInstance;
}
export function CheckoutSessionItemEntityFromJSON(json) {
    return CheckoutSessionItemEntityFromJSONTyped(json, false);
}
export function CheckoutSessionItemEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'itemId': json['itemId'],
        'sessionId': json['sessionId'],
        'description': json['description'],
        'quantity': json['quantity'],
        'unitPrice': json['unitPrice'],
        'total': json['total'],
        'subtotal': !exists(json, 'subtotal') ? undefined : json['subtotal'],
        'discountTotal': !exists(json, 'discountTotal') ? undefined : json['discountTotal'],
        'taxTotal': !exists(json, 'taxTotal') ? undefined : json['taxTotal'],
        'feesTotal': !exists(json, 'feesTotal') ? undefined : json['feesTotal'],
    };
    return removeNullUndefined(typed);
}
export function CheckoutSessionItemEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'itemId': value.itemId,
        'sessionId': value.sessionId,
        'description': value.description,
        'quantity': value.quantity,
        'unitPrice': value.unitPrice,
        'total': value.total,
        'subtotal': value.subtotal,
        'discountTotal': value.discountTotal,
        'taxTotal': value.taxTotal,
        'feesTotal': value.feesTotal,
    };
}
