// @cp/node-sdk/dist/mjs/models/ConnectorServiceTokenEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const ConnectorServiceTokenEntityConnectorEnum = {
    Epay: 'EPAY',
    ChannelPayments: 'CHANNEL_PAYMENTS',
    Tsys: 'TSYS'
};
/**
 * Check if a given object implements the ConnectorServiceTokenEntity interface.
 */
export function instanceOfConnectorServiceTokenEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethodId" in value;
    isInstance = isInstance && "connectorServiceTokenId" in value;
    isInstance = isInstance && "token" in value;
    isInstance = isInstance && "connectorId" in value;
    isInstance = isInstance && "connector" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    return isInstance;
}
export function ConnectorServiceTokenEntityFromJSON(json) {
    return ConnectorServiceTokenEntityFromJSONTyped(json, false);
}
export function ConnectorServiceTokenEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethodId': json['paymentMethodId'],
        'connectorServiceTokenId': json['connectorServiceTokenId'],
        'token': json['token'],
        'connectorId': json['connectorId'],
        'connector': json['connector'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'approvalUrl': !exists(json, 'approvalUrl') ? undefined : json['approvalUrl'],
    };
    return removeNullUndefined(typed);
}
export function ConnectorServiceTokenEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethodId': value.paymentMethodId,
        'connectorServiceTokenId': value.connectorServiceTokenId,
        'token': value.token,
        'connectorId': value.connectorId,
        'connector': value.connector,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'approvalUrl': value.approvalUrl,
    };
}
