// @cp/node-sdk/dist/mjs/models/CreateTransactionRequestDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { PaymentMethodDtoFromJSON, PaymentMethodDtoToJSON, } from './PaymentMethodDto';
import { TagDtoFromJSON, TagDtoToJSON, } from './TagDto';
/**
 * @export
 */
export const CreateTransactionRequestDtoCurrencyEnum = {
    Usd: 'USD',
    Cad: 'CAD',
    Aud: 'AUD'
};
/**
 * @export
 */
export const CreateTransactionRequestDtoIntentEnum = {
    Authorize: 'authorize',
    Capture: 'capture',
    Debit: 'debit'
};
/**
 * @export
 */
export const CreateTransactionRequestDtoSecCodeEnum = {
    Ppd: 'PPD',
    Ccd: 'CCD',
    Web: 'WEB',
    Tel: 'TEL'
};
/**
 * Check if a given object implements the CreateTransactionRequestDto interface.
 */
export function instanceOfCreateTransactionRequestDto(value) {
    let isInstance = true;
    isInstance = isInstance && "amount" in value;
    isInstance = isInstance && "currency" in value;
    isInstance = isInstance && "paymentMethod" in value;
    isInstance = isInstance && "intent" in value;
    return isInstance;
}
export function CreateTransactionRequestDtoFromJSON(json) {
    return CreateTransactionRequestDtoFromJSONTyped(json, false);
}
export function CreateTransactionRequestDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'amount': json['amount'],
        'currency': json['currency'],
        'paymentMethod': PaymentMethodDtoFromJSON(json['paymentMethod']),
        'intent': json['intent'],
        'excludeCFee': !exists(json, 'excludeCFee') ? undefined : json['excludeCFee'],
        'country': !exists(json, 'country') ? undefined : json['country'],
        'externalTransactionId': !exists(json, 'externalTransactionId') ? undefined : json['externalTransactionId'],
        'secCode': !exists(json, 'secCode') ? undefined : json['secCode'],
        'tags': !exists(json, 'tags') ? undefined : (json['tags'].map(TagDtoFromJSON)),
    };
    return removeNullUndefined(typed);
}
export function CreateTransactionRequestDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'amount': value.amount,
        'currency': value.currency,
        'paymentMethod': PaymentMethodDtoToJSON(value.paymentMethod),
        'intent': value.intent,
        'excludeCFee': value.excludeCFee,
        'country': value.country,
        'externalTransactionId': value.externalTransactionId,
        'secCode': value.secCode,
        'tags': value.tags === undefined ? undefined : (value.tags.map(TagDtoToJSON)),
    };
}
