// @cp/node-sdk/dist/mjs/models/CreateMerchantDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateMerchantAccountWithoutMerchantIdDtoFromJSON, CreateMerchantAccountWithoutMerchantIdDtoToJSON, } from './CreateMerchantAccountWithoutMerchantIdDto';
import { CreateMerchantAddressDtoFromJSON, CreateMerchantAddressDtoToJSON, } from './CreateMerchantAddressDto';
import { MerchantSettingsBaseDtoFromJSON, MerchantSettingsBaseDtoToJSON, } from './MerchantSettingsBaseDto';
/**
 * Check if a given object implements the CreateMerchantDto interface.
 */
export function instanceOfCreateMerchantDto(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "phoneNumber" in value;
    isInstance = isInstance && "address" in value;
    return isInstance;
}
export function CreateMerchantDtoFromJSON(json) {
    return CreateMerchantDtoFromJSONTyped(json, false);
}
export function CreateMerchantDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': json['name'],
        'email': json['email'],
        'phoneNumber': json['phoneNumber'],
        'address': CreateMerchantAddressDtoFromJSON(json['address']),
        'website': !exists(json, 'website') ? undefined : json['website'],
        'parentMerchantId': !exists(json, 'parentMerchantId') ? undefined : json['parentMerchantId'],
        'merchantAccounts': !exists(json, 'merchantAccounts') ? undefined : (json['merchantAccounts'].map(CreateMerchantAccountWithoutMerchantIdDtoFromJSON)),
        'merchantSettings': !exists(json, 'merchantSettings') ? undefined : MerchantSettingsBaseDtoFromJSON(json['merchantSettings']),
    };
    return removeNullUndefined(typed);
}
export function CreateMerchantDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'email': value.email,
        'phoneNumber': value.phoneNumber,
        'address': CreateMerchantAddressDtoToJSON(value.address),
        'website': value.website,
        'parentMerchantId': value.parentMerchantId,
        'merchantAccounts': value.merchantAccounts === undefined ? undefined : (value.merchantAccounts.map(CreateMerchantAccountWithoutMerchantIdDtoToJSON)),
        'merchantSettings': MerchantSettingsBaseDtoToJSON(value.merchantSettings),
    };
}
