// @cp/node-sdk/dist/mjs/models/CreatePaymentMethodSessionDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the CreatePaymentMethodSessionDto interface.
 */
export function instanceOfCreatePaymentMethodSessionDto(value) {
    let isInstance = true;
    isInstance = isInstance && "origin" in value;
    return isInstance;
}
export function CreatePaymentMethodSessionDtoFromJSON(json) {
    return CreatePaymentMethodSessionDtoFromJSONTyped(json, false);
}
export function CreatePaymentMethodSessionDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'origin': json['origin'],
        'buyerId': !exists(json, 'buyerId') ? undefined : json['buyerId'],
        'externalBuyerId': !exists(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
    };
    return removeNullUndefined(typed);
}
export function CreatePaymentMethodSessionDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'origin': value.origin,
        'buyerId': value.buyerId,
        'externalBuyerId': value.externalBuyerId,
    };
}
