// @cp/node-sdk/dist/mjs/models/CreateTransactionsReportsTemplateDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateTransactionsReportsFiltersDtoFromJSON, CreateTransactionsReportsFiltersDtoToJSON, } from './CreateTransactionsReportsFiltersDto';
/**
 * @export
 */
export const CreateTransactionsReportsTemplateDtoOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * @export
 */
export const CreateTransactionsReportsTemplateDtoDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the CreateTransactionsReportsTemplateDto interface.
 */
export function instanceOfCreateTransactionsReportsTemplateDto(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "dateRange" in value;
    isInstance = isInstance && "startTime" in value;
    isInstance = isInstance && "endTime" in value;
    isInstance = isInstance && "columns" in value;
    isInstance = isInstance && "transactionsFilters" in value;
    return isInstance;
}
export function CreateTransactionsReportsTemplateDtoFromJSON(json) {
    return CreateTransactionsReportsTemplateDtoFromJSONTyped(json, false);
}
export function CreateTransactionsReportsTemplateDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'order': !exists(json, 'order') ? undefined : json['order'],
        'dateRange': json['dateRange'],
        'startTime': json['startTime'],
        'endTime': json['endTime'],
        'columns': json['columns'],
        'transactionsFilters': CreateTransactionsReportsFiltersDtoFromJSON(json['transactionsFilters']),
    };
    return removeNullUndefined(typed);
}
export function CreateTransactionsReportsTemplateDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'order': value.order,
        'dateRange': value.dateRange,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'columns': value.columns,
        'transactionsFilters': CreateTransactionsReportsFiltersDtoToJSON(value.transactionsFilters),
    };
}
