// @cp/node-sdk/dist/mjs/models/CreateMerchantCredentialsDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the CreateMerchantCredentialsDto interface.
 */
export function instanceOfCreateMerchantCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "credentialsName" in value;
    return isInstance;
}
export function CreateMerchantCredentialsDtoFromJSON(json) {
    return CreateMerchantCredentialsDtoFromJSONTyped(json, false);
}
export function CreateMerchantCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'credentialsName': json['credentialsName'],
    };
    return removeNullUndefined(typed);
}
export function CreateMerchantCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'credentialsName': value.credentialsName,
    };
}
