// @cp/node-sdk/dist/mjs/models/CreateInitialUserDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const CreateInitialUserDtoRoleEnum = {
    User: 'user',
    Agent: 'agent',
    Admin: 'admin',
    SuperAdmin: 'superAdmin'
};
/**
 * Check if a given object implements the CreateInitialUserDto interface.
 */
export function instanceOfCreateInitialUserDto(value) {
    let isInstance = true;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "role" in value;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
export function CreateInitialUserDtoFromJSON(json) {
    return CreateInitialUserDtoFromJSONTyped(json, false);
}
export function CreateInitialUserDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'email': json['email'],
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'role': json['role'],
        'phoneNumber': !exists(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'merchantId': json['merchantId'],
    };
    return removeNullUndefined(typed);
}
export function CreateInitialUserDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'email': value.email,
        'givenName': value.givenName,
        'familyName': value.familyName,
        'role': value.role,
        'phoneNumber': value.phoneNumber,
        'merchantId': value.merchantId,
    };
}
