// @cp/node-sdk/dist/mjs/models/CreateCheckoutSessionDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateCheckoutSessionItemDtoFromJSON, CreateCheckoutSessionItemDtoToJSON, } from './CreateCheckoutSessionItemDto';
import { TagDtoFromJSON, TagDtoToJSON, } from './TagDto';
/**
 * @export
 */
export const CreateCheckoutSessionDtoSecCodeEnum = {
    Ppd: 'PPD',
    Ccd: 'CCD',
    Web: 'WEB',
    Tel: 'TEL'
};
/**
 * @export
 */
export const CreateCheckoutSessionDtoCurrencyEnum = {
    Usd: 'USD',
    Cad: 'CAD',
    Aud: 'AUD'
};
/**
 * Check if a given object implements the CreateCheckoutSessionDto interface.
 */
export function instanceOfCreateCheckoutSessionDto(value) {
    let isInstance = true;
    isInstance = isInstance && "items" in value;
    return isInstance;
}
export function CreateCheckoutSessionDtoFromJSON(json) {
    return CreateCheckoutSessionDtoFromJSONTyped(json, false);
}
export function CreateCheckoutSessionDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'items': (json['items'].map(CreateCheckoutSessionItemDtoFromJSON)),
        'excludeCFee': !exists(json, 'excludeCFee') ? undefined : json['excludeCFee'],
        'successRedirectUrl': !exists(json, 'successRedirectUrl') ? undefined : json['successRedirectUrl'],
        'cancelRedirectUrl': !exists(json, 'cancelRedirectUrl') ? undefined : json['cancelRedirectUrl'],
        'buyerId': !exists(json, 'buyerId') ? undefined : json['buyerId'],
        'secCode': !exists(json, 'secCode') ? undefined : json['secCode'],
        'externalBuyerId': !exists(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'currency': !exists(json, 'currency') ? undefined : json['currency'],
        'country': !exists(json, 'country') ? undefined : json['country'],
        'maxAttempts': !exists(json, 'maxAttempts') ? undefined : json['maxAttempts'],
        'duration': !exists(json, 'duration') ? undefined : json['duration'],
        'externalTransactionId': !exists(json, 'externalTransactionId') ? undefined : json['externalTransactionId'],
        'tags': !exists(json, 'tags') ? undefined : (json['tags'].map(TagDtoFromJSON)),
    };
    return removeNullUndefined(typed);
}
export function CreateCheckoutSessionDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'items': (value.items.map(CreateCheckoutSessionItemDtoToJSON)),
        'excludeCFee': value.excludeCFee,
        'successRedirectUrl': value.successRedirectUrl,
        'cancelRedirectUrl': value.cancelRedirectUrl,
        'buyerId': value.buyerId,
        'secCode': value.secCode,
        'externalBuyerId': value.externalBuyerId,
        'currency': value.currency,
        'country': value.country,
        'maxAttempts': value.maxAttempts,
        'duration': value.duration,
        'externalTransactionId': value.externalTransactionId,
        'tags': value.tags === undefined ? undefined : (value.tags.map(TagDtoToJSON)),
    };
}
