// @cp/node-sdk/dist/mjs/models/CreateEventConfigDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the CreateEventConfigDto interface.
 */
export function instanceOfCreateEventConfigDto(value) {
    let isInstance = true;
    isInstance = isInstance && "selected" in value;
    return isInstance;
}
export function CreateEventConfigDtoFromJSON(json) {
    return CreateEventConfigDtoFromJSONTyped(json, false);
}
export function CreateEventConfigDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'selected': json['selected'],
    };
    return removeNullUndefined(typed);
}
export function CreateEventConfigDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'selected': value.selected,
    };
}
