// @cp/node-sdk/dist/mjs/models/CreateBuyerDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateBuyerAddressDtoFromJSON, CreateBuyerAddressDtoToJSON, } from './CreateBuyerAddressDto';
/**
 * Check if a given object implements the CreateBuyerDto interface.
 */
export function instanceOfCreateBuyerDto(value) {
    let isInstance = true;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "address" in value;
    return isInstance;
}
export function CreateBuyerDtoFromJSON(json) {
    return CreateBuyerDtoFromJSONTyped(json, false);
}
export function CreateBuyerDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'email': json['email'],
        'address': CreateBuyerAddressDtoFromJSON(json['address']),
        'phoneNumber': !exists(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'externalBuyerId': !exists(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
    };
    return removeNullUndefined(typed);
}
export function CreateBuyerDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'address': CreateBuyerAddressDtoToJSON(value.address),
        'phoneNumber': value.phoneNumber,
        'externalBuyerId': value.externalBuyerId,
    };
}
