// @cp/node-sdk/dist/mjs/models/CreateMerchantConnectorWithoutMerchantAccountIdDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateMerchantConnectorCredentialsDtoFromJSON, CreateMerchantConnectorCredentialsDtoToJSON, } from './CreateMerchantConnectorCredentialsDto';
/**
 * @export
 */
export const CreateMerchantConnectorWithoutMerchantAccountIdDtoConnectorEnum = {
    Epay: 'EPAY',
    AuthorizeNet: 'AUTHORIZE_NET',
    ChannelPayments: 'CHANNEL_PAYMENTS',
    Tsys: 'TSYS'
};
/**
 * @export
 */
export const CreateMerchantConnectorWithoutMerchantAccountIdDtoTypeEnum = {
    Gateway: 'GATEWAY',
    Terminal: 'TERMINAL'
};
/**
 * Check if a given object implements the CreateMerchantConnectorWithoutMerchantAccountIdDto interface.
 */
export function instanceOfCreateMerchantConnectorWithoutMerchantAccountIdDto(value) {
    let isInstance = true;
    isInstance = isInstance && "connector" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
export function CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSON(json) {
    return CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSONTyped(json, false);
}
export function CreateMerchantConnectorWithoutMerchantAccountIdDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'connector': json['connector'],
        'type': json['type'],
        'connectorCredentials': !exists(json, 'connectorCredentials') ? undefined : CreateMerchantConnectorCredentialsDtoFromJSON(json['connectorCredentials']),
        'issuedConnectorId': !exists(json, 'issuedConnectorId') ? undefined : json['issuedConnectorId'],
        'name': !exists(json, 'name') ? undefined : json['name'],
    };
    return removeNullUndefined(typed);
}
export function CreateMerchantConnectorWithoutMerchantAccountIdDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'connector': value.connector,
        'type': value.type,
        'connectorCredentials': CreateMerchantConnectorCredentialsDtoToJSON(value.connectorCredentials),
        'issuedConnectorId': value.issuedConnectorId,
        'name': value.name,
    };
}
