// @cp/node-sdk/dist/mjs/models/CreateCFeeDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const CreateCFeeDtoTypeEnum = {
    Percentage: 'PERCENTAGE',
    Fixed: 'FIXED'
};
/**
 * Check if a given object implements the CreateCFeeDto interface.
 */
export function instanceOfCreateCFeeDto(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "rate" in value;
    return isInstance;
}
export function CreateCFeeDtoFromJSON(json) {
    return CreateCFeeDtoFromJSONTyped(json, false);
}
export function CreateCFeeDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'type': json['type'],
        'rate': json['rate'],
    };
    return removeNullUndefined(typed);
}
export function CreateCFeeDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'rate': value.rate,
    };
}
