// @cp/node-sdk/dist/mjs/models/CreateWebhookDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateEventDomainDtoFromJSON, CreateEventDomainDtoToJSON, } from './CreateEventDomainDto';
/**
 * Check if a given object implements the CreateWebhookDto interface.
 */
export function instanceOfCreateWebhookDto(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "url" in value;
    isInstance = isInstance && "eventDomains" in value;
    return isInstance;
}
export function CreateWebhookDtoFromJSON(json) {
    return CreateWebhookDtoFromJSONTyped(json, false);
}
export function CreateWebhookDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': json['merchantId'],
        'url': json['url'],
        'eventDomains': CreateEventDomainDtoFromJSON(json['eventDomains']),
        'name': !exists(json, 'name') ? undefined : json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'username': !exists(json, 'username') ? undefined : json['username'],
        'password': !exists(json, 'password') ? undefined : json['password'],
    };
    return removeNullUndefined(typed);
}
export function CreateWebhookDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'url': value.url,
        'eventDomains': CreateEventDomainDtoToJSON(value.eventDomains),
        'name': value.name,
        'description': value.description,
        'username': value.username,
        'password': value.password,
    };
}
