// @cp/node-sdk/dist/mjs/models/CreateCheckoutSessionItemDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the CreateCheckoutSessionItemDto interface.
 */
export function instanceOfCreateCheckoutSessionItemDto(value) {
    let isInstance = true;
    isInstance = isInstance && "unitPrice" in value;
    isInstance = isInstance && "description" in value;
    return isInstance;
}
export function CreateCheckoutSessionItemDtoFromJSON(json) {
    return CreateCheckoutSessionItemDtoFromJSONTyped(json, false);
}
export function CreateCheckoutSessionItemDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'unitPrice': json['unitPrice'],
        'description': json['description'],
        'quantity': !exists(json, 'quantity') ? undefined : json['quantity'],
        'taxTotal': !exists(json, 'taxTotal') ? undefined : json['taxTotal'],
        'feesTotal': !exists(json, 'feesTotal') ? undefined : json['feesTotal'],
        'discountTotal': !exists(json, 'discountTotal') ? undefined : json['discountTotal'],
    };
    return removeNullUndefined(typed);
}
export function CreateCheckoutSessionItemDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'unitPrice': value.unitPrice,
        'description': value.description,
        'quantity': value.quantity,
        'taxTotal': value.taxTotal,
        'feesTotal': value.feesTotal,
        'discountTotal': value.discountTotal,
    };
}
