// @cp/node-sdk/dist/mjs/models/CreateMerchantConnectorCredentialsDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the CreateMerchantConnectorCredentialsDto interface.
 */
export function instanceOfCreateMerchantConnectorCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "connectorId" in value;
    return isInstance;
}
export function CreateMerchantConnectorCredentialsDtoFromJSON(json) {
    return CreateMerchantConnectorCredentialsDtoFromJSONTyped(json, false);
}
export function CreateMerchantConnectorCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'connectorId': json['connectorId'],
        'username': !exists(json, 'username') ? undefined : json['username'],
        'password': !exists(json, 'password') ? undefined : json['password'],
    };
    return removeNullUndefined(typed);
}
export function CreateMerchantConnectorCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'connectorId': value.connectorId,
        'username': value.username,
        'password': value.password,
    };
}
