// @cp/node-sdk/dist/mjs/models/CreateEventDomainDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
import { CreateEventConfigDtoFromJSON, CreateEventConfigDtoToJSON, } from './CreateEventConfigDto';
/**
 * Check if a given object implements the CreateEventDomainDto interface.
 */
export function instanceOfCreateEventDomainDto(value) {
    let isInstance = true;
    isInstance = isInstance && "transactions" in value;
    isInstance = isInstance && "buyers" in value;
    isInstance = isInstance && "users" in value;
    isInstance = isInstance && "paymentMethods" in value;
    isInstance = isInstance && "merchants" in value;
    return isInstance;
}
export function CreateEventDomainDtoFromJSON(json) {
    return CreateEventDomainDtoFromJSONTyped(json, false);
}
export function CreateEventDomainDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'transactions': CreateEventConfigDtoFromJSON(json['transactions']),
        'buyers': CreateEventConfigDtoFromJSON(json['buyers']),
        'users': CreateEventConfigDtoFromJSON(json['users']),
        'paymentMethods': CreateEventConfigDtoFromJSON(json['paymentMethods']),
        'merchants': CreateEventConfigDtoFromJSON(json['merchants']),
    };
    return removeNullUndefined(typed);
}
export function CreateEventDomainDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'transactions': CreateEventConfigDtoToJSON(value.transactions),
        'buyers': CreateEventConfigDtoToJSON(value.buyers),
        'users': CreateEventConfigDtoToJSON(value.users),
        'paymentMethods': CreateEventConfigDtoToJSON(value.paymentMethods),
        'merchants': CreateEventConfigDtoToJSON(value.merchants),
    };
}
