// @cp/node-sdk/dist/mjs/models/CreateCheckoutSessionTransactionDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
import { BillingAddressDtoFromJSON, BillingAddressDtoToJSON, } from './BillingAddressDto';
import { CreateCheckoutSessionPaymentMethodDtoFromJSON, CreateCheckoutSessionPaymentMethodDtoToJSON, } from './CreateCheckoutSessionPaymentMethodDto';
/**
 * Check if a given object implements the CreateCheckoutSessionTransactionDto interface.
 */
export function instanceOfCreateCheckoutSessionTransactionDto(value) {
    let isInstance = true;
    isInstance = isInstance && "sessionId" in value;
    isInstance = isInstance && "givenName" in value;
    isInstance = isInstance && "familyName" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "billingAddress" in value;
    isInstance = isInstance && "paymentMethod" in value;
    return isInstance;
}
export function CreateCheckoutSessionTransactionDtoFromJSON(json) {
    return CreateCheckoutSessionTransactionDtoFromJSONTyped(json, false);
}
export function CreateCheckoutSessionTransactionDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'sessionId': json['sessionId'],
        'givenName': json['givenName'],
        'familyName': json['familyName'],
        'email': json['email'],
        'billingAddress': BillingAddressDtoFromJSON(json['billingAddress']),
        'paymentMethod': CreateCheckoutSessionPaymentMethodDtoFromJSON(json['paymentMethod']),
    };
    return removeNullUndefined(typed);
}
export function CreateCheckoutSessionTransactionDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'sessionId': value.sessionId,
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'billingAddress': BillingAddressDtoToJSON(value.billingAddress),
        'paymentMethod': CreateCheckoutSessionPaymentMethodDtoToJSON(value.paymentMethod),
    };
}
