// @cp/node-sdk/dist/mjs/models/CreateFeatureTargetDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const CreateFeatureTargetDtoTargetTypeEnum = {
    Merchant: 'merchant',
    User: 'user',
    PaymentMethod: 'payment_method',
    Buyer: 'buyer',
    Webhook: 'webhook',
    Job: 'job'
};
/**
 * Check if a given object implements the CreateFeatureTargetDto interface.
 */
export function instanceOfCreateFeatureTargetDto(value) {
    let isInstance = true;
    isInstance = isInstance && "featureId" in value;
    isInstance = isInstance && "targetEntityId" in value;
    return isInstance;
}
export function CreateFeatureTargetDtoFromJSON(json) {
    return CreateFeatureTargetDtoFromJSONTyped(json, false);
}
export function CreateFeatureTargetDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'featureId': json['featureId'],
        'targetEntityId': json['targetEntityId'],
        'targetType': !exists(json, 'targetType') ? undefined : json['targetType'],
        'rolloutPercentage': !exists(json, 'rolloutPercentage') ? undefined : json['rolloutPercentage'],
    };
    return removeNullUndefined(typed);
}
export function CreateFeatureTargetDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'featureId': value.featureId,
        'targetEntityId': value.targetEntityId,
        'targetType': value.targetType,
        'rolloutPercentage': value.rolloutPercentage,
    };
}
