// @cp/node-sdk/dist/mjs/models/CreateProvisionedCardTokenByMerchantDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const CreateProvisionedCardTokenByMerchantDtoConnectorEnum = {
    Epay: 'EPAY',
    Tsys: 'TSYS'
};
/**
 * Check if a given object implements the CreateProvisionedCardTokenByMerchantDto interface.
 */
export function instanceOfCreateProvisionedCardTokenByMerchantDto(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethodId" in value;
    isInstance = isInstance && "connector" in value;
    isInstance = isInstance && "merchantId" in value;
    return isInstance;
}
export function CreateProvisionedCardTokenByMerchantDtoFromJSON(json) {
    return CreateProvisionedCardTokenByMerchantDtoFromJSONTyped(json, false);
}
export function CreateProvisionedCardTokenByMerchantDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethodId': json['paymentMethodId'],
        'securityCode': !exists(json, 'securityCode') ? undefined : json['securityCode'],
        'connector': json['connector'],
        'redirectUrl': !exists(json, 'redirectUrl') ? undefined : json['redirectUrl'],
        'merchantId': json['merchantId'],
        'token': !exists(json, 'token') ? undefined : json['token'],
    };
    return removeNullUndefined(typed);
}
export function CreateProvisionedCardTokenByMerchantDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethodId': value.paymentMethodId,
        'securityCode': value.securityCode,
        'connector': value.connector,
        'redirectUrl': value.redirectUrl,
        'merchantId': value.merchantId,
        'token': value.token,
    };
}
