// @cp/node-sdk/dist/mjs/models/CreateProvisionedCardTokenDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const CreateProvisionedCardTokenDtoConnectorEnum = {
    Epay: 'EPAY'
};
/**
 * Check if a given object implements the CreateProvisionedCardTokenDto interface.
 */
export function instanceOfCreateProvisionedCardTokenDto(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethodId" in value;
    isInstance = isInstance && "connector" in value;
    return isInstance;
}
export function CreateProvisionedCardTokenDtoFromJSON(json) {
    return CreateProvisionedCardTokenDtoFromJSONTyped(json, false);
}
export function CreateProvisionedCardTokenDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethodId': json['paymentMethodId'],
        'securityCode': !exists(json, 'securityCode') ? undefined : json['securityCode'],
        'connector': json['connector'],
        'redirectUrl': !exists(json, 'redirectUrl') ? undefined : json['redirectUrl'],
    };
    return removeNullUndefined(typed);
}
export function CreateProvisionedCardTokenDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethodId': value.paymentMethodId,
        'securityCode': value.securityCode,
        'connector': value.connector,
        'redirectUrl': value.redirectUrl,
    };
}
