// @cp/node-sdk/dist/mjs/models/CreateFeatureDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the CreateFeatureDto interface.
 */
export function instanceOfCreateFeatureDto(value) {
    let isInstance = true;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "description" in value;
    return isInstance;
}
export function CreateFeatureDtoFromJSON(json) {
    return CreateFeatureDtoFromJSONTyped(json, false);
}
export function CreateFeatureDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': json['name'],
        'description': json['description'],
        'version': !exists(json, 'version') ? undefined : json['version'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
    };
    return removeNullUndefined(typed);
}
export function CreateFeatureDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'version': value.version,
        'enabled': value.enabled,
    };
}
