// @cp/node-sdk/dist/mjs/models/FeatureEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the FeatureEntity interface.
 */
export function instanceOfFeatureEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "featureId" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "description" in value;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "version" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    return isInstance;
}
export function FeatureEntityFromJSON(json) {
    return FeatureEntityFromJSONTyped(json, false);
}
export function FeatureEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'featureId': json['featureId'],
        'name': json['name'],
        'description': json['description'],
        'enabled': json['enabled'],
        'version': json['version'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
    };
    return removeNullUndefined(typed);
}
export function FeatureEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'featureId': value.featureId,
        'name': value.name,
        'description': value.description,
        'enabled': value.enabled,
        'version': value.version,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
    };
}
