// @cp/node-sdk/dist/mjs/models/FeaturesEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { FeatureEntityFromJSON, FeatureEntityToJSON, } from './FeatureEntity';
/**
 * Check if a given object implements the FeaturesEntity interface.
 */
export function instanceOfFeaturesEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "features" in value;
    return isInstance;
}
export function FeaturesEntityFromJSON(json) {
    return FeaturesEntityFromJSONTyped(json, false);
}
export function FeaturesEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'features': (json['features'].map(FeatureEntityFromJSON)),
        'featuresDict': !exists(json, 'featuresDict') ? undefined : json['featuresDict'],
    };
    return removeNullUndefined(typed);
}
export function FeaturesEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'features': (value.features.map(FeatureEntityToJSON)),
        'featuresDict': value.featuresDict,
    };
}
