// @cp/node-sdk/dist/mjs/models/FeeDetailsEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const FeeDetailsEntityTypeEnum = {
    ConvenienceFee: 'CONVENIENCE_FEE'
};
/**
 * @export
 */
export const FeeDetailsEntityChargeTypeEnum = {
    IncludedInAmount: 'INCLUDED_IN_AMOUNT',
    ChargedSeparately: 'CHARGED_SEPARATELY'
};
/**
 * Check if a given object implements the FeeDetailsEntity interface.
 */
export function instanceOfFeeDetailsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    isInstance = isInstance && "chargeType" in value;
    isInstance = isInstance && "amount" in value;
    return isInstance;
}
export function FeeDetailsEntityFromJSON(json) {
    return FeeDetailsEntityFromJSONTyped(json, false);
}
export function FeeDetailsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'type': json['type'],
        'chargeType': json['chargeType'],
        'amount': json['amount'],
    };
    return removeNullUndefined(typed);
}
export function FeeDetailsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'chargeType': value.chargeType,
        'amount': value.amount,
    };
}
