// @cp/node-sdk/dist/mjs/models/LogoUploadEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the LogoUploadEntity interface.
 */
export function instanceOfLogoUploadEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "imageUrl" in value;
    isInstance = isInstance && "uploadUrl" in value;
    return isInstance;
}
export function LogoUploadEntityFromJSON(json) {
    return LogoUploadEntityFromJSONTyped(json, false);
}
export function LogoUploadEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'imageUrl': json['imageUrl'],
        'uploadUrl': json['uploadUrl'],
    };
    return removeNullUndefined(typed);
}
export function LogoUploadEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'imageUrl': value.imageUrl,
        'uploadUrl': value.uploadUrl,
    };
}
