// @cp/node-sdk/dist/mjs/models/MerchantAccountsSearchEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { MerchantAccountServiceEntityFromJSON, MerchantAccountServiceEntityToJSON, } from './MerchantAccountServiceEntity';
/**
 * Check if a given object implements the MerchantAccountsSearchEntity interface.
 */
export function instanceOfMerchantAccountsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantAccounts" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
export function MerchantAccountsSearchEntityFromJSON(json) {
    return MerchantAccountsSearchEntityFromJSONTyped(json, false);
}
export function MerchantAccountsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantAccounts': (json['merchantAccounts'].map(MerchantAccountServiceEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !exists(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !exists(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return removeNullUndefined(typed);
}
export function MerchantAccountsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantAccounts': (value.merchantAccounts.map(MerchantAccountServiceEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
