// @cp/node-sdk/dist/mjs/models/MerchantConnectorCredentialsServiceEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the MerchantConnectorCredentialsServiceEntity interface.
 */
export function instanceOfMerchantConnectorCredentialsServiceEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "credentialsId" in value;
    isInstance = isInstance && "connectorId" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    return isInstance;
}
export function MerchantConnectorCredentialsServiceEntityFromJSON(json) {
    return MerchantConnectorCredentialsServiceEntityFromJSONTyped(json, false);
}
export function MerchantConnectorCredentialsServiceEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'credentialsId': json['credentialsId'],
        'connectorId': json['connectorId'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'username': !exists(json, 'username') ? undefined : json['username'],
        'password': !exists(json, 'password') ? undefined : json['password'],
    };
    return removeNullUndefined(typed);
}
export function MerchantConnectorCredentialsServiceEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'credentialsId': value.credentialsId,
        'connectorId': value.connectorId,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'username': value.username,
        'password': value.password,
    };
}
