// @cp/node-sdk/dist/mjs/models/MerchantThemeLogoEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const MerchantThemeLogoEntityTypeEnum = {
    None: 'NONE',
    Default: 'DEFAULT',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the MerchantThemeLogoEntity interface.
 */
export function instanceOfMerchantThemeLogoEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
export function MerchantThemeLogoEntityFromJSON(json) {
    return MerchantThemeLogoEntityFromJSONTyped(json, false);
}
export function MerchantThemeLogoEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'type': json['type'],
        'customLogo': !exists(json, 'customLogo') ? undefined : json['customLogo'],
    };
    return removeNullUndefined(typed);
}
export function MerchantThemeLogoEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'type': value.type,
        'customLogo': value.customLogo,
    };
}
