// @cp/node-sdk/dist/mjs/models/MerchantThemeDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { MerchantColorsDtoFromJSON, MerchantColorsDtoToJSON, } from './MerchantColorsDto';
import { MerchantThemeLogoDtoFromJSON, MerchantThemeLogoDtoToJSON, } from './MerchantThemeLogoDto';
/**
 * Check if a given object implements the MerchantThemeDto interface.
 */
export function instanceOfMerchantThemeDto(value) {
    let isInstance = true;
    isInstance = isInstance && "enabled" in value;
    isInstance = isInstance && "logo" in value;
    return isInstance;
}
export function MerchantThemeDtoFromJSON(json) {
    return MerchantThemeDtoFromJSONTyped(json, false);
}
export function MerchantThemeDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'enabled': json['enabled'],
        'logo': MerchantThemeLogoDtoFromJSON(json['logo']),
        'colors': !exists(json, 'colors') ? undefined : MerchantColorsDtoFromJSON(json['colors']),
    };
    return removeNullUndefined(typed);
}
export function MerchantThemeDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'enabled': value.enabled,
        'logo': MerchantThemeLogoDtoToJSON(value.logo),
        'colors': MerchantColorsDtoToJSON(value.colors),
    };
}
