// @cp/node-sdk/dist/mjs/models/MerchantFeeDetailsEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
import { PaymentMethodFeeDetailsEntityFromJSON, PaymentMethodFeeDetailsEntityToJSON, } from './PaymentMethodFeeDetailsEntity';
/**
 * Check if a given object implements the MerchantFeeDetailsEntity interface.
 */
export function instanceOfMerchantFeeDetailsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "subtotal" in value;
    isInstance = isInstance && "feeDetails" in value;
    return isInstance;
}
export function MerchantFeeDetailsEntityFromJSON(json) {
    return MerchantFeeDetailsEntityFromJSONTyped(json, false);
}
export function MerchantFeeDetailsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': json['merchantId'],
        'subtotal': json['subtotal'],
        'feeDetails': (json['feeDetails'].map(PaymentMethodFeeDetailsEntityFromJSON)),
    };
    return removeNullUndefined(typed);
}
export function MerchantFeeDetailsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'subtotal': value.subtotal,
        'feeDetails': (value.feeDetails.map(PaymentMethodFeeDetailsEntityToJSON)),
    };
}
