// @cp/node-sdk/dist/mjs/models/MerchantSettingsDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { MerchantThemeDtoFromJSON, MerchantThemeDtoToJSON, } from './MerchantThemeDto';
/**
 * Check if a given object implements the MerchantSettingsDto interface.
 */
export function instanceOfMerchantSettingsDto(value) {
    let isInstance = true;
    return isInstance;
}
export function MerchantSettingsDtoFromJSON(json) {
    return MerchantSettingsDtoFromJSONTyped(json, false);
}
export function MerchantSettingsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'checkoutTheme': !exists(json, 'checkoutTheme') ? undefined : MerchantThemeDtoFromJSON(json['checkoutTheme']),
    };
    return removeNullUndefined(typed);
}
export function MerchantSettingsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'checkoutTheme': MerchantThemeDtoToJSON(value.checkoutTheme),
    };
}
