// @cp/node-sdk/dist/mjs/models/MerchantCredentialsEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the MerchantCredentialsEntity interface.
 */
export function instanceOfMerchantCredentialsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantCredentialsId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "credentialsName" in value;
    isInstance = isInstance && "thumbprint" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    isInstance = isInstance && "algorithm" in value;
    isInstance = isInstance && "createdBy" in value;
    return isInstance;
}
export function MerchantCredentialsEntityFromJSON(json) {
    return MerchantCredentialsEntityFromJSONTyped(json, false);
}
export function MerchantCredentialsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantCredentialsId': json['merchantCredentialsId'],
        'merchantId': json['merchantId'],
        'credentialsName': json['credentialsName'],
        'thumbprint': json['thumbprint'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'algorithm': json['algorithm'],
        'createdBy': json['createdBy'],
        'signingKey': !exists(json, 'signingKey') ? undefined : json['signingKey'],
    };
    return removeNullUndefined(typed);
}
export function MerchantCredentialsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantCredentialsId': value.merchantCredentialsId,
        'merchantId': value.merchantId,
        'credentialsName': value.credentialsName,
        'thumbprint': value.thumbprint,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'algorithm': value.algorithm,
        'createdBy': value.createdBy,
        'signingKey': value.signingKey,
    };
}
