// @cp/node-sdk/dist/mjs/models/MerchantColorsEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the MerchantColorsEntity interface.
 */
export function instanceOfMerchantColorsEntity(value) {
    let isInstance = true;
    return isInstance;
}
export function MerchantColorsEntityFromJSON(json) {
    return MerchantColorsEntityFromJSONTyped(json, false);
}
export function MerchantColorsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'primary': !exists(json, 'primary') ? undefined : json['primary'],
        'secondary': !exists(json, 'secondary') ? undefined : json['secondary'],
        'tertiary': !exists(json, 'tertiary') ? undefined : json['tertiary'],
    };
    return removeNullUndefined(typed);
}
export function MerchantColorsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'primary': value.primary,
        'secondary': value.secondary,
        'tertiary': value.tertiary,
    };
}
