// @cp/node-sdk/dist/mjs/models/MerchantServiceEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { MerchantAccountServiceEntityFromJSON, MerchantAccountServiceEntityToJSON, } from './MerchantAccountServiceEntity';
import { MerchantAddressEntityFromJSON, MerchantAddressEntityToJSON, } from './MerchantAddressEntity';
import { MerchantSettingsEntityFromJSON, MerchantSettingsEntityToJSON, } from './MerchantSettingsEntity';
/**
 * Check if a given object implements the MerchantServiceEntity interface.
 */
export function instanceOfMerchantServiceEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "address" in value;
    isInstance = isInstance && "email" in value;
    isInstance = isInstance && "phoneNumber" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    isInstance = isInstance && "integrationId" in value;
    return isInstance;
}
export function MerchantServiceEntityFromJSON(json) {
    return MerchantServiceEntityFromJSONTyped(json, false);
}
export function MerchantServiceEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': json['merchantId'],
        'name': json['name'],
        'address': MerchantAddressEntityFromJSON(json['address']),
        'email': json['email'],
        'phoneNumber': json['phoneNumber'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'integrationId': json['integrationId'],
        'logo': !exists(json, 'logo') ? undefined : json['logo'],
        'parentMerchantId': !exists(json, 'parentMerchantId') ? undefined : json['parentMerchantId'],
        'website': !exists(json, 'website') ? undefined : json['website'],
        'merchantAccounts': !exists(json, 'merchantAccounts') ? undefined : (json['merchantAccounts'].map(MerchantAccountServiceEntityFromJSON)),
        'cancelRedirectUrl': !exists(json, 'cancelRedirectUrl') ? undefined : json['cancelRedirectUrl'],
        'successRedirectUrl': !exists(json, 'successRedirectUrl') ? undefined : json['successRedirectUrl'],
        'merchantSettings': !exists(json, 'merchantSettings') ? undefined : MerchantSettingsEntityFromJSON(json['merchantSettings']),
    };
    return removeNullUndefined(typed);
}
export function MerchantServiceEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'name': value.name,
        'address': MerchantAddressEntityToJSON(value.address),
        'email': value.email,
        'phoneNumber': value.phoneNumber,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'integrationId': value.integrationId,
        'logo': value.logo,
        'parentMerchantId': value.parentMerchantId,
        'website': value.website,
        'merchantAccounts': value.merchantAccounts === undefined ? undefined : (value.merchantAccounts.map(MerchantAccountServiceEntityToJSON)),
        'cancelRedirectUrl': value.cancelRedirectUrl,
        'successRedirectUrl': value.successRedirectUrl,
        'merchantSettings': MerchantSettingsEntityToJSON(value.merchantSettings),
    };
}
