// @cp/node-sdk/dist/mjs/models/PaymentMethodDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const PaymentMethodDtoMethodEnum = {
    Ach: 'ACH',
    Bank: 'BANK',
    Card: 'CARD'
};
/**
 * @export
 */
export const PaymentMethodDtoAccountTypeEnum = {
    Checking: 'CHECKING',
    Savings: 'SAVINGS'
};
/**
 * Check if a given object implements the PaymentMethodDto interface.
 */
export function instanceOfPaymentMethodDto(value) {
    let isInstance = true;
    isInstance = isInstance && "method" in value;
    return isInstance;
}
export function PaymentMethodDtoFromJSON(json) {
    return PaymentMethodDtoFromJSONTyped(json, false);
}
export function PaymentMethodDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'method': json['method'],
        'number': !exists(json, 'number') ? undefined : json['number'],
        'securityCode': !exists(json, 'securityCode') ? undefined : json['securityCode'],
        'expirationDate': !exists(json, 'expirationDate') ? undefined : json['expirationDate'],
        'externalBuyerId': !exists(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'buyerId': !exists(json, 'buyerId') ? undefined : json['buyerId'],
        'paymentMethodId': !exists(json, 'paymentMethodId') ? undefined : json['paymentMethodId'],
        'store': !exists(json, 'store') ? undefined : json['store'],
        'abaNumber': !exists(json, 'abaNumber') ? undefined : json['abaNumber'],
        'accountNumber': !exists(json, 'accountNumber') ? undefined : json['accountNumber'],
        'accountType': !exists(json, 'accountType') ? undefined : json['accountType'],
        'accountHolderName': !exists(json, 'accountHolderName') ? undefined : json['accountHolderName'],
    };
    return removeNullUndefined(typed);
}
export function PaymentMethodDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'method': value.method,
        'number': value.number,
        'securityCode': value.securityCode,
        'expirationDate': value.expirationDate,
        'externalBuyerId': value.externalBuyerId,
        'buyerId': value.buyerId,
        'paymentMethodId': value.paymentMethodId,
        'store': value.store,
        'abaNumber': value.abaNumber,
        'accountNumber': value.accountNumber,
        'accountType': value.accountType,
        'accountHolderName': value.accountHolderName,
    };
}
