// @cp/node-sdk/dist/mjs/models/PaymentMethodFeeDetailsEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
import { FeeDetailsEntityFromJSON, FeeDetailsEntityToJSON, } from './FeeDetailsEntity';
/**
 * @export
 */
export const PaymentMethodFeeDetailsEntityMethodEnum = {
    Ach: 'ACH',
    Bank: 'BANK',
    Card: 'CARD'
};
/**
 * Check if a given object implements the PaymentMethodFeeDetailsEntity interface.
 */
export function instanceOfPaymentMethodFeeDetailsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "method" in value;
    isInstance = isInstance && "fees" in value;
    isInstance = isInstance && "feesTotal" in value;
    isInstance = isInstance && "adjustedTotal" in value;
    return isInstance;
}
export function PaymentMethodFeeDetailsEntityFromJSON(json) {
    return PaymentMethodFeeDetailsEntityFromJSONTyped(json, false);
}
export function PaymentMethodFeeDetailsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'method': json['method'],
        'fees': (json['fees'].map(FeeDetailsEntityFromJSON)),
        'feesTotal': json['feesTotal'],
        'adjustedTotal': json['adjustedTotal'],
    };
    return removeNullUndefined(typed);
}
export function PaymentMethodFeeDetailsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'method': value.method,
        'fees': (value.fees.map(FeeDetailsEntityToJSON)),
        'feesTotal': value.feesTotal,
        'adjustedTotal': value.adjustedTotal,
    };
}
