// @cp/node-sdk/dist/mjs/models/PaymentMethodsSearchEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { PaymentMethodEntityFromJSON, PaymentMethodEntityToJSON, } from './PaymentMethodEntity';
/**
 * Check if a given object implements the PaymentMethodsSearchEntity interface.
 */
export function instanceOfPaymentMethodsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethods" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
export function PaymentMethodsSearchEntityFromJSON(json) {
    return PaymentMethodsSearchEntityFromJSONTyped(json, false);
}
export function PaymentMethodsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethods': (json['paymentMethods'].map(PaymentMethodEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !exists(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !exists(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return removeNullUndefined(typed);
}
export function PaymentMethodsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethods': (value.paymentMethods.map(PaymentMethodEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
