// @cp/node-sdk/dist/mjs/models/PaymentMethodSessionBaseEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { AuthorizedPaymentInstrumentsEntityFromJSON, AuthorizedPaymentInstrumentsEntityToJSON, } from './AuthorizedPaymentInstrumentsEntity';
import { PaymentMethodSessionAttemptEntityFromJSON, PaymentMethodSessionAttemptEntityToJSON, } from './PaymentMethodSessionAttemptEntity';
/**
 * @export
 */
export const PaymentMethodSessionBaseEntityStatusEnum = {
    Pending: 'PENDING',
    Expired: 'EXPIRED',
    Completed: 'COMPLETED'
};
/**
 * @export
 */
export const PaymentMethodSessionBaseEntityExpiredReasonEnum = {
    Timeout: 'TIMEOUT',
    MaxAttempts: 'MAX_ATTEMPTS',
    UserCancelled: 'USER_CANCELLED'
};
/**
 * Check if a given object implements the PaymentMethodSessionBaseEntity interface.
 */
export function instanceOfPaymentMethodSessionBaseEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "sessionId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "attempts" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "origin" in value;
    isInstance = isInstance && "expiresAt" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    isInstance = isInstance && "deleted" in value;
    return isInstance;
}
export function PaymentMethodSessionBaseEntityFromJSON(json) {
    return PaymentMethodSessionBaseEntityFromJSONTyped(json, false);
}
export function PaymentMethodSessionBaseEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'sessionId': json['sessionId'],
        'merchantId': json['merchantId'],
        'attempts': (json['attempts'].map(PaymentMethodSessionAttemptEntityFromJSON)),
        'status': json['status'],
        'origin': json['origin'],
        'expiresAt': (new Date(json['expiresAt'])),
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'deleted': json['deleted'],
        'sessionToken': !exists(json, 'sessionToken') ? undefined : json['sessionToken'],
        'authorizedPaymentInstruments': !exists(json, 'authorizedPaymentInstruments') ? undefined : AuthorizedPaymentInstrumentsEntityFromJSON(json['authorizedPaymentInstruments']),
        'paymentMethodId': !exists(json, 'paymentMethodId') ? undefined : json['paymentMethodId'],
        'buyerId': !exists(json, 'buyerId') ? undefined : json['buyerId'],
        'externalBuyerId': !exists(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'expiredReason': !exists(json, 'expiredReason') ? undefined : json['expiredReason'],
        'expiredAt': !exists(json, 'expiredAt') ? undefined : (new Date(json['expiredAt'])),
        'completedAt': !exists(json, 'completedAt') ? undefined : (new Date(json['completedAt'])),
        'deletedAt': !exists(json, 'deletedAt') ? undefined : (new Date(json['deletedAt'])),
    };
    return removeNullUndefined(typed);
}
export function PaymentMethodSessionBaseEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'sessionId': value.sessionId,
        'merchantId': value.merchantId,
        'attempts': (value.attempts.map(PaymentMethodSessionAttemptEntityToJSON)),
        'status': value.status,
        'origin': value.origin,
        'expiresAt': (value.expiresAt.toISOString()),
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'deleted': value.deleted,
        'sessionToken': value.sessionToken,
        'authorizedPaymentInstruments': AuthorizedPaymentInstrumentsEntityToJSON(value.authorizedPaymentInstruments),
        'paymentMethodId': value.paymentMethodId,
        'buyerId': value.buyerId,
        'externalBuyerId': value.externalBuyerId,
        'expiredReason': value.expiredReason,
        'expiredAt': value.expiredAt === undefined ? undefined : (value.expiredAt.toISOString()),
        'completedAt': value.completedAt === undefined ? undefined : (value.completedAt.toISOString()),
        'deletedAt': value.deletedAt === undefined ? undefined : (value.deletedAt.toISOString()),
    };
}
