// @cp/node-sdk/dist/mjs/models/ReportCreatorEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the ReportCreatorEntity interface.
 */
export function instanceOfReportCreatorEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "userId" in value;
    isInstance = isInstance && "fullName" in value;
    return isInstance;
}
export function ReportCreatorEntityFromJSON(json) {
    return ReportCreatorEntityFromJSONTyped(json, false);
}
export function ReportCreatorEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'userId': json['userId'],
        'fullName': json['fullName'],
    };
    return removeNullUndefined(typed);
}
export function ReportCreatorEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'userId': value.userId,
        'fullName': value.fullName,
    };
}
