// @cp/node-sdk/dist/mjs/models/ReturnNotificationsSearchEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { ReturnNotificationEntityFromJSON, ReturnNotificationEntityToJSON, } from './ReturnNotificationEntity';
/**
 * Check if a given object implements the ReturnNotificationsSearchEntity interface.
 */
export function instanceOfReturnNotificationsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "returnNotifications" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    return isInstance;
}
export function ReturnNotificationsSearchEntityFromJSON(json) {
    return ReturnNotificationsSearchEntityFromJSONTyped(json, false);
}
export function ReturnNotificationsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'returnNotifications': (json['returnNotifications'].map(ReturnNotificationEntityFromJSON)),
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !exists(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !exists(json, 'prevCursor') ? undefined : json['prevCursor'],
    };
    return removeNullUndefined(typed);
}
export function ReturnNotificationsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'returnNotifications': (value.returnNotifications.map(ReturnNotificationEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
    };
}
