// @cp/node-sdk/dist/mjs/models/TagEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the TagEntity interface.
 */
export function instanceOfTagEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "key" in value;
    isInstance = isInstance && "value" in value;
    return isInstance;
}
export function TagEntityFromJSON(json) {
    return TagEntityFromJSONTyped(json, false);
}
export function TagEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'key': json['key'],
        'value': json['value'],
    };
    return removeNullUndefined(typed);
}
export function TagEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'key': value.key,
        'value': value.value,
    };
}
