// @cp/node-sdk/dist/mjs/models/TransactionNoteHistoryEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the TransactionNoteHistoryEntity interface.
 */
export function instanceOfTransactionNoteHistoryEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "content" in value;
    isInstance = isInstance && "modifiedBy" in value;
    isInstance = isInstance && "modifiedAt" in value;
    return isInstance;
}
export function TransactionNoteHistoryEntityFromJSON(json) {
    return TransactionNoteHistoryEntityFromJSONTyped(json, false);
}
export function TransactionNoteHistoryEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'content': json['content'],
        'modifiedBy': json['modifiedBy'],
        'modifiedAt': (new Date(json['modifiedAt'])),
    };
    return removeNullUndefined(typed);
}
export function TransactionNoteHistoryEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'content': value.content,
        'modifiedBy': value.modifiedBy,
        'modifiedAt': (value.modifiedAt.toISOString()),
    };
}
