// @cp/node-sdk/dist/mjs/models/TransactionFeeDetailsEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const TransactionFeeDetailsEntityCurrencyEnum = {
    Usd: 'USD',
    Cad: 'CAD',
    Aud: 'AUD'
};
/**
 * @export
 */
export const TransactionFeeDetailsEntityTypeEnum = {
    ConvenienceFee: 'CONVENIENCE_FEE'
};
/**
 * @export
 */
export const TransactionFeeDetailsEntityChargeTypeEnum = {
    IncludedInAmount: 'INCLUDED_IN_AMOUNT',
    ChargedSeparately: 'CHARGED_SEPARATELY'
};
/**
 * @export
 */
export const TransactionFeeDetailsEntityStatusEnum = {
    Processing: 'PROCESSING',
    Processed: 'PROCESSED'
};
/**
 * Check if a given object implements the TransactionFeeDetailsEntity interface.
 */
export function instanceOfTransactionFeeDetailsEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "amount" in value;
    isInstance = isInstance && "type" in value;
    return isInstance;
}
export function TransactionFeeDetailsEntityFromJSON(json) {
    return TransactionFeeDetailsEntityFromJSONTyped(json, false);
}
export function TransactionFeeDetailsEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'amount': json['amount'],
        'currency': !exists(json, 'currency') ? undefined : json['currency'],
        'type': json['type'],
        'chargeType': !exists(json, 'chargeType') ? undefined : json['chargeType'],
        'status': !exists(json, 'status') ? undefined : json['status'],
        'transactionId': !exists(json, 'transactionId') ? undefined : json['transactionId'],
        'description': !exists(json, 'description') ? undefined : json['description'],
    };
    return removeNullUndefined(typed);
}
export function TransactionFeeDetailsEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'amount': value.amount,
        'currency': value.currency,
        'type': value.type,
        'chargeType': value.chargeType,
        'status': value.status,
        'transactionId': value.transactionId,
        'description': value.description,
    };
}
