// @cp/node-sdk/dist/mjs/models/TransactionsReportTemplateEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { ReportCreatorEntityFromJSON, ReportCreatorEntityToJSON, } from './ReportCreatorEntity';
import { TransactionsReportsFiltersEntityFromJSON, TransactionsReportsFiltersEntityToJSON, } from './TransactionsReportsFiltersEntity';
/**
 * @export
 */
export const TransactionsReportTemplateEntityReportDomainEnum = {
    Transactions: 'TRANSACTIONS'
};
/**
 * @export
 */
export const TransactionsReportTemplateEntityOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * @export
 */
export const TransactionsReportTemplateEntityDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the TransactionsReportTemplateEntity interface.
 */
export function instanceOfTransactionsReportTemplateEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "reportTemplateId" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "reportDomain" in value;
    isInstance = isInstance && "isFavorite" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    isInstance = isInstance && "dateRange" in value;
    isInstance = isInstance && "startTime" in value;
    isInstance = isInstance && "endTime" in value;
    isInstance = isInstance && "columns" in value;
    isInstance = isInstance && "transactionsFilters" in value;
    return isInstance;
}
export function TransactionsReportTemplateEntityFromJSON(json) {
    return TransactionsReportTemplateEntityFromJSONTyped(json, false);
}
export function TransactionsReportTemplateEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'reportTemplateId': json['reportTemplateId'],
        'merchantId': !exists(json, 'merchantId') ? undefined : json['merchantId'],
        'createdBy': ReportCreatorEntityFromJSON(json['createdBy']),
        'reportDomain': json['reportDomain'],
        'isFavorite': json['isFavorite'],
        'name': json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'order': !exists(json, 'order') ? undefined : json['order'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'deleted': !exists(json, 'deleted') ? undefined : json['deleted'],
        'deletedAt': !exists(json, 'deletedAt') ? undefined : (new Date(json['deletedAt'])),
        'dateRange': json['dateRange'],
        'startTime': json['startTime'],
        'endTime': json['endTime'],
        'columns': json['columns'],
        'transactionsFilters': TransactionsReportsFiltersEntityFromJSON(json['transactionsFilters']),
    };
    return removeNullUndefined(typed);
}
export function TransactionsReportTemplateEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'reportTemplateId': value.reportTemplateId,
        'merchantId': value.merchantId,
        'createdBy': ReportCreatorEntityToJSON(value.createdBy),
        'reportDomain': value.reportDomain,
        'isFavorite': value.isFavorite,
        'name': value.name,
        'description': value.description,
        'order': value.order,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'deleted': value.deleted,
        'deletedAt': value.deletedAt === undefined ? undefined : (value.deletedAt.toISOString()),
        'dateRange': value.dateRange,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'columns': value.columns,
        'transactionsFilters': TransactionsReportsFiltersEntityToJSON(value.transactionsFilters),
    };
}
