// @cp/node-sdk/dist/mjs/models/TransactionsReportScheduleEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { ReportCreatorEntityFromJSON, ReportCreatorEntityToJSON, } from './ReportCreatorEntity';
import { TransactionsReportsFiltersEntityFromJSON, TransactionsReportsFiltersEntityToJSON, } from './TransactionsReportsFiltersEntity';
/**
 * @export
 */
export const TransactionsReportScheduleEntityOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * @export
 */
export const TransactionsReportScheduleEntityIntervalEnum = {
    Daily: 'DAILY',
    Weekly: 'WEEKLY',
    Monthly: 'MONTHLY'
};
/**
 * @export
 */
export const TransactionsReportScheduleEntityDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the TransactionsReportScheduleEntity interface.
 */
export function instanceOfTransactionsReportScheduleEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "reportScheduleId" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "interval" in value;
    isInstance = isInstance && "dateRange" in value;
    isInstance = isInstance && "startTime" in value;
    isInstance = isInstance && "endTime" in value;
    isInstance = isInstance && "recipients" in value;
    isInstance = isInstance && "nextRun" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    isInstance = isInstance && "columns" in value;
    isInstance = isInstance && "transactionsFilters" in value;
    return isInstance;
}
export function TransactionsReportScheduleEntityFromJSON(json) {
    return TransactionsReportScheduleEntityFromJSONTyped(json, false);
}
export function TransactionsReportScheduleEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'reportScheduleId': json['reportScheduleId'],
        'name': json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'order': !exists(json, 'order') ? undefined : json['order'],
        'createdBy': ReportCreatorEntityFromJSON(json['createdBy']),
        'deletedBy': !exists(json, 'deletedBy') ? undefined : ReportCreatorEntityFromJSON(json['deletedBy']),
        'merchantId': json['merchantId'],
        'interval': json['interval'],
        'dateRange': json['dateRange'],
        'startTime': json['startTime'],
        'endTime': json['endTime'],
        'recipients': json['recipients'],
        'lastRun': !exists(json, 'lastRun') ? undefined : (new Date(json['lastRun'])),
        'nextRun': (new Date(json['nextRun'])),
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'deleted': !exists(json, 'deleted') ? undefined : json['deleted'],
        'deletedAt': !exists(json, 'deletedAt') ? undefined : (new Date(json['deletedAt'])),
        'columns': json['columns'],
        'transactionsFilters': TransactionsReportsFiltersEntityFromJSON(json['transactionsFilters']),
    };
    return removeNullUndefined(typed);
}
export function TransactionsReportScheduleEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'reportScheduleId': value.reportScheduleId,
        'name': value.name,
        'description': value.description,
        'order': value.order,
        'createdBy': ReportCreatorEntityToJSON(value.createdBy),
        'deletedBy': ReportCreatorEntityToJSON(value.deletedBy),
        'merchantId': value.merchantId,
        'interval': value.interval,
        'dateRange': value.dateRange,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'recipients': value.recipients,
        'lastRun': value.lastRun === undefined ? undefined : (value.lastRun.toISOString()),
        'nextRun': (value.nextRun.toISOString()),
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'deleted': value.deleted,
        'deletedAt': value.deletedAt === undefined ? undefined : (value.deletedAt.toISOString()),
        'columns': value.columns,
        'transactionsFilters': TransactionsReportsFiltersEntityToJSON(value.transactionsFilters),
    };
}
