// @cp/node-sdk/dist/mjs/models/TransactionRefundsSearchEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { TransactionRefundBaseEntityFromJSON, TransactionRefundBaseEntityToJSON, } from './TransactionRefundBaseEntity';
/**
 * Check if a given object implements the TransactionRefundsSearchEntity interface.
 */
export function instanceOfTransactionRefundsSearchEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "refunds" in value;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "prevCursor" in value;
    return isInstance;
}
export function TransactionRefundsSearchEntityFromJSON(json) {
    return TransactionRefundsSearchEntityFromJSONTyped(json, false);
}
export function TransactionRefundsSearchEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'refunds': (json['refunds'].map(TransactionRefundBaseEntityFromJSON)),
        'limit': json['limit'],
        'page': !exists(json, 'page') ? undefined : json['page'],
        'pageCount': !exists(json, 'pageCount') ? undefined : json['pageCount'],
        'totalCount': !exists(json, 'totalCount') ? undefined : json['totalCount'],
        'filteredCount': !exists(json, 'filteredCount') ? undefined : json['filteredCount'],
        'prevCursor': json['prevCursor'],
        'nextCursor': !exists(json, 'nextCursor') ? undefined : json['nextCursor'],
    };
    return removeNullUndefined(typed);
}
export function TransactionRefundsSearchEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'refunds': (value.refunds.map(TransactionRefundBaseEntityToJSON)),
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'prevCursor': value.prevCursor,
        'nextCursor': value.nextCursor,
    };
}
