// @cp/node-sdk/dist/mjs/models/TransactionsReportEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { ReportCreatorEntityFromJSON, ReportCreatorEntityToJSON, } from './ReportCreatorEntity';
import { TransactionsReportsFiltersEntityFromJSON, TransactionsReportsFiltersEntityToJSON, } from './TransactionsReportsFiltersEntity';
/**
 * @export
 */
export const TransactionsReportEntityStatusEnum = {
    Pending: 'PENDING',
    Processing: 'PROCESSING',
    Generated: 'GENERATED',
    Failed: 'FAILED'
};
/**
 * @export
 */
export const TransactionsReportEntityOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * @export
 */
export const TransactionsReportEntityDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the TransactionsReportEntity interface.
 */
export function instanceOfTransactionsReportEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "reportId" in value;
    isInstance = isInstance && "name" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "createdBy" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    isInstance = isInstance && "deletedAt" in value;
    isInstance = isInstance && "startDate" in value;
    isInstance = isInstance && "endDate" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "order" in value;
    isInstance = isInstance && "recipients" in value;
    isInstance = isInstance && "dateRange" in value;
    isInstance = isInstance && "startTime" in value;
    isInstance = isInstance && "endTime" in value;
    isInstance = isInstance && "columns" in value;
    isInstance = isInstance && "transactionsFilters" in value;
    return isInstance;
}
export function TransactionsReportEntityFromJSON(json) {
    return TransactionsReportEntityFromJSONTyped(json, false);
}
export function TransactionsReportEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'reportId': json['reportId'],
        'name': json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'merchantId': json['merchantId'],
        'createdBy': ReportCreatorEntityFromJSON(json['createdBy']),
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'deletedAt': (new Date(json['deletedAt'])),
        'deleted': !exists(json, 'deleted') ? undefined : json['deleted'],
        'startDate': json['startDate'],
        'endDate': json['endDate'],
        'status': json['status'],
        'order': json['order'],
        'recipients': json['recipients'],
        'dateRange': json['dateRange'],
        'startTime': json['startTime'],
        'endTime': json['endTime'],
        'columns': json['columns'],
        'transactionsFilters': TransactionsReportsFiltersEntityFromJSON(json['transactionsFilters']),
    };
    return removeNullUndefined(typed);
}
export function TransactionsReportEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'reportId': value.reportId,
        'name': value.name,
        'description': value.description,
        'merchantId': value.merchantId,
        'createdBy': ReportCreatorEntityToJSON(value.createdBy),
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'deletedAt': (value.deletedAt.toISOString()),
        'deleted': value.deleted,
        'startDate': value.startDate,
        'endDate': value.endDate,
        'status': value.status,
        'order': value.order,
        'recipients': value.recipients,
        'dateRange': value.dateRange,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'columns': value.columns,
        'transactionsFilters': TransactionsReportsFiltersEntityToJSON(value.transactionsFilters),
    };
}
