// @cp/node-sdk/dist/mjs/models/TransactionsReportsSearchQueryResponseEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { TransactionsReportEntityFromJSON, TransactionsReportEntityToJSON, } from './TransactionsReportEntity';
/**
 * Check if a given object implements the TransactionsReportsSearchQueryResponseEntity interface.
 */
export function instanceOfTransactionsReportsSearchQueryResponseEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "limit" in value;
    isInstance = isInstance && "page" in value;
    isInstance = isInstance && "pageCount" in value;
    isInstance = isInstance && "totalCount" in value;
    isInstance = isInstance && "filteredCount" in value;
    isInstance = isInstance && "items" in value;
    return isInstance;
}
export function TransactionsReportsSearchQueryResponseEntityFromJSON(json) {
    return TransactionsReportsSearchQueryResponseEntityFromJSONTyped(json, false);
}
export function TransactionsReportsSearchQueryResponseEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'limit': json['limit'],
        'page': json['page'],
        'pageCount': json['pageCount'],
        'totalCount': json['totalCount'],
        'filteredCount': json['filteredCount'],
        'nextCursor': !exists(json, 'nextCursor') ? undefined : json['nextCursor'],
        'prevCursor': !exists(json, 'prevCursor') ? undefined : json['prevCursor'],
        'items': (json['items'].map(TransactionsReportEntityFromJSON)),
    };
    return removeNullUndefined(typed);
}
export function TransactionsReportsSearchQueryResponseEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'limit': value.limit,
        'page': value.page,
        'pageCount': value.pageCount,
        'totalCount': value.totalCount,
        'filteredCount': value.filteredCount,
        'nextCursor': value.nextCursor,
        'prevCursor': value.prevCursor,
        'items': (value.items.map(TransactionsReportEntityToJSON)),
    };
}
