// @cp/node-sdk/dist/mjs/models/TransactionsRecordEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const TransactionsRecordEntityStatusEnum = {
    Processing: 'processing',
    BuyerApprovalPending: 'buyerApprovalPending',
    Authorized: 'authorized',
    AuthorizationFailed: 'authorizationFailed',
    CaptureFailed: 'captureFailed',
    Declined: 'declined',
    CapturePending: 'capturePending',
    Captured: 'captured',
    VoidPending: 'voidPending',
    Voided: 'voided',
    Hold: 'hold',
    Submitted: 'submitted',
    Transmitted: 'transmitted',
    Settled: 'settled',
    Returned: 'returned',
    Rejected: 'rejected'
};
/**
 * @export
 */
export const TransactionsRecordEntityPaymentMethodEnum = {
    Ach: 'ACH',
    Bank: 'BANK',
    Card: 'CARD'
};
/**
 * @export
 */
export const TransactionsRecordEntityAccountTypeEnum = {
    Checking: 'CHECKING',
    Savings: 'SAVINGS'
};
/**
 * @export
 */
export const TransactionsRecordEntityIntentEnum = {
    Capture: 'capture',
    Authorize: 'authorize',
    Debit: 'debit',
    Credit: 'credit'
};
/**
 * @export
 */
export const TransactionsRecordEntityCurrencyEnum = {
    Usd: 'USD',
    Cad: 'CAD',
    Aud: 'AUD'
};
/**
 * @export
 */
export const TransactionsRecordEntityCardTypeEnum = {
    Credit: 'CREDIT',
    Debit: 'DEBIT',
    Prepaid: 'PREPAID'
};
/**
 * Check if a given object implements the TransactionsRecordEntity interface.
 */
export function instanceOfTransactionsRecordEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "transactionId" in value;
    isInstance = isInstance && "status" in value;
    isInstance = isInstance && "amount" in value;
    isInstance = isInstance && "paymentMethod" in value;
    isInstance = isInstance && "netAmount" in value;
    isInstance = isInstance && "fee" in value;
    isInstance = isInstance && "intent" in value;
    isInstance = isInstance && "currency" in value;
    isInstance = isInstance && "country" in value;
    return isInstance;
}
export function TransactionsRecordEntityFromJSON(json) {
    return TransactionsRecordEntityFromJSONTyped(json, false);
}
export function TransactionsRecordEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': json['merchantId'],
        'createdAt': (new Date(json['createdAt'])),
        'transactionId': json['transactionId'],
        'status': json['status'],
        'amount': json['amount'],
        'paymentMethod': json['paymentMethod'],
        'accountType': !exists(json, 'accountType') ? undefined : json['accountType'],
        'externalTransactionId': !exists(json, 'externalTransactionId') ? undefined : json['externalTransactionId'],
        'netAmount': json['netAmount'],
        'fee': json['fee'],
        'intent': json['intent'],
        'capturedAt': !exists(json, 'capturedAt') ? undefined : (new Date(json['capturedAt'])),
        'authorizedAt': !exists(json, 'authorizedAt') ? undefined : (new Date(json['authorizedAt'])),
        'rejectedAt': !exists(json, 'rejectedAt') ? undefined : (new Date(json['rejectedAt'])),
        'returnedAt': !exists(json, 'returnedAt') ? undefined : (new Date(json['returnedAt'])),
        'settledAt': !exists(json, 'settledAt') ? undefined : (new Date(json['settledAt'])),
        'updatedAt': !exists(json, 'updatedAt') ? undefined : (new Date(json['updatedAt'])),
        'voidedAt': !exists(json, 'voidedAt') ? undefined : (new Date(json['voidedAt'])),
        'capturedAmount': !exists(json, 'capturedAmount') ? undefined : json['capturedAmount'],
        'refundedAmount': !exists(json, 'refundedAmount') ? undefined : json['refundedAmount'],
        'currency': json['currency'],
        'country': json['country'],
        'buyerId': !exists(json, 'buyerId') ? undefined : json['buyerId'],
        'externalBuyerId': !exists(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'cardType': !exists(json, 'cardType') ? undefined : json['cardType'],
        'label': !exists(json, 'label') ? undefined : json['label'],
        'scheme': !exists(json, 'scheme') ? undefined : json['scheme'],
        'expirationDate': !exists(json, 'expirationDate') ? undefined : json['expirationDate'],
        'accountHolderName': !exists(json, 'accountHolderName') ? undefined : json['accountHolderName'],
        'rawResponseCode': !exists(json, 'rawResponseCode') ? undefined : json['rawResponseCode'],
        'rawResponseDescription': !exists(json, 'rawResponseDescription') ? undefined : json['rawResponseDescription'],
        'avsResponseCode': !exists(json, 'avsResponseCode') ? undefined : json['avsResponseCode'],
        'cvvResponseCode': !exists(json, 'cvvResponseCode') ? undefined : json['cvvResponseCode'],
        'sessionId': !exists(json, 'sessionId') ? undefined : json['sessionId'],
        'accountEnding': !exists(json, 'accountEnding') ? undefined : json['accountEnding'],
        'tags': !exists(json, 'tags') ? undefined : json['tags'],
    };
    return removeNullUndefined(typed);
}
export function TransactionsRecordEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'createdAt': (value.createdAt.toISOString()),
        'transactionId': value.transactionId,
        'status': value.status,
        'amount': value.amount,
        'paymentMethod': value.paymentMethod,
        'accountType': value.accountType,
        'externalTransactionId': value.externalTransactionId,
        'netAmount': value.netAmount,
        'fee': value.fee,
        'intent': value.intent,
        'capturedAt': value.capturedAt === undefined ? undefined : (value.capturedAt.toISOString()),
        'authorizedAt': value.authorizedAt === undefined ? undefined : (value.authorizedAt.toISOString()),
        'rejectedAt': value.rejectedAt === undefined ? undefined : (value.rejectedAt.toISOString()),
        'returnedAt': value.returnedAt === undefined ? undefined : (value.returnedAt.toISOString()),
        'settledAt': value.settledAt === undefined ? undefined : (value.settledAt.toISOString()),
        'updatedAt': value.updatedAt === undefined ? undefined : (value.updatedAt.toISOString()),
        'voidedAt': value.voidedAt === undefined ? undefined : (value.voidedAt.toISOString()),
        'capturedAmount': value.capturedAmount,
        'refundedAmount': value.refundedAmount,
        'currency': value.currency,
        'country': value.country,
        'buyerId': value.buyerId,
        'externalBuyerId': value.externalBuyerId,
        'cardType': value.cardType,
        'label': value.label,
        'scheme': value.scheme,
        'expirationDate': value.expirationDate,
        'accountHolderName': value.accountHolderName,
        'rawResponseCode': value.rawResponseCode,
        'rawResponseDescription': value.rawResponseDescription,
        'avsResponseCode': value.avsResponseCode,
        'cvvResponseCode': value.cvvResponseCode,
        'sessionId': value.sessionId,
        'accountEnding': value.accountEnding,
        'tags': value.tags,
    };
}
