// @cp/node-sdk/dist/mjs/models/TransactionRefundEntity.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { TransactionNoteEntityFromJSON, TransactionNoteEntityToJSON, } from './TransactionNoteEntity';
/**
 * @export
 */
export const TransactionRefundEntityStatusEnum = {
    Processing: 'processing',
    Succeeded: 'succeeded',
    Declined: 'declined',
    Failed: 'failed',
    Voided: 'voided',
    Hold: 'hold',
    Submitted: 'submitted',
    Transmitted: 'transmitted',
    Settled: 'settled',
    Returned: 'returned'
};
/**
 * Check if a given object implements the TransactionRefundEntity interface.
 */
export function instanceOfTransactionRefundEntity(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "refundId" in value;
    isInstance = isInstance && "transactionId" in value;
    isInstance = isInstance && "refundedAmount" in value;
    isInstance = isInstance && "createdAt" in value;
    isInstance = isInstance && "updatedAt" in value;
    isInstance = isInstance && "status" in value;
    return isInstance;
}
export function TransactionRefundEntityFromJSON(json) {
    return TransactionRefundEntityFromJSONTyped(json, false);
}
export function TransactionRefundEntityFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': json['merchantId'],
        'refundId': json['refundId'],
        'transactionId': json['transactionId'],
        'refundTransactionId': !exists(json, 'refundTransactionId') ? undefined : json['refundTransactionId'],
        'refundedAmount': json['refundedAmount'],
        'createdAt': (new Date(json['createdAt'])),
        'updatedAt': (new Date(json['updatedAt'])),
        'status': json['status'],
        'reason': !exists(json, 'reason') ? undefined : json['reason'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'currency': !exists(json, 'currency') ? undefined : json['currency'],
        'notes': !exists(json, 'notes') ? undefined : (json['notes'].map(TransactionNoteEntityFromJSON)),
    };
    return removeNullUndefined(typed);
}
export function TransactionRefundEntityToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'refundId': value.refundId,
        'transactionId': value.transactionId,
        'refundTransactionId': value.refundTransactionId,
        'refundedAmount': value.refundedAmount,
        'createdAt': (value.createdAt.toISOString()),
        'updatedAt': (value.updatedAt.toISOString()),
        'status': value.status,
        'reason': value.reason,
        'description': value.description,
        'currency': value.currency,
        'notes': value.notes === undefined ? undefined : (value.notes.map(TransactionNoteEntityToJSON)),
    };
}
