// @cp/node-sdk/dist/mjs/models/UpdateMerchantCredentialsDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the UpdateMerchantCredentialsDto interface.
 */
export function instanceOfUpdateMerchantCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "merchantCredentialsId" in value;
    isInstance = isInstance && "credentialsName" in value;
    return isInstance;
}
export function UpdateMerchantCredentialsDtoFromJSON(json) {
    return UpdateMerchantCredentialsDtoFromJSONTyped(json, false);
}
export function UpdateMerchantCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantCredentialsId': json['merchantCredentialsId'],
        'credentialsName': json['credentialsName'],
    };
    return removeNullUndefined(typed);
}
export function UpdateMerchantCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantCredentialsId': value.merchantCredentialsId,
        'credentialsName': value.credentialsName,
    };
}
