// @cp/node-sdk/dist/mjs/models/UpdateMerchantAccountDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateCFeeDtoFromJSON, CreateCFeeDtoToJSON, } from './CreateCFeeDto';
/**
 * @export
 */
export const UpdateMerchantAccountDtoPspEnum = {
    Tsys: 'TSYS',
    Nashville: 'NASHVILLE',
    North: 'NORTH',
    ChannelPayments: 'CHANNEL_PAYMENTS'
};
/**
 * @export
 */
export const UpdateMerchantAccountDtoCardTypesEnum = {
    Credit: 'CREDIT',
    Debit: 'DEBIT',
    Prepaid: 'PREPAID'
};
/**
 * @export
 */
export const UpdateMerchantAccountDtoCardSchemesEnum = {
    Visa: 'VISA',
    Mastercard: 'MASTERCARD',
    Amex: 'AMEX',
    Discover: 'DISCOVER',
    Jcb: 'JCB',
    DinersClub: 'DINERS_CLUB',
    Maestro: 'MAESTRO'
};
/**
 * @export
 */
export const UpdateMerchantAccountDtoCurrenciesEnum = {
    Usd: 'USD',
    Aud: 'AUD',
    Cad: 'CAD'
};
/**
 * Check if a given object implements the UpdateMerchantAccountDto interface.
 */
export function instanceOfUpdateMerchantAccountDto(value) {
    let isInstance = true;
    return isInstance;
}
export function UpdateMerchantAccountDtoFromJSON(json) {
    return UpdateMerchantAccountDtoFromJSONTyped(json, false);
}
export function UpdateMerchantAccountDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'psp': !exists(json, 'psp') ? undefined : json['psp'],
        'cFee': !exists(json, 'cFee') ? undefined : CreateCFeeDtoFromJSON(json['cFee']),
        'cardTypes': !exists(json, 'cardTypes') ? undefined : json['cardTypes'],
        'cardSchemes': !exists(json, 'cardSchemes') ? undefined : json['cardSchemes'],
        'currencies': !exists(json, 'currencies') ? undefined : json['currencies'],
        'mcc': !exists(json, 'mcc') ? undefined : json['mcc'],
        'bin': !exists(json, 'bin') ? undefined : json['bin'],
        'chain': !exists(json, 'chain') ? undefined : json['chain'],
        'store': !exists(json, 'store') ? undefined : json['store'],
        'agent': !exists(json, 'agent') ? undefined : json['agent'],
        'isDefault': !exists(json, 'isDefault') ? undefined : json['isDefault'],
        'lastPolledAt': !exists(json, 'lastPolledAt') ? undefined : json['lastPolledAt'],
    };
    return removeNullUndefined(typed);
}
export function UpdateMerchantAccountDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'psp': value.psp,
        'cFee': CreateCFeeDtoToJSON(value.cFee),
        'cardTypes': value.cardTypes,
        'cardSchemes': value.cardSchemes,
        'currencies': value.currencies,
        'mcc': value.mcc,
        'bin': value.bin,
        'chain': value.chain,
        'store': value.store,
        'agent': value.agent,
        'isDefault': value.isDefault,
        'lastPolledAt': value.lastPolledAt,
    };
}
