// @cp/node-sdk/dist/mjs/models/UpdateFeatureDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the UpdateFeatureDto interface.
 */
export function instanceOfUpdateFeatureDto(value) {
    let isInstance = true;
    return isInstance;
}
export function UpdateFeatureDtoFromJSON(json) {
    return UpdateFeatureDtoFromJSONTyped(json, false);
}
export function UpdateFeatureDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'description': !exists(json, 'description') ? undefined : json['description'],
        'version': !exists(json, 'version') ? undefined : json['version'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
    };
    return removeNullUndefined(typed);
}
export function UpdateFeatureDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'description': value.description,
        'version': value.version,
        'enabled': value.enabled,
    };
}
