// @cp/node-sdk/dist/mjs/models/UpdateTransactionsReportsTemplateDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateTransactionsReportsFiltersDtoFromJSON, CreateTransactionsReportsFiltersDtoToJSON, } from './CreateTransactionsReportsFiltersDto';
/**
 * @export
 */
export const UpdateTransactionsReportsTemplateDtoOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * @export
 */
export const UpdateTransactionsReportsTemplateDtoDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * Check if a given object implements the UpdateTransactionsReportsTemplateDto interface.
 */
export function instanceOfUpdateTransactionsReportsTemplateDto(value) {
    let isInstance = true;
    isInstance = isInstance && "reportTemplateId" in value;
    return isInstance;
}
export function UpdateTransactionsReportsTemplateDtoFromJSON(json) {
    return UpdateTransactionsReportsTemplateDtoFromJSONTyped(json, false);
}
export function UpdateTransactionsReportsTemplateDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'order': !exists(json, 'order') ? undefined : json['order'],
        'dateRange': !exists(json, 'dateRange') ? undefined : json['dateRange'],
        'startTime': !exists(json, 'startTime') ? undefined : json['startTime'],
        'endTime': !exists(json, 'endTime') ? undefined : json['endTime'],
        'columns': !exists(json, 'columns') ? undefined : json['columns'],
        'transactionsFilters': !exists(json, 'transactionsFilters') ? undefined : CreateTransactionsReportsFiltersDtoFromJSON(json['transactionsFilters']),
        'reportTemplateId': json['reportTemplateId'],
        'isFavorite': !exists(json, 'isFavorite') ? undefined : json['isFavorite'],
    };
    return removeNullUndefined(typed);
}
export function UpdateTransactionsReportsTemplateDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'order': value.order,
        'dateRange': value.dateRange,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'columns': value.columns,
        'transactionsFilters': CreateTransactionsReportsFiltersDtoToJSON(value.transactionsFilters),
        'reportTemplateId': value.reportTemplateId,
        'isFavorite': value.isFavorite,
    };
}
