// @cp/node-sdk/dist/mjs/models/UpdateTransactionsReportsScheduleDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateTransactionsReportsFiltersDtoFromJSON, CreateTransactionsReportsFiltersDtoToJSON, } from './CreateTransactionsReportsFiltersDto';
/**
 * @export
 */
export const UpdateTransactionsReportsScheduleDtoIntervalEnum = {
    Daily: 'DAILY',
    Weekly: 'WEEKLY',
    Monthly: 'MONTHLY'
};
/**
 * @export
 */
export const UpdateTransactionsReportsScheduleDtoDateRangeEnum = {
    Today: 'TODAY',
    Yesterday: 'YESTERDAY',
    Last7Days: 'LAST_7_DAYS',
    Last30Days: 'LAST_30_DAYS',
    Last60Days: 'LAST_60_DAYS',
    Last90Days: 'LAST_90_DAYS',
    LastWeek: 'LAST_WEEK',
    LastMonth: 'LAST_MONTH',
    LastQuarter: 'LAST_QUARTER',
    LastYear: 'LAST_YEAR',
    Custom: 'CUSTOM'
};
/**
 * @export
 */
export const UpdateTransactionsReportsScheduleDtoOrderEnum = {
    Asc: 'asc',
    Desc: 'desc'
};
/**
 * Check if a given object implements the UpdateTransactionsReportsScheduleDto interface.
 */
export function instanceOfUpdateTransactionsReportsScheduleDto(value) {
    let isInstance = true;
    isInstance = isInstance && "reportScheduleId" in value;
    return isInstance;
}
export function UpdateTransactionsReportsScheduleDtoFromJSON(json) {
    return UpdateTransactionsReportsScheduleDtoFromJSONTyped(json, false);
}
export function UpdateTransactionsReportsScheduleDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'name': !exists(json, 'name') ? undefined : json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'interval': !exists(json, 'interval') ? undefined : json['interval'],
        'dateRange': !exists(json, 'dateRange') ? undefined : json['dateRange'],
        'recipients': !exists(json, 'recipients') ? undefined : json['recipients'],
        'order': !exists(json, 'order') ? undefined : json['order'],
        'startTime': !exists(json, 'startTime') ? undefined : json['startTime'],
        'endTime': !exists(json, 'endTime') ? undefined : json['endTime'],
        'nextRun': !exists(json, 'nextRun') ? undefined : json['nextRun'],
        'columns': !exists(json, 'columns') ? undefined : json['columns'],
        'transactionsFilters': !exists(json, 'transactionsFilters') ? undefined : CreateTransactionsReportsFiltersDtoFromJSON(json['transactionsFilters']),
        'reportScheduleId': json['reportScheduleId'],
    };
    return removeNullUndefined(typed);
}
export function UpdateTransactionsReportsScheduleDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'name': value.name,
        'description': value.description,
        'interval': value.interval,
        'dateRange': value.dateRange,
        'recipients': value.recipients,
        'order': value.order,
        'startTime': value.startTime,
        'endTime': value.endTime,
        'nextRun': value.nextRun,
        'columns': value.columns,
        'transactionsFilters': CreateTransactionsReportsFiltersDtoToJSON(value.transactionsFilters),
        'reportScheduleId': value.reportScheduleId,
    };
}
