// @cp/node-sdk/dist/mjs/models/UpdateWebhookDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateEventDomainDtoFromJSON, CreateEventDomainDtoToJSON, } from './CreateEventDomainDto';
/**
 * Check if a given object implements the UpdateWebhookDto interface.
 */
export function instanceOfUpdateWebhookDto(value) {
    let isInstance = true;
    isInstance = isInstance && "webhookId" in value;
    return isInstance;
}
export function UpdateWebhookDtoFromJSON(json) {
    return UpdateWebhookDtoFromJSONTyped(json, false);
}
export function UpdateWebhookDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'merchantId': !exists(json, 'merchantId') ? undefined : json['merchantId'],
        'url': !exists(json, 'url') ? undefined : json['url'],
        'eventDomains': !exists(json, 'eventDomains') ? undefined : CreateEventDomainDtoFromJSON(json['eventDomains']),
        'name': !exists(json, 'name') ? undefined : json['name'],
        'description': !exists(json, 'description') ? undefined : json['description'],
        'username': !exists(json, 'username') ? undefined : json['username'],
        'password': !exists(json, 'password') ? undefined : json['password'],
        'webhookId': json['webhookId'],
        'enabled': !exists(json, 'enabled') ? undefined : json['enabled'],
    };
    return removeNullUndefined(typed);
}
export function UpdateWebhookDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'merchantId': value.merchantId,
        'url': value.url,
        'eventDomains': CreateEventDomainDtoToJSON(value.eventDomains),
        'name': value.name,
        'description': value.description,
        'username': value.username,
        'password': value.password,
        'webhookId': value.webhookId,
        'enabled': value.enabled,
    };
}
