// @cp/node-sdk/dist/mjs/models/UpdateUserDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const UpdateUserDtoRoleEnum = {
    User: 'user',
    Agent: 'agent',
    Admin: 'admin',
    SuperAdmin: 'superAdmin'
};
/**
 * @export
 */
export const UpdateUserDtoStatusEnum = {
    Invited: 'invited',
    Active: 'active',
    Inactive: 'inactive',
    Blocked: 'blocked'
};
/**
 * Check if a given object implements the UpdateUserDto interface.
 */
export function instanceOfUpdateUserDto(value) {
    let isInstance = true;
    isInstance = isInstance && "userId" in value;
    return isInstance;
}
export function UpdateUserDtoFromJSON(json) {
    return UpdateUserDtoFromJSONTyped(json, false);
}
export function UpdateUserDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': !exists(json, 'givenName') ? undefined : json['givenName'],
        'familyName': !exists(json, 'familyName') ? undefined : json['familyName'],
        'role': !exists(json, 'role') ? undefined : json['role'],
        'phoneNumber': !exists(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'userId': json['userId'],
        'status': !exists(json, 'status') ? undefined : json['status'],
    };
    return removeNullUndefined(typed);
}
export function UpdateUserDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'role': value.role,
        'phoneNumber': value.phoneNumber,
        'userId': value.userId,
        'status': value.status,
    };
}
