// @cp/node-sdk/dist/mjs/models/UpdateMerchantConnectorCredentialsDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the UpdateMerchantConnectorCredentialsDto interface.
 */
export function instanceOfUpdateMerchantConnectorCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "credentialsId" in value;
    return isInstance;
}
export function UpdateMerchantConnectorCredentialsDtoFromJSON(json) {
    return UpdateMerchantConnectorCredentialsDtoFromJSONTyped(json, false);
}
export function UpdateMerchantConnectorCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'username': !exists(json, 'username') ? undefined : json['username'],
        'password': !exists(json, 'password') ? undefined : json['password'],
        'credentialsId': json['credentialsId'],
    };
    return removeNullUndefined(typed);
}
export function UpdateMerchantConnectorCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'username': value.username,
        'password': value.password,
        'credentialsId': value.credentialsId,
    };
}
