// @cp/node-sdk/dist/mjs/models/UpdatePaymentMethodStatusDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * @export
 */
export const UpdatePaymentMethodStatusDtoStatusEnum = {
    Processing: 'processing',
    ApprovalRequired: 'approvalRequired',
    Succeeded: 'succeeded',
    Failed: 'failed',
    Expired: 'expired'
};
/**
 * Check if a given object implements the UpdatePaymentMethodStatusDto interface.
 */
export function instanceOfUpdatePaymentMethodStatusDto(value) {
    let isInstance = true;
    isInstance = isInstance && "paymentMethodId" in value;
    isInstance = isInstance && "merchantId" in value;
    isInstance = isInstance && "status" in value;
    return isInstance;
}
export function UpdatePaymentMethodStatusDtoFromJSON(json) {
    return UpdatePaymentMethodStatusDtoFromJSONTyped(json, false);
}
export function UpdatePaymentMethodStatusDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'paymentMethodId': json['paymentMethodId'],
        'merchantId': json['merchantId'],
        'status': json['status'],
    };
    return removeNullUndefined(typed);
}
export function UpdatePaymentMethodStatusDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'paymentMethodId': value.paymentMethodId,
        'merchantId': value.merchantId,
        'status': value.status,
    };
}
