// @cp/node-sdk/dist/mjs/models/UpdateBuyerDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
import { CreateBuyerAddressDtoFromJSON, CreateBuyerAddressDtoToJSON, } from './CreateBuyerAddressDto';
/**
 * Check if a given object implements the UpdateBuyerDto interface.
 */
export function instanceOfUpdateBuyerDto(value) {
    let isInstance = true;
    isInstance = isInstance && "buyerId" in value;
    return isInstance;
}
export function UpdateBuyerDtoFromJSON(json) {
    return UpdateBuyerDtoFromJSONTyped(json, false);
}
export function UpdateBuyerDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'givenName': !exists(json, 'givenName') ? undefined : json['givenName'],
        'familyName': !exists(json, 'familyName') ? undefined : json['familyName'],
        'email': !exists(json, 'email') ? undefined : json['email'],
        'address': !exists(json, 'address') ? undefined : CreateBuyerAddressDtoFromJSON(json['address']),
        'phoneNumber': !exists(json, 'phoneNumber') ? undefined : json['phoneNumber'],
        'externalBuyerId': !exists(json, 'externalBuyerId') ? undefined : json['externalBuyerId'],
        'buyerId': json['buyerId'],
    };
    return removeNullUndefined(typed);
}
export function UpdateBuyerDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'givenName': value.givenName,
        'familyName': value.familyName,
        'email': value.email,
        'address': CreateBuyerAddressDtoToJSON(value.address),
        'phoneNumber': value.phoneNumber,
        'externalBuyerId': value.externalBuyerId,
        'buyerId': value.buyerId,
    };
}
