// @cp/node-sdk/dist/mjs/models/UpdateFeatureTargetDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { exists, removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the UpdateFeatureTargetDto interface.
 */
export function instanceOfUpdateFeatureTargetDto(value) {
    let isInstance = true;
    return isInstance;
}
export function UpdateFeatureTargetDtoFromJSON(json) {
    return UpdateFeatureTargetDtoFromJSONTyped(json, false);
}
export function UpdateFeatureTargetDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'rolloutPercentage': !exists(json, 'rolloutPercentage') ? undefined : json['rolloutPercentage'],
    };
    return removeNullUndefined(typed);
}
export function UpdateFeatureTargetDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'rolloutPercentage': value.rolloutPercentage,
    };
}
