// @cp/node-sdk/dist/mjs/models/ValidateMerchantCredentialsDto.js
/* tslint:disable */
/* eslint-disable */
/**
 * Channel Payments API
 *
 * NOTE: This class is auto generated. Do not edit the class manually.
 *
 */
import { removeNullUndefined } from '../runtime';
/**
 * Check if a given object implements the ValidateMerchantCredentialsDto interface.
 */
export function instanceOfValidateMerchantCredentialsDto(value) {
    let isInstance = true;
    isInstance = isInstance && "accessToken" in value;
    return isInstance;
}
export function ValidateMerchantCredentialsDtoFromJSON(json) {
    return ValidateMerchantCredentialsDtoFromJSONTyped(json, false);
}
export function ValidateMerchantCredentialsDtoFromJSONTyped(json, ignoreDiscriminator) {
    if ((json === undefined) || (json === null)) {
        return json;
    }
    const typed = {
        'accessToken': json['accessToken'],
    };
    return removeNullUndefined(typed);
}
export function ValidateMerchantCredentialsDtoToJSON(value) {
    if (value === undefined) {
        return undefined;
    }
    if (value === null) {
        return null;
    }
    return {
        'accessToken': value.accessToken,
    };
}
